<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>C075: Annotating the Grant (C075) retinal epithelial cells data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2020-08-28"/>
  <meta property="article:created" itemprop="dateCreated" content="2020-08-28"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C075: Annotating the Grant (C075) retinal epithelial cells data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C075"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C075: Annotating the Grant (C075) retinal epithelial cells data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Computational assignment of cell-cycle stage from single-cell transcriptome data;citation_publication_date=2015;citation_volume=85;citation_author=A. Scialdone;citation_author=K. N. Natarajan;citation_author=L. R. Saraiva;citation_author=V. Proserpio;citation_author=S. A. Teichmann;citation_author=O. Stegle;citation_author=J. C. Marioni;citation_author=F. Buettner"/>
  <meta name="citation_reference" content="citation_title=Bias, robustness and scalability in single-cell differential expression analysis;citation_publication_date=2018;citation_volume=15;citation_author=C. Soneson;citation_author=M. D. Robinson"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Annotating the Grant (C075) retinal epithelial cells data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Single Cell Open Research Endeavour (SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2020-08-28"]},{"type":"character","attributes":{},"value":["distill::distill_article"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .custom p {
    margin-bottom: 0.5em;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
  <script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.14/datatables.js"></script>
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Annotating the Grant (C075) retinal epithelial cells data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Single Cell Open Research Endeavour (SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score"}],"publishedDate":"2020-08-28T00:00:00.000+10:00","citationText":"Hickey, 2020"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C075</a>
<a href="index.html">Introduction</a>
<a href="C075_Grant_Coultas.preprocess.html">Preprocessing</a>
<a href="C075_Grant_Coultas.cell_selection.html">Selecting biologically relevant cells</a>
<a href="C075_Grant_Coultas.annotate.html">Annotating cells</a>
<a href="C075_Grant_Coultas.multi-sample_comparisons.html">Differential analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/zoes_marker_genes/">Zoe's marker gene tables</a>
<a href="../output/cluster_marker_genes/">Cluster marker gene tables</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
<a href="../output/glimma-plots/">Glimma plots</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C075_Grant_Coultas">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Annotating the Grant (C075) retinal epithelial cells data set</h1>

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Single Cell Open Research Endeavour (SCORE), WEHI)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2020-08-28
</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#preparing-the-data">Preparing the data</a></li>
<li><a href="#motivation">Motivation</a></li>
<li><a href="#pre-defined-marker-genes">Pre-defined marker genes</a><ul>
<li><a href="#zoes-gene-list">Zoe’s gene list</a><ul>
<li><a href="#motivation-1">Motivation</a></li>
<li><a href="#analysis">Analysis</a></li>
</ul></li>
<li><a href="#facs-markers">FACS markers</a></li>
</ul></li>
<li><a href="#cluster-marker-gene-detection">Cluster marker gene detection</a><ul>
<li><a href="#motivation-2">Motivation</a></li>
<li><a href="#statistical-methodology-and-considerations">Statistical methodology and considerations</a><ul>
<li><a href="#gene-lists">Gene lists</a></li>
<li><a href="#use-of-pairwise-comparisons">Use of pairwise comparisons</a></li>
<li><a href="#blocking">Blocking</a></li>
<li><a href="#choice-of-test-statistic">Choice of test statistic</a></li>
<li><a href="#direction-and-magnitude-of-the-log-fold-change">Direction and magnitude of the log-fold change</a></li>
<li><a href="#choice-of-combining-pairwise-de-results-into-a-marker-list">Choice of combining pairwise DE results into a marker list</a></li>
</ul></li>
<li><a href="#analysis-1">Analysis</a><ul>
<li><a href="#t_up_all"><code>t_up_all</code></a></li>
<li><a href="#t_up_some"><code>t_up_some</code></a></li>
<li><a href="#t_any_any"><code>t_any_any</code></a></li>
<li><a href="#targeted">Targeted</a></li>
</ul></li>
</ul></li>
<li><a href="#cell-cycle-assignment">Cell cycle assignment</a><ul>
<li><a href="#using-the-cyclins">Using the cyclins</a></li>
<li><a href="#using-reference-profiles">Using reference profiles</a></li>
<li><a href="#using-the-cyclone-classifier">Using the <code>cyclone()</code> classifier</a></li>
<li><a href="#summary">Summary</a></li>
</ul></li>
<li><a href="#visualization">Visualization</a><ul>
<li><a href="#choice-of-expression-values-logcounts-vs.-corrected">Choice of expression values: <code>logcounts</code> vs. <code>corrected</code></a></li>
<li><a href="#limitations-of-the-t_up_all-strategy">Limitations of the <code>t_up_all</code> strategy</a></li>
<li><a href="#exclusion-of-certain-types-of-genes">Exclusion of certain types of genes</a></li>
<li><a href="#choice-of-gradient-colour-scale">Choice of gradient colour scale</a></li>
<li><a href="#making-your-own-umap-plot-coloured-by-gene-expression">Making your own UMAP plot coloured by gene expression</a></li>
</ul></li>
<li><a href="#summary-1">Summary</a></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="preparing-the-data">Preparing the data</h1>
<p>We start from the preprocessed <em>SingleCellExperiment</em> object created in <a href="C075_Grant_Coultas.cell_selection.html">‘Selection of biologically relevant cells for the Grant (C075) retinal epithelial cells data set’</a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="motivation">Motivation</h1>
<p>The most challenging task in scRNA-seq data analysis is arguably the annotation of cells with meaningful labels. We might attempt to label cells individually or to label clusters (with cells assigned to those clusters ‘inheriting’ the cluster-level label).</p>
<p>In <a href="C075_Grant_Coultas.cell_selection.html">‘Selection of biologically relevant cells for the Grant (C075) retinal epithelial cells data set’</a>, we obtained one set of labels:</p>
<ol type="1">
<li>‘cluster’ labels using clustering. Obtaining clusters of cells is fairly straightforward, but it is more difficult to determine what biological state is represented by each of those clusters.</li>
</ol>
<p>In this section we use <a href="#pre-defined-marker-genes">Pre-defined marker genes</a> and perform <a href="#cluster-marker-gene-detection">Cluster marker gene detection</a> to construct <em>de novo</em> marker gene lists that can be used to aid the interpretation of the clustering results. The <em>de novo</em> marker genes lists use a data driven approach to identify those genes that drive separation between clusters whereas the pre-defined marker genes are a more targeted approach to cluster annotation. These two approaches complement one another.</p>
<p>As a reminder, Figure <a href="#fig:clusterplot-umap">1</a> shows the clustering results overlaid on the UMAP plot and Figure <a href="#fig:cluster-barplot">2</a> breaks down the clusters by key experimental factors.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:clusterplot-umap"></span>
<img src="C075_Grant_Coultas.annotate_files/clusterplot-umap-1.png" alt="UMAP plot, where each point represents a cell and is coloured according to the legend." width="624" />
<p class="caption">
Figure 1: UMAP plot, where each point represents a cell and is coloured according to the legend.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cluster-barplot"></span>
<img src="C075_Grant_Coultas.annotate_files/cluster-barplot-1.png" alt="Breakdown of clusters by experimental factors." width="624" />
<p class="caption">
Figure 2: Breakdown of clusters by experimental factors.
</p>
</div>
</div>
<p>We conclude by performing <a href="#cell-cycle-assignment">Cell cycle assignment</a> and by describing the choice of <a href="#visualization">Visualization</a> options for the plots used in this report, as well instructions for how to make your own plots.</p>
<h1 id="pre-defined-marker-genes">Pre-defined marker genes</h1>
<p>We examine <a href="#zoes-gene-list">Zoe’s gene list</a> and also perform a simple visualization of the <a href="#facs-markers">FACS markers</a>.</p>
<h2 id="zoes-gene-list">Zoe’s gene list</h2>
<h3 id="motivation-1">Motivation</h3>
<p>Zoe sent me a list of marker genes to help characterise the cells<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-de80daf81e596f644f7c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-de80daf81e596f644f7c">{"x":{"filter":"none","caption":"<caption>Zoe's supplied list of marker genes, their 'Endothelial identity', the reference (if supplied), and whether they are expressed in the current dataset.<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84"],["Sox17","Gja4","Vegfc","Sema3g","Cxcl12","Ptn","Sat1","Efnb2","Cxcr4","Dll4","Esm1","Angpt2","Apln","Flt4","Adm","Col4a1","Ednrb","Igfbp3","Lama4","Lamb1","Mcam","Nid2","Trp53i11","Rhoc","Aplnr","Prcp","Bgn","Slc38a5","Nr2f2","Gatm","Apod","Hspg2","Vwa1","Eng","Egr1","Kdr","Rgcc","Sgk1","Cd200","BC028528","Edn1","Epas1","Cav1","Spock2","Sparc","Sparcl1","Ccne1","Ccne2","Ccna2","Ccnb1","Ccnb2","Ccnd1","Ccnd2","Ccnd3","Jag1","Nrarp","Hes1","Hey1","Vegfr1","Ackr1","Aqp1","C1qtnf9","Cd36","Csrp","Ehd4","Fbln5","Hsbp1","Iigp1","Il6st","Jam2","Lgals3","Lrg1","Meox2","Plscr2","Sdpr","Selp","Spint2","Tgfbi","Tgm2","Tmem176a","Tmem176b","Tmem252","Tspan7","Vwf"],["Artery","Artery","Artery","Artery","Artery","Artery","Artery","Artery, tip","Artery, tip","Artery, tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Tip","Vein","Vein","Vein","Vein","Vein","Vein","Vein","Immature","Immature","Immature","Capillary","Capillary","Capillary","Capillary","Capillary","Capillary","Capillary","Capillary","Capillary","Capillary","Capillary","Angiogenic","Cell cycle","Cell cycle","Cell cycle","Cell cycle","Cell cycle","Cell cycle","Cell cycle","Cell cycle","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem","Stem"],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,false,true,true,true,true,false,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Strasser 2010","Strasser 2010","Strasser 2010","Strasser 2010","Strasser 2010","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018","Zhao 2018"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>Endothelial identity<\/th>\n      <th>Expressed<\/th>\n      <th>Reference<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<h3 id="analysis">Analysis</h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:zoes-markers-heatmap">3</a> is a heatmap of Zoe’s marker genes. It is challenging to immediately associate clusters with a cell type on the basis of these marker genes, but some patterns are clear, such as:</p>
<ul>
<li><code>Artery</code> marker genes are highly expressed in cluster 7.</li>
<li><code>Artery, tip</code> marker genes are highly expressed in clusters 5 and 7.</li>
<li><code>Tip</code> marker genes are highly expressed in cluster 5.</li>
</ul>
<p>The expressions patterns for the other marker gene sets are more complex, such as a subset of the gene set being not expressed, restricted to particular cluster(s), or ubiquitously expressed across clusters.</p>
<aside>
Violin plots of per-gene expression measurements and heatmaps (<code>logcounts</code> and <code>reconstructed</code>) can be found in the <a href="../output/zoes_marker_genes/"><code>output/zoes_marker_genes/</code></a> directory.
</aside>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:zoes-markers-heatmap"></span>
<img src="C075_Grant_Coultas.annotate_files/zoes-markers-heatmap-1.png" alt="Heatmap of row-normalized `logcounts` for Zoe's marker genes. Each column is a sample, each row a gene. Samples are ordered by `cluster` then by `genotype`." width="624" />
<p class="caption">
Figure 3: Heatmap of row-normalized <code>logcounts</code> for Zoe’s marker genes. Each column is a sample, each row a gene. Samples are ordered by <code>cluster</code> then by <code>genotype</code>.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h2 id="facs-markers">FACS markers</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Several FACS markers were collected for these cells: FSC_A, FSC_W, FSC_H, SSC_A, SSC_W, SSC_H, R660_20_A_APC, B530_30_A_GFP, Y582_15_A_td_Tomato, V450_50_A_DAPI, and Y780_60_A_PE_Cy7.</p>
<p>Absolute FACS measurements are generally not comparable across samples for a number of reasons<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> and normalization methods are underdeveloped. We apply a simple ‘scaled rank’ normalization method, which is to rank within each plate the FACS measurements between 0 and 1 (0 being the lowest and 1 being the highest). This is a very drastic normalization that removes much of the quantitative information potentially available in the FACS data.</p>
<aside>
Any sort of sophisticated analysis of the FACS data is very challenging because of this lack of standardization in data collection. The normalization of this data and integration with scRNA-seq data is a research question that is beyond the scope of this analysis.
</aside>
<p>Figure <a href="#fig:umap-with-facs">4</a> presents a simple summary of the FACS data by overlaying it on the UMAP plot.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:umap-with-facs"></span>
<img src="C075_Grant_Coultas.annotate_files/umap-with-facs-1.png" alt="Overlay of index sorting data on UMAP plot. For each marker, the left-hand plot shows the 'raw' or 'pseudo-logged' fluorescence intensity and the right-side plots the 'scaled rank' of the raw intensity. The pseudo-log transformation is a transformation mapping numbers to a signed logarithmic scale with a smooth transition to linear scale around 0. This transformation is commonly used when plotting fluorescence intensities from FACS. The scaled rank is applied within each `plate_number` and assigns the maximum fluorescence intensity a value of one and the minimum fluorescence intensities a value of zero. It can be thought of as a crude normalization of the FACS data that allows us to compare fluorescence intensities from different plates." width="624" />
<p class="caption">
Figure 4: Overlay of index sorting data on UMAP plot. For each marker, the left-hand plot shows the ‘raw’ or ‘pseudo-logged’ fluorescence intensity and the right-side plots the ‘scaled rank’ of the raw intensity. The pseudo-log transformation is a transformation mapping numbers to a signed logarithmic scale with a smooth transition to linear scale around 0. This transformation is commonly used when plotting fluorescence intensities from FACS. The scaled rank is applied within each <code>plate_number</code> and assigns the maximum fluorescence intensity a value of one and the minimum fluorescence intensities a value of zero. It can be thought of as a crude normalization of the FACS data that allows us to compare fluorescence intensities from different plates.
</p>
</div>
</div>
<h1 id="cluster-marker-gene-detection">Cluster marker gene detection</h1>
<h2 id="motivation-2">Motivation</h2>
<p>To interpret our clustering results, we identify the genes that drive separation between clusters. These marker genes allow us to assign biological meaning to each cluster based on their functional annotation. In the most obvious case, the marker genes for each cluster are <em>a priori</em> associated with particular cell types, allowing us to treat the clustering as a proxy for cell type identity. The same principle can be applied to more subtle differences in activation status or differentiation state.</p>
<p>Identification of marker genes is usually based around the retrospective detection of differential expression between clusters<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>. Genes that are more strongly DE are more likely to have driven cluster separation in the first place. The top DE genes are likely to be good candidate markers as they can effectively distinguish between cells in different clusters.</p>
<h2 id="statistical-methodology-and-considerations">Statistical methodology and considerations</h2>
<p>It is important to have some understanding of the statistical methodology used for marker gene detection. This section gives an overview.</p>
<h3 id="gene-lists">Gene lists</h3>
<p>For each cluster, the DE results of the relevant comparisons are consolidated into a single output table. This allows a set of marker genes to be easily defined by taking the top DE genes from each pairwise comparison between clusters. Other statistics are also reported for each gene, including the adjusted p-values<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> and the log-fold changes relative to every other cluster.</p>
<h3 id="use-of-pairwise-comparisons">Use of pairwise comparisons</h3>
<p>We intentionally use pairwise comparisons between clusters rather than comparing each cluster to the average of all other cells. The latter approach is sensitive to the population composition, potentially resulting in wildly different sets of markers when cell type abundances change in different contexts. In the worst case, the presence of a single dominant subpopulation will drive the selection of top markers for every other cluster, pushing out useful genes that can resolve the various minor subpopulations. Moreover, pairwise comparisons naturally provide more information to interpret of the utility of a marker, e.g., by providing log-fold changes to indicate which clusters are distinguished by this gene.</p>
<h3 id="blocking">Blocking</h3>
<p>We perform intra-plate comparisons by blocking on the <code>plate_number</code> to avoid confounding effects from differential expression between plates<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="choice-of-test-statistic">Choice of test statistic</h3>
<p>The Welch <span class="math inline">\(t\)</span>-test is an obvious choice of statistical method to test for differences in expression between clusters. It is quickly computed and has good statistical properties for large numbers of cells <span class="citation" data-cites="soneson2018bias">(Soneson and Robinson <a href="#ref-soneson2018bias" role="doc-biblioref">2018</a>)</span>.</p>
<p>Alternatively, we could consider the Wilcoxon rank sum test (also known as the Wilcoxon-Mann-Whitney test, or WMW test). Its strength lies in the fact that it directly assesses separation between the expression distributions of different clusters. The WMW test statistic is proportional to the area-under-the-curve (AUC), i.e., the concordance probability, which is the probability of a random cell from one cluster having higher expression than a random cell from another cluster. In a pairwise comparison, AUCs of 1 or 0 indicate that the two clusters have perfectly separated expression distributions. Thus, the WMW test directly addresses the most desirable property of a candidate marker gene, while the <span class="math inline">\(t\)</span>-test only does so indirectly via the difference in the means and the intra-group variance.</p>
<p>Another alternative, the binomial test, identifies genes that differ in the proportion of expressing cells between clusters<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>. This represents a much more stringent definition of marker genes compared to the other methods, as differences in expression between clusters are effectively ignored if both distributions of expression values are not near zero. The premise is that genes are more likely to contribute to important biological decisions if they were active in one cluster and silent in another, compared to more subtle ‘tuning’ effects from changing the expression of an active gene. From a practical perspective, a binary measure of presence/absence is easier to validate.</p>
<p>The Welch <span class="math inline">\(t\)</span>-test is our default method for identifying marker genes.</p>
<h3 id="direction-and-magnitude-of-the-log-fold-change">Direction and magnitude of the log-fold change</h3>
<p>There are 3 ways of specifying the direction of the log-fold change used in the statistical test when comparing cluster <span class="math inline">\(X\)</span> to cluster <span class="math inline">\(Y\)</span>:</p>
<ol type="1">
<li><code>direction = "any"</code>: Identify genes that are upregulated or downregulated in <span class="math inline">\(X\)</span> compared to <span class="math inline">\(Y\)</span>.</li>
<li><code>direction = "up"</code>: Identify genes that are upregulated in <span class="math inline">\(X\)</span> compared to <span class="math inline">\(Y\)</span>.</li>
<li><code>direction = "down"</code>: Identify genes that are downregulated in <span class="math inline">\(X\)</span> compared to <span class="math inline">\(Y\)</span>.</li>
</ol>
<p>Generally speaking, downregulated genes are less appealing as markers as it is more difficult to interpret and experimentally validate an absence of expression. We therefore mostly focus on identifying genes that are upregulated (i.e. <code>direction = "up"</code>) in the chosen cluster relative to any/some/all other clusters . Of course, this increased stringency is not without cost. If only upregulated genes are requested then any cluster defined by downregulation of a marker gene will not contain that gene among the top set of features in its gene list. This is occasionally relevant for subtypes or other states that are distinguished by high versus low expression of particular genes.</p>
<h3 id="choice-of-combining-pairwise-de-results-into-a-marker-list">Choice of combining pairwise DE results into a marker list</h3>
<p>There are 3 ways of combining the pairwise DE results to obtain the per-cluster marker gene lists:</p>
<ol type="1">
<li><a href="#pval.type-any-consolidating-with-de-against-any-other-cluster"><code>pval.type = "any"</code>: Consolidating with DE against <em>any</em> other cluster</a></li>
<li><a href="#pval.type-all-consolidating-with-de-against-all-other-clusters"><code>pval.type = "all"</code>: Consolidating with DE against <em>all</em> other clusters</a></li>
<li><a href="#pval.type-some-consolidating-with-de-against-some-other-clusters"><code>pval.type = "some"</code>: Consolidating with DE against <em>some</em> other clusters</a></li>
</ol>
<p>It is important to understand the differences between these strategies, which are described below in some detail.</p>
<h4 id="pval.type-any-consolidating-with-de-against-any-other-cluster"><code>pval.type = "any"</code>: Consolidating with DE against <em>any</em> other cluster</h4>
<p>If <code>pval.type = "any"</code>, the null hypothesis is that the gene is not DE in <em>any</em> pairwise comparison. The genes in each cluster’s gene list are sorted by the minimum rank (by significance) across all pairwise comparisons (called the <code>Top</code> value). Taking all rows with <code>Top</code> values less than or equal to <span class="math inline">\(T\)</span> yields a marker set containing the top <span class="math inline">\(T\)</span> genes from each pairwise comparison.</p>
<p>This strategy guarantees the inclusion of genes that can distinguish between any two clusters.</p>
<p>To demonstrate, let us define a marker set with a <span class="math inline">\(T\)</span> of 1 for a given cluster. The set of genes with <code>Top</code><span class="math inline">\(\leq1\)</span> will contain the top gene from each pairwise comparison to every other cluster. If <span class="math inline">\(T\)</span> is instead, say, <span class="math inline">\(5\)</span>, the set will consist of the union of the top 5 genes from each pairwise comparison. Obviously, multiple genes can have the same <code>Top</code> as different genes may have the same rank across different pairwise comparisons. Conversely, the marker set may be smaller than the product of <code>Top</code> and the number of other clusters, as the same gene may be shared across different comparisons.</p>
<p>This approach does not explicitly favour genes that are uniquely expressed in a cluster. Rather, it focuses on combinations of genes that - together - drive separation of a cluster from the others. This is more general and robust but tends to yield a less focused marker set compared to the other <code>pval.type</code> settings.</p>
<p>For each gene and cluster, the summary effect size is defined as the effect size from the pairwise comparison with the lowest p-value. The combined p-value is computed by applying Simes’ method to all p-values. Neither of these values are directly used for ranking and are only reported for the sake of the user.</p>
<h4 id="pval.type-all-consolidating-with-de-against-all-other-clusters"><code>pval.type = "all"</code>: Consolidating with DE against <em>all</em> other clusters</h4>
<p>If <code>pval.type = "all"</code>, the null hypothesis is that the gene is not DE in <em>all</em> pairwise comparisons. A combined p-value for each gene is computed using Berger’s intersection union test (IUT). Ranking based on the IUT p-value will focus on genes that are DE in that cluster compared to all other clusters.</p>
<p>This strategy is particularly effective when dealing with distinct clusters that have a unique expression profile. In such cases, it yields a highly focused marker set that concisely captures the differences between clusters.</p>
<p>However, it can be too stringent if the cluster’s separation is driven by combinations of gene expression. For example, consider a situation involving four clusters expressing each combination of two marker genes A and B. With <code>pval.type = "all"</code>, neither A nor B would be detected as markers as it is not uniquely defined in any one cluster. This is especially detrimental with overclustering where an otherwise acceptable marker is discarded if it is not DE between two adjacent clusters.</p>
<p>For each gene and cluster, the summary effect size is defined as the effect size from the pairwise comparison with the largest p-value. This reflects the fact that, with this approach, a gene is only as significant as its weakest DE. Again, this value is not directly used for ranking and are only reported for the sake of the user.</p>
<h4 id="pval.type-some-consolidating-with-de-against-some-other-clusters"><code>pval.type = "some"</code>: Consolidating with DE against <em>some</em> other clusters</h4>
<p>If <code>pval.type = "some"</code>, the null hypothesis is that the gene is not DE in <em>some</em> of the pairwise comparisons. Thus, <code>pval.type = "some"</code> serves as a compromise between the <code>pval.type = "all"</code> and <code>pval.type = "any"</code> strategies. The definition of ‘some’ is formalized by the minimum proportion (<code>min.prop</code>) of significant comparisons per gene and can be tuned to the specifics of the dataset.</p>
<p>For example, suppose we require that the gene is significant in at least <code>min.prop = 0.5</code> (i.e. 50%) of comparisons. A combined p-value is calculated by taking the middlemost value of the Holm-corrected p-values for each gene. Here, the null hypothesis is that the gene is not DE in at least half of the contrasts.</p>
<p>Genes are then ranked by the combined p-value. The aim is to provide a more focused marker set without being overly stringent, However, a downside is that it loses the theoretical guarantees of the <code>pval.type = "all"</code> and <code>pval.type = "any"</code> strategies. For example, there is no guarantee that the top set contains genes that can distinguish a cluster from any other cluster, which would have been possible with <code>pval.type = "any"</code>.</p>
<p>For each gene and cluster, the summary effect size is defined as the effect size from the pairwise comparison with the <code>min.prop</code>-smallest p-value. This mirrors the p-value calculation but, again, is reported only for the benefit of the user.</p>
<h2 id="analysis-1">Analysis</h2>
<p>CSV files of the cluster marker gene lists and PDFs files of heatmaps for <em>selected</em> genes from these cluster marker gene lists can be found in the <a href="../output/cluster_marker_genes/"><code>output/cluster_marker_genes/</code></a> directory.</p>
<aside>
See <a href="#visualization">Visualization</a> for further details of these heatmaps.
</aside>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>These files are organised as follows:</p>
<table>
<thead>
<tr class="header">
<th>prefix</th>
<th><code>test.type</code></th>
<th><code>direction</code></th>
<th><code>pval.type</code></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>t_up_all</code></td>
<td>t</td>
<td>up</td>
<td>all</td>
</tr>
<tr class="even">
<td><code>t_up_some</code></td>
<td>t</td>
<td>up</td>
<td>some</td>
</tr>
<tr class="odd">
<td><code>t_any_any</code></td>
<td>t</td>
<td>any</td>
<td>any</td>
</tr>
</tbody>
</table>
<h3 id="t_up_all"><code>t_up_all</code></h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_all/"><code>output/cluster_marker_genes/t_up_all/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-11">Table 1: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">132</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">60</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">74</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="t_up_some"><code>t_up_some</code></h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We set <code>min.prop</code> = <span class="math inline">\(\frac{2}{n - 1}\)</span> where <span class="math inline">\(n\)</span> is the number of clusters (i.e. <code>min.prop</code> = <span class="math inline">\(\frac{2}{7 - 1} = \frac{1}{3}\)</span>) so that ‘some’ means ‘two or more pairwise comparisons’.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_some/"><code>output/cluster_marker_genes/t_up_some/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-15">Table 2: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">1654</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">2360</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">327</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">74</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">459</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">895</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">255</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="t_any_any"><code>t_any_any</code></h3>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_any/"><code>output/cluster_marker_genes/t_up_any/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-18">Table 3: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">3693</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">4669</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">3998</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">3607</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">2358</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">3173</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">3669</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="targeted">Targeted</h3>
<p>We can also perform targeted comparisons between particular clusters of interest. For example, this can be useful when two clusters (<span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span>) share a set of marker genes that distinguish <span class="math inline">\((A, B)\)</span> from the remaining clusters and we now want to hone in on what specifically distinguishes <span class="math inline">\(A\)</span> from <span class="math inline">\(B\)</span>.</p>
<h4 id="t_up_all_clusters_01_and_02"><code>t_up_all_clusters_01_and_02</code></h4>
<p>This analysis compares clusters 1 and 2.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_all_clusters_01_and_02/"><code>output/cluster_marker_genes/t_up_all_clusters_01_and_02/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-21">Table 4: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">176</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">37</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="t_up_all_clusters_03_and_04"><code>t_up_all_clusters_03_and_04</code></h4>
<p>This analysis compares clusters 3 and 4.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_all_clusters_03_and_04/"><code>output/cluster_marker_genes/t_up_all_clusters_03_and_04/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-24">Table 5: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">241</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: left;">4</td>
<td style="text-align: right;">35</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="t_up_all_clusters_03_and_06"><code>t_up_all_clusters_03_and_06</code></h4>
<p>This analysis compares clusters 3 and 6.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_all_clusters_03_and_06/"><code>output/cluster_marker_genes/t_up_all_clusters_03_and_06/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-27">Table 6: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">117</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: left;">6</td>
<td style="text-align: right;">1311</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h4 id="t_up_all_clusters_04_and_06"><code>t_up_all_clusters_04_and_06</code></h4>
<p>This analysis compares clusters 4 and 6.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The following table gives the number of marker genes for each cluster by this approach.</p>
<aside>
For spreadsheets and heatmaps of these gene lists see <a href="../output/cluster_marker_genes/t_up_all_clusters_04_and_06/"><code>output/cluster_marker_genes/t_up_all_clusters_04_and_06/</code></a>.
</aside>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-30">Table 7: </span>Number of marker genes for each cluster (FDR &lt; 0.05).</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">4</td>
<td style="text-align: left;">4</td>
<td style="text-align: right;">24</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: left;">6</td>
<td style="text-align: right;">1239</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="cell-cycle-assignment">Cell cycle assignment</h1>
<p>On occasion, it can be desirable to determine cell cycle activity from scRNA-seq data. In and of itself, the distribution of cells across phases of the cell cycle is not usually informative, but we can use this to determine if there are differences in cell cycle progression between subpopulations or across treatment conditions. Many of the key events in the cell cycle (e.g., passage through checkpoints) are post-translational and thus not directly visible in transcriptomic data; nonetheless, there are enough changes in expression that can be exploited to determine cell cycle phase.</p>
<h2 id="using-the-cyclins">Using the cyclins</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The cyclins control progression through the cell cycle and have well-characterized patterns of expression across cell cycle phases. The cyclins available in this dataset are as follows: Ccnd2, Ccnd3, Ccne1, Ccna2, Ccnb2, Ccnd1, Ccnb1, Ccne2, and Ccna1.</p>
<ul>
<li>Cyclin A is expressed across S and G2.</li>
<li>cyclin B is expressed highest in late G2 and mitosis.</li>
<li>Cyclin D is expressed throughout but peaks at G1.</li>
<li>Cyclin E is expressed highest in the G1/S transition.</li>
</ul>
<p>Inspection of the relative expression of cyclins across the population can often be sufficient to determine the relative cell cycle activity in each cluster.</p>
<p>For example, Figure <a href="#fig:cyclin-heatmap">5</a> shows that clusters 1 and 2 have higher expression of cyclins A, B, and E and therefore may be in S,G2 or G2/M phase while the other clusters are scattered across the later phases.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cyclin-heatmap"></span>
<img src="C075_Grant_Coultas.annotate_files/cyclin-heatmap-1.png" alt="Heatmap of `logcounts` (left) and row-normalized `logcounts` values (right) for Zoe's marker genes. Each column is a sample, each row a gene. Samples are ordered by `cluster` then by `genotype`. Row (gene) and column (cell) order are preserved across the two heatmaps." width="624" />
<p class="caption">
Figure 5: Heatmap of <code>logcounts</code> (left) and row-normalized <code>logcounts</code> values (right) for Zoe’s marker genes. Each column is a sample, each row a gene. Samples are ordered by <code>cluster</code> then by <code>genotype</code>. Row (gene) and column (cell) order are preserved across the two heatmaps.
</p>
</div>
</div>
<p>We can also use standard differential expression methods to look for upregulation of each cyclin, allowing us to determine if there are more cells in the corresponding phase of the cell cycle between subpopulations.</p>
<p>Here we use pairwise Wilcoxon rank sum tests to compute differential expression of cyclin genes between each pair of clusters. The Wilcoxon test estimates effect sizes as overlap proportions (<code>overlap</code> in the tables below). The overlap proportion is defined as the probability that a randomly selected cell in <span class="math inline">\(A\)</span> has a greater expression value of <span class="math inline">\(X\)</span> than a randomly selected cell in <span class="math inline">\(B\)</span>. Overlap proportions near 0 (<span class="math inline">\(A\)</span> is lower than <span class="math inline">\(B\)</span>) or 1 (<span class="math inline">\(A\)</span> is higher than <span class="math inline">\(B\)</span>) indicate that the expression distributions are well-separated. The Wilcoxon rank sum test effectively tests for significant deviations from an overlap proportion of 0.5.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We can infer that clusters 1 and 2 likely have more cells in S, G2, and M phases than the other clusters, based on higher expression of some of the cyclin A and B genes.</p>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-6a3de5c58624f619c90a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6a3de5c58624f619c90a">{"x":{"filter":"none","caption":"<caption>Cyclin DE analysis comparing cluster 1 to all other clusters.<\/caption>","data":[["Ccna2","Ccne1","Ccnb1","Ccne2","Ccnb2","Ccnd1","Ccnd3","Ccnd2","Ccna1"],[1,1,2,2,4,4,4,5,9],[6.73253094045781e-39,7.70465431546744e-20,1.56656192358886e-15,3.10403968032191e-16,1.68362795527801e-09,1.00252448299575e-07,0.00420146487606784,0.999999999999984,1],[6.059277846412e-38,3.46709444196035e-19,3.52476432807492e-15,9.31211904096571e-16,3.03053031950042e-09,1.50378672449362e-07,0.00540188341208722,1,1],[0.828943077647741,0.659386351483414,0.66023225479184,0.655486511902347,0.642215297655712,0.698306714384915,0.575147972636587,0.511412363766303,0.5],[0.571210023226729,0.612794800786135,0.494137484366625,0.597228425942469,0.344648919063784,0.43809183491156,0.542857780954083,0.511412363766303,0.5],[0.828943077647741,0.653130548208268,0.66023225479184,0.653592521468169,0.642215297655712,0.531051849704699,0.574377694844016,0.337023080546397,0.5],[0.81366511164789,0.659386351483414,0.640099201593126,0.655486511902347,0.638596426529973,0.455331624656574,0.575147972636587,0.353161359319971,0.5],[0.837067637585772,0.627153059795547,0.680401904495169,0.621271530597955,0.638636045371797,0.261027867245484,0.450987256686739,0.236941604817252,0.5],[0.835221896823609,0.633532327100698,0.655136291957648,0.621142149132688,0.603987384546069,0.395832394683487,0.510520387474656,0.371998197792296,0.5],[0.831259013941563,0.663132311308157,0.647481437957374,0.67039688050852,0.648149137332408,0.698306714384915,0.52579990385129,0.230970567811549,0.5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Top<\/th>\n      <th>p.value<\/th>\n      <th>FDR<\/th>\n      <th>summary.AUC<\/th>\n      <th>AUC.2<\/th>\n      <th>AUC.3<\/th>\n      <th>AUC.4<\/th>\n      <th>AUC.5<\/th>\n      <th>AUC.6<\/th>\n      <th>AUC.7<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":2,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":3,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":4,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":5,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":6,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":7,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":8,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":9,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":10,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render","options.columnDefs.7.render","options.columnDefs.8.render"],"jsHooks":[]}</script>
</div>
<div class="layout-chunk" data-layout="l-body">
<div id="htmlwidget-260fcda6fd5637e8dc10" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-260fcda6fd5637e8dc10">{"x":{"filter":"none","caption":"<caption>Cyclin DE analysis comparing cluster 2 to all other clusters.<\/caption>","data":[["Ccna2","Ccnb2","Ccnd1","Ccnb1","Ccnd2","Ccne1","Ccne2","Ccnd3","Ccna1"],[1,1,2,3,4,4,4,5,9],[1.56219584299472e-28,6.46395803969461e-33,1.25769384028163e-13,4.07235973738877e-18,1,3.83397748371743e-05,4.32644451368762e-05,0.22913155739734,1],[7.02988129347626e-28,5.81756223572513e-32,2.82981114063365e-13,1.22170792121663e-17,1,6.48966677053142e-05,6.48966677053142e-05,0.294597716653722,1],[0.762674825174825,0.784573185435254,0.763864201919659,0.672044550277309,0.488587636233697,0.551409682169591,0.564685851024666,0.536430850411093,0.5],[0.428789976773271,0.655351080936216,0.56190816508844,0.505862515633375,0.488587636233697,0.387205199213865,0.402771574057531,0.457142219045917,0.5],[0.762674825174825,0.784573185435254,0.589160839160839,0.672044550277309,0.326576440800579,0.543291837472872,0.561384735953701,0.533472992524717,0.5],[0.752109154497484,0.770232543870414,0.504617130936311,0.651897472082464,0.342035832617499,0.551409682169591,0.564685851024666,0.536430850411093,0.5],[0.767998602050326,0.78989981360671,0.298928238583411,0.693528657968313,0.23153541472507,0.513746505125815,0.523182665424045,0.406628611369991,0.5],[0.769162293853073,0.761656671664168,0.451930284857571,0.667962893553223,0.362181409295352,0.52107383808096,0.525665292353823,0.469471514242879,0.5],[0.770440810522574,0.787460007109847,0.763864201919659,0.658971738357625,0.22400462140064,0.555434589406328,0.579608069676502,0.488313188766441,0.5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Top<\/th>\n      <th>p.value<\/th>\n      <th>FDR<\/th>\n      <th>summary.AUC<\/th>\n      <th>AUC.1<\/th>\n      <th>AUC.3<\/th>\n      <th>AUC.4<\/th>\n      <th>AUC.5<\/th>\n      <th>AUC.6<\/th>\n      <th>AUC.7<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":2,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":3,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":4,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":5,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":6,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":7,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":8,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":9,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":10,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render","options.columnDefs.7.render","options.columnDefs.8.render"],"jsHooks":[]}</script>
</div>
<p>Direct examination of cyclin expression is easily understood, interpreted and validated with other technologies. However, it is best suited for statements about relative cell cycle activity; for example, we would find it difficult to assign cell cycle phase in Figure <a href="#fig:cyclin-heatmap">5</a> without the presence of clusters spanning all phases to provide benchmarks for “high” and “low” expression of each cyclin. We also assume that cyclin expression is not affected by biological processes other than the cell cycle, which may be a strong assumption in some cases, e.g., malignant cells.</p>
<h2 id="using-reference-profiles">Using reference profiles</h2>
<p>Cell cycle assignment can be considered a specialized case of cell annotation, which suggests that we can use the <em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> package with an appropriate reference dataset. For example, given a reference dataset containing mouse ESCs with known cell cycle phases<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a>, we could use <em><a href="https://bioconductor.org/packages/3.11/SingleR">SingleR</a></em> to determine the phase of each cell in a test dataset.</p>
<p>We use the reference dataset to identify phase-specific markers from genes with annotated roles in cell cycle. The use of prior annotation aims to avoid detecting markers for other biological processes that happen to be correlated with the cell cycle in the reference dataset, which would reduce classification performance if those processes are absent or uncorrelated in the test dataset.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Unlike the cyclin-based analysis, this approach yields “absolute” assignments of cell cycle phase that do not need to be interpreted relative to other cells in the same dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-37">Table 8: </span>Assignement of cells in each cluster to a cell cycle phase with the ‘reference profiles’ approach.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">G1</th>
<th style="text-align: right;">G2M</th>
<th style="text-align: right;">S</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">64</td>
<td style="text-align: right;">110</td>
<td style="text-align: right;">19</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">157</td>
<td style="text-align: right;">51</td>
<td style="text-align: right;">24</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">263</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">22</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">238</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">27</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">70</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">210</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">19</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">15</td>
</tr>
</tbody>
</table>
</div>
<p>The key assumption here is that the cell cycle is orthogonal to cell type and other aspects of cell behaviour. This justifies the use of a reference involving cell types that are quite different from the cells in the test dataset, provided that the cell cycle transcriptional program is conserved across datasets<a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a>. However, it is not difficult to find routine violations of this assumption<a href="#fn9" class="footnote-ref" id="fnref9" role="doc-noteref"><sup>9</sup></a>.</p>
<p>Thus, a healthy dose of scepticism is required when interpreting these assignments. Our hope is that any systematic assignment error is consistent across clusters and conditions such that they cancel out in comparisons of phase frequencies, which is the more interesting analysis anyway. Indeed, while the availability of absolute phase calls may be more appealing, it may not make much practical difference to the conclusions if the frequencies are ultimately interpreted in a relative sense (e.g., using a chi-squared test).</p>
<h2 id="using-the-cyclone-classifier">Using the <code>cyclone()</code> classifier</h2>
<p>The prediction method described by <span class="citation" data-cites="scialdone2015computational">(Scialdone et al. <a href="#ref-scialdone2015computational" role="doc-biblioref">2015</a>)</span> is another approach for classifying cells into cell cycle phases. Using a reference dataset, we first compute the sign of the difference in expression between each pair of genes. Pairs with changes in the sign across cell cycle phases are chosen as markers. Cells in a test dataset can then be classified into the appropriate phase, based on whether the observed sign for each marker pair is consistent with one phase or another. This approach is implemented in the <code>cyclone()</code> function from the <code>R BiocStyle::Biocpkg("scran")</code> package.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The phase assignment result for each cell in the dataset is shown in Figure <a href="#fig:cyclone">6</a>. For each cell, a higher score for a phase corresponds to a higher probability that the cell is in that phase. We recommend focusing on the G1 and G2/M scores as these are the most informative for classification.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:cyclone"></span>
<img src="C075_Grant_Coultas.annotate_files/cyclone-1.png" alt="Cell cycle phase scores from applying the pair-based classifier on the dataset. Each point represents a cell, plotted according to its scores for G1 and G2/M phases, and coloured according to cluster." width="624" />
<p class="caption">
Figure 6: Cell cycle phase scores from applying the pair-based classifier on the dataset. Each point represents a cell, plotted according to its scores for G1 and G2/M phases, and coloured according to cluster.
</p>
</div>
</div>
<p>Cells are classified as being in G1 phase if the G1 score is above 0.5 and greater than the G2/M score; in G2/M phase if the G2/M score is above 0.5 and greater than the G1 score; and in S phase if neither score is above 0.5.</p>
<p>Figure <a href="#fig:clusters-by-cyclone-phase">7</a> shows most cells are estimated as being in G1 phase.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:clusters-by-cyclone-phase"></span>
<img src="C075_Grant_Coultas.annotate_files/clusters-by-cyclone-phase-1.png" alt="Breakdown of clusters by phase (as predicted using cyclone)." width="624" />
<p class="caption">
Figure 7: Breakdown of clusters by phase (as predicted using cyclone).
</p>
</div>
</div>
<p>The data used to generate Figure <a href="#fig:clusters-by-cyclone-phase">7</a> are also given in the table below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-39">Table 9: </span>Number of cells from each cluster estimated to be in each cell cycle phase using the <code>cyclone()</code> classifier.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">cluster</th>
<th style="text-align: right;">G1</th>
<th style="text-align: right;">G2M</th>
<th style="text-align: right;">S</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">54</td>
<td style="text-align: right;">130</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: right;">133</td>
<td style="text-align: right;">90</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: right;">276</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: right;">252</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: right;">73</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: right;">222</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: right;">92</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
<p>The same considerations and caveats described for the ‘reference profiles’ approach are also applicable here.</p>
<h2 id="summary">Summary</h2>
<p>We can conclude that clusters 1 and 2 likely have more cells in S, G2, and M phases than the other clusters and that the remaining clusters are likely in G1 (based on fairly ubiquitous expression of cyclin D genes and no strong upregulation of the other cyclins).</p>
<p>For some time, it was popular to regress out the cell cycle phase prior to downstream analyses. The aim was to remove uninteresting variation due to cell cycle, thus improving resolution of other biological processes of interest. However, we do not consider adjusting for cell cycle to be a necessary step in routine scRNA-seq analyses.</p>
<p>Any attempt at removal would need to assume that the cell cycle effect is orthogonal to other biological processes. If this was violated (i.e. the cell cycle effect was associated with other biological processes), then the regression would potentially remove interesting signal if cell cycle activity varied across clusters or conditions.</p>
<h1 id="visualization">Visualization</h1>
<p>For each marker gene strategy (e.g. <code>t_up_all</code>) we provide PDF heatmaps of <em>selected</em> marker genes for each cluster in <a href="../output/cluster_marker_genes/"><code>output/cluster_marker_genes/</code></a>.</p>
<aside>
The supplied heatmaps are limited to showing a maximum of 100 genes (any more and they fast become illegible without considerable manual tweaking). We can assist with customised figures for publication.
</aside>
<p>In this section we describe the rationale behind choices made in creating these heatmaps, namely:</p>
<ul>
<li>The considerations behind the <a href="#choice-of-expression-values-logcounts-vs.-corrected">Choice of expression values: <code>logcounts</code> vs. <code>corrected</code></a> used for visualization.</li>
<li>The <a href="#exclusion-of-certain-types-of-genes">Exclusion of certain types of genes</a> from the provided heatmaps.</li>
</ul>
<p>We also use the heatmaps to illustrate the <a href="#limitations-of-the-t_up_all-strategy">Limitations of the <code>t_up_all</code> strategy</a> to further motivate the need for the other marker gene lists.</p>
<h2 id="choice-of-expression-values-logcounts-vs.-corrected">Choice of expression values: <code>logcounts</code> vs. <code>corrected</code></h2>
<p>A reminder that the <code>logcounts</code> (log-transformed normalized expression values) are the basis for any and all gene-based analyses, like DE-based marker gene detection. However, we can visualize the results of such an analysis using either the <code>logcounts</code> or the <code>reconstructed</code> values (batch-corrected <code>logcounts</code>). The advantage of using the <code>reconstructed</code> values for visualization is that these will generally eliminate some of the visual differences between the batches (in this case, between the plates), which can result in ‘cleaner’ figures. As always, however, it is essential to also examine the ‘raw data’ (in this case the <code>logcounts</code>) to ensure that the ‘cleaner’ figures do not belie the truth. If you opt to use the <code>reconstructed</code> data for a figure it is vital that you make note of this (e.g., in the figure caption).</p>
<aside>
See the ‘Data integration’ section of <a href="../docs/C075_Grant_Coultas.cell_selection.html"><code>docs/C075_Grant_Coultas.cell_selection.html</code></a> for details of the <code>reconstructed</code> data.
</aside>
<p>To give some examples of the difference between visualizing <code>logcounts</code> and <code>reconstructed</code> values, Figure <a href="#fig:visualization-violin">8</a> illustrates this with a violin plot of <em>Sox17</em> expression while Figure <a href="#fig:visualization-heatmap">9</a> illustrates this with a heatmap of ribosomal protein gene expression. Together these figures demonstrate that:</p>
<ul>
<li>An arbitrary correction algorithm, such as MNN, is not obliged to preserve the magnitude (or even direction) of differences in per-gene expression when attempting to align multiple batches. This can lead to ‘exaggerated’ differences (Figure <a href="#fig:visualization-violin">8</a>) or, conversely, artificial agreement across batches (Figure <a href="#fig:visualization-heatmap">9</a>).</li>
<li>This means that absolute values (like zero) lose their meaning, i.e. <code>reconstructed</code> = 0 does not mean anything special. In contrast, <code>logcounts</code> = 0 means that there were zero <code>counts</code>, i.e. the gene was not detected in that sample.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:visualization-violin"></span>
<img src="C075_Grant_Coultas.annotate_files/visualization-violin-1.png" alt="Violin plot of *Sox17* gene expression using the `logcounts` (left) and the `reconstructed` values (right) for cells in each cluster." width="624" />
<p class="caption">
Figure 8: Violin plot of <em>Sox17</em> gene expression using the <code>logcounts</code> (left) and the <code>reconstructed</code> values (right) for cells in each cluster.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:visualization-heatmap"></span>
<img src="C075_Grant_Coultas.annotate_files/visualization-heatmap-1.png" alt="Heatmap of some ribosomoal protein gene expression using the `logcounts` (left) and the `reconstructed` values (right) for cells in each cluster. Row (gene) and column (cell) order are preserved across the two heatmaps." width="624" />
<p class="caption">
Figure 9: Heatmap of some ribosomoal protein gene expression using the <code>logcounts</code> (left) and the <code>reconstructed</code> values (right) for cells in each cluster. Row (gene) and column (cell) order are preserved across the two heatmaps.
</p>
</div>
</div>
<h2 id="limitations-of-the-t_up_all-strategy">Limitations of the <code>t_up_all</code> strategy</h2>
<p>Figure <a href="#fig:t-up-all-heatmap">10</a> is a heatmap showing the expression of the top-10 marker genes (<span class="math inline">\(FDR &lt; 0.05\)</span>) for each cluster from the <a href="#t_up_all"><code>t_up_all</code></a> analysis. From this figure we can observe that some clusters do not have any marker genes, and some other clusters only have a small number of marker genes, using the <code>t_up_all</code> strategy. The lack of marker genes for some clusters under the <a href="#t_up_all"><code>t_up_all</code></a> strategy necessitates the other strategies (<a href="#t_up_some"><code>t_up_some</code></a> and <a href="#t_any_any"><code>t_any_any</code></a>).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:t-up-all-heatmap"></span>
<img src="C075_Grant_Coultas.annotate_files/t-up-all-heatmap-1.png" alt="Heatmap of top-10 `t_up_all` marker genes (FDR &lt; 0.05) using the row-normalized `logcounts` for cells in each cluster. Each column is a sample, each row a gene. Note that some clusters have as few as zero upregulated cluster-specific marker genes." width="624" />
<p class="caption">
Figure 10: Heatmap of top-10 <code>t_up_all</code> marker genes (FDR &lt; 0.05) using the row-normalized <code>logcounts</code> for cells in each cluster. Each column is a sample, each row a gene. Note that some clusters have as few as zero upregulated cluster-specific marker genes.
</p>
</div>
</div>
<h2 id="exclusion-of-certain-types-of-genes">Exclusion of certain types of genes</h2>
<p>Figure <a href="#fig:t-up-some-cluster-2-heatmap">11</a> is a heatmap showing the expression of the top-50 marker genes for cluster 2 from the <a href="#t_up_some"><code>t_up_some</code></a> analysis. From this figure we can observe that many of the cluster 2 <code>t_up_some</code> marker genes are pseudogenes, while several others are ribosomal protein genes and sex chromosome genes. This prompted Zoe and Anne to request heatmaps that exclude mitochondrial genes, ribosomal protein genes, sex chromosome genes, and pseudogenes.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:t-up-some-cluster-2-heatmap"></span>
<img src="C075_Grant_Coultas.annotate_files/t-up-some-cluster-2-heatmap-1.png" alt="Heatmap of top-50 `t_up_all` marker genes for cluster 2 using the row-normalized `logcounts` for cells in each cluster. Each column is a sample, each row a gene. Genes are annotated according to `gene_type`." width="624" />
<p class="caption">
Figure 11: Heatmap of top-50 <code>t_up_all</code> marker genes for cluster 2 using the row-normalized <code>logcounts</code> for cells in each cluster. Each column is a sample, each row a gene. Genes are annotated according to <code>gene_type</code>.
</p>
</div>
</div>
<h2 id="choice-of-gradient-colour-scale">Choice of gradient colour scale</h2>
<p>A commonly used plot is a UMAP plot overlaid with the expression of a selected gene using a gradient colour scale. For example, Figure <a href="#fig:sox17-umap-viridis">12</a> shows this for <em>Sox17</em>.</p>
<aside>
Similar plots are shown in Figure <a href="#fig:umap-with-facs">4</a> but visualizing cell surface protein expression as measured by FACS index sorting instead of gene expression.
</aside>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-viridis"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-viridis-1.png" alt="UMAP plot with points coloured by Sox17 expression using the viridis colour gradient." width="624" />
<p class="caption">
Figure 12: UMAP plot with points coloured by Sox17 expression using the viridis colour gradient.
</p>
</div>
</div>
<p>The gene expression measurements are shown using the ‘viridis’ colour gradient, which is one of the five viridis colour palettes (‘viridis’, ‘magma’, ‘plasma’, ‘inferno’, and ‘cividis’).</p>
<aside>
Yep, <em>viridis</em> is both the name of the set of palettes <em>and</em> the name of one of the individual palettes. ¯\_(ツ)_/¯
</aside>
<p>I strongly recommend using one of the viridis colour palettes to show the expression values because they are designed to be<a href="#fn10" class="footnote-ref" id="fnref10" role="doc-noteref"><sup>10</sup></a>:</p>
<ul>
<li><strong>Colourful</strong>, spanning as wide a palette as possible so as to make differences easy to see</li>
<li><strong>Perceptually uniform</strong>, meaning that values close to each other have similar-appearing colours and values far away from each other have more different-appearing colours, consistently across the range of values</li>
<li><strong>Robust to colour blindness</strong>, so that the above properties hold true for people with common forms of colourblindness, as well as in grey scale printing</li>
</ul>
<p>Figure <a href="#fig:sox-17-lightblue-red">13</a> compares the viridis palette to the commonly used lightblue-red colour gradient for these same data, revealing that the lightblue-red colour gradient is not perceptually uniform, e.g.,:</p>
<ul>
<li>The bottom half of the lightblue-red colour gradient is perceptually ‘compressed’</li>
<li>It is hard to distinguish values between 3–5 using the lightblue-red gradient</li>
<li>The perceptual midpoint of the viridis colour gradient matches the midpoint of the 0–5 range of expression values whereas the perceptual midpoint of the lightblue-red colour gradient occurs somewhere below 1</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox-17-lightblue-red"></span>
<img src="C075_Grant_Coultas.annotate_files/sox-17-lightblue-red-1.png" alt="UMAP plot with points coloured by Sox17 expression using the lightblue-red colour gradient." width="624" />
<p class="caption">
Figure 13: UMAP plot with points coloured by Sox17 expression using the lightblue-red colour gradient.
</p>
</div>
</div>
<p>Figure <a href="#fig:sox17-umap-viridis-cvd">14</a> to <a href="#fig:sox17-umap-cividis-cvd">18</a> simulate what this same plot would like to people with various common colour-deficiencies using each of the viridis palettes and shows that the data are still very much interpretable in all cases using any of these palettes.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-viridis-cvd"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-viridis-cvd-1.png" alt="UMAP plot with points coloured by Sox17 expression using the viridis colour gradient and simulating various common colour deficiencies." width="624" />
<p class="caption">
Figure 14: UMAP plot with points coloured by Sox17 expression using the viridis colour gradient and simulating various common colour deficiencies.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-magma-cvd"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-magma-cvd-1.png" alt="UMAP plot with points coloured by Sox17 expression using the magma colour gradient and simulating various common colour deficiencies." width="624" />
<p class="caption">
Figure 15: UMAP plot with points coloured by Sox17 expression using the magma colour gradient and simulating various common colour deficiencies.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-inferno-cvd"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-inferno-cvd-1.png" alt="UMAP plot with points coloured by Sox17 expression using the inferno colour gradient and simulating various common colour deficiencies." width="624" />
<p class="caption">
Figure 16: UMAP plot with points coloured by Sox17 expression using the inferno colour gradient and simulating various common colour deficiencies.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-plasma-cvd"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-plasma-cvd-1.png" alt="UMAP plot with points coloured by Sox17 expression using the plasma colour gradient and simulating various common colour deficiencies." width="624" />
<p class="caption">
Figure 17: UMAP plot with points coloured by Sox17 expression using the plasma colour gradient and simulating various common colour deficiencies.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-cividis-cvd"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-cividis-cvd-1.png" alt="UMAP plot with points coloured by Sox17 expression using the cividis colour gradient and simulating various common colour deficiencies." width="624" />
<p class="caption">
Figure 18: UMAP plot with points coloured by Sox17 expression using the cividis colour gradient and simulating various common colour deficiencies.
</p>
</div>
</div>
<p>Finally, if a greyscale version of the plot is required then it may be preferable to reverse the direction of the colour gradient (i.e. low is lightgrey, high is black). Figure <a href="#fig:sox17-umap-greyscale">19</a> shows both options.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:sox17-umap-greyscale"></span>
<img src="C075_Grant_Coultas.annotate_files/sox17-umap-greyscale-1.png" alt="UMAP plot with points coloured by Sox17 expression using a greyscale colour gradient running in each direction." width="624" />
<p class="caption">
Figure 19: UMAP plot with points coloured by Sox17 expression using a greyscale colour gradient running in each direction.
</p>
</div>
</div>
<h2 id="making-your-own-umap-plot-coloured-by-gene-expression">Making your own UMAP plot coloured by gene expression</h2>
<p>To make your own UMAP plot coloured by gene expression is fairly straightforward. Here I give some example code demonstrating how I made the UMAP plot showing <em>Sox17</em> expression and how you might extend this. Each plot can be saved as a PDF within RStudio (see the ‘Export’ tab in the ‘Plots’ pane.)</p>
<p>To run these examples you will need R version 4.0 and I also recommend using RStudio. You will first need to install a few R packages (if they are not already installed). These packages can be installed and loaded as follows:</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE)) {
    install.packages(&quot;BiocManager&quot;)
}

if (!requireNamespace(&quot;scater&quot;, quietly = TRUE)) {
    BiocManager::install(&quot;scater&quot;)
}

if (!requireNamespace(&quot;cowplot&quot;, quietly = TRUE)) {
    BiocManager::install(&quot;cowplot&quot;)
}</code></pre>
</div>
<p>Once these are installed, you can load the required packages into the R session as follows:</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>You will then need to load the <code>data/SCEs/C075_Grant_Coultas.cells_selected.SCE.rds</code> <em>SingleCellExperiment</em> object into your R session. One way of doing this is to adapt the following code:</p>
<aside>
Alternatively, you can load the object from the ‘Files’ pane in RStudio.
</aside>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
# NOTE: You will need to modify this to where the object is saved on your 
#       computer.
path_to_se &lt;- &quot;data/SCEs/C075_Grant_Coultas.annotated.SCE.rds&quot;

sce &lt;- readRDS(path_to_se)</code></pre>
</div>
<p>You can then create a plot in your R session. Let’s re-make the the UMAP plot showing <em>Sox17</em> expression with the viridis colour palette:</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ggcells(sce, features = &quot;Sox17&quot;) + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17)) +
  theme_cowplot() +
  scale_colour_viridis_c(option = &quot;viridis&quot;)</code></pre>
<p><img src="C075_Grant_Coultas.annotate_files/unnamed-chunk-43-1.png" width="624" /></p>
</div>
<p>To make a version of that plot using the greyscale colour palette requires us to change just the final line of that command:</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ggcells(sce, features = &quot;Sox17&quot;) + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17)) +
  theme_cowplot() +
  scale_colour_gradient(low = &quot;grey90&quot;, high = &quot;black&quot;)</code></pre>
<p><img src="C075_Grant_Coultas.annotate_files/unnamed-chunk-44-1.png" width="624" /></p>
</div>
<p>To plot a different gene requires just two small changes, e.g., to plot <em>Hist1h2ap</em> expression:</p>
<div class="layout-chunk" data-layout="l-body">
<pre class="r"><code>
ggcells(sce, features = &quot;Hist1h2ap&quot;) + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Hist1h2ap)) +
  theme_cowplot() +
  scale_colour_viridis_c(option = &quot;viridis&quot;)</code></pre>
<p><img src="C075_Grant_Coultas.annotate_files/unnamed-chunk-45-1.png" width="624" /></p>
</div>
<h1 id="summary-1">Summary</h1>
<p>We have performed a focused analysis using Zoe’s marker genes and the cyclin genes, and identified cluster marker genes for each cluster using several strategies. Some preliminary high-level summaries of the clusters are as follows:</p>
<ul>
<li>Clusters 1, 5, and 7 can be readily distinguished from the other clusters (<code>t_up_all</code> analysis).
<ul>
<li>Cluster 1 has very high and distinct expression of many genes, most obviously including <em>Hist1h2</em> genes.</li>
<li>Cluster 5 has high and fairly distinct expression of Zoe’s <code>Artery, Tip</code> and <code>Tip</code> marker genes.</li>
<li>Cluster 7 has very high and distinct expression of many genes, including Zoe’s <code>Artery</code> marker genes.</li>
</ul></li>
<li>Cluster 2 shares many marker genes with cluster 1 (<code>t_up_some</code> analysis).</li>
<li>Clusters 3, 4, and 6 are perhaps less distinct than the other clusters and share some marker genes with each other and also with clusters 1, 2, 5, and 7 (<code>t_up_all</code> and <code>t_up_some</code> analyses).
<ul>
<li>Cluster 3 shares some markers with clusters <span class="math inline">\(4 - 7\)</span>.</li>
<li>Cluster 4 is perhaps the least distinct cluster in terms of marker genes.</li>
<li>Cluster 6 shares some markers with clusters 1 and 2 and others with clusters 3, 4, and 7.</li>
</ul></li>
</ul>
<aside>
The <a href="#targeted">Targeted</a> gene lists may help identify genes that specifically distinguish clusters 2 and 6 as well as clusters 3, 4, and 6. We can add further targeted analyses as required.
</aside>
<p>As mentioned, cluster 4 is perhaps the least distinct cluster in terms of marker genes. Cluster 4 contains the fewest cells of all the clusters, with these cells coming from all plates, but is is also the cluster most enriched for <code>Mutant</code> cells (in terms of the percentage of cells in that cluster with each genotype; Figure <a href="#fig:cluster-barplot">2</a>). Figure <a href="#fig:qcplot-by-cluster">20</a> shows that cells in this cluster have on average poorer QC metrics than cells in most other clusters, perhaps meaning that this cluster is comprised of a ‘mixed bag’ of cells that are hard to place with their ‘correct’ cluster.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:qcplot-by-cluster"></span>
<img src="C075_Grant_Coultas.annotate_files/qcplot-by-cluster-1.png" alt="Distributions of various QC metrics for all cells in the data set. This includes the library sizes, number of expressed genes, and proportion of reads mapped to spike-in transcripts or mitochondrial genes." width="624" />
<p class="caption">
Figure 20: Distributions of various QC metrics for all cells in the data set. This includes the library sizes, number of expressed genes, and proportion of reads mapped to spike-in transcripts or mitochondrial genes.
</p>
</div>
</div>
<p>This preliminary summary is just that, preliminary. The cluster marker gene lists can be analysed like any other gene list, such as by using gene set enrichment analysis against the Gene Ontology (GO) or Kyoto Encyclopedia of Genes and Genomes (KEGG) databases. Used this way, the provided cluster marker gene lists can help develop biologically relevant, meaningful, and useful labels for cells/clusters in this dataset.</p>
<aside>
We strongly recommend that you visualise the expression patterns of any genes of interest to better understand what is driving their inclusion in the marker gene lists.
</aside>
<h1 id="concluding-remarks">Concluding remarks</h1>
<p>Zoe requested that clusters 3, 4, and 6 be aggregated for the purposes of differential analyses.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/C075_Grant_Coultas.annotated.SCE.rds"><code>data/SCEs/C075_Grant_Coultas.annotated.SCE.rds</code></a>). This will be used in downstream analyses, e.g., differential expression analysis between conditions within each cluster and differential abundance analyses between conditions for each cluster.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.0 (2020-04-24)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-08-28                  

─ Packages ─────────────────────────────────────────────────────────
 ! package                * version  date       lib
 P AnnotationDbi          * 1.50.1   2020-06-29 [?]
 P AnnotationFilter       * 1.12.0   2020-04-27 [?]
 P AnnotationHub            2.20.0   2020-04-27 [?]
 P askpass                  1.1      2019-01-13 [?]
 P assertthat               0.2.1    2019-03-21 [?]
 P backports                1.1.8    2020-06-17 [?]
 P beeswarm                 0.2.3    2016-04-25 [?]
 P Biobase                * 2.48.0   2020-04-27 [?]
 P BiocFileCache            1.12.0   2020-04-27 [?]
 P BiocGenerics           * 0.34.0   2020-04-27 [?]
 P BiocManager              1.30.10  2019-11-16 [?]
 P BiocNeighbors            1.6.0    2020-04-27 [?]
 P BiocParallel           * 1.22.0   2020-04-27 [?]
 P BiocSingular             1.4.0    2020-04-27 [?]
 P BiocStyle                2.16.0   2020-04-27 [?]
 P BiocVersion              3.11.1   2019-11-13 [?]
 P biomaRt                  2.44.1   2020-06-17 [?]
 P Biostrings               2.56.0   2020-04-27 [?]
 P bit                      1.1-15.2 2020-02-10 [?]
 P bit64                    0.9-7.1  2020-07-15 [?]
 P bitops                   1.0-6    2013-08-17 [?]
 P blob                     1.2.1    2020-01-20 [?]
 P cellranger               1.1.0    2016-07-27 [?]
 P cli                      2.0.2    2020-02-28 [?]
 P colorblindr              0.1.0    2020-08-13 [?]
 P colorspace               1.4-1    2019-03-18 [?]
 P cowplot                * 1.0.0    2019-07-11 [?]
 P crayon                   1.3.4    2017-09-16 [?]
 P crosstalk                1.1.0.1  2020-03-13 [?]
 P curl                     4.3      2019-12-02 [?]
 P DBI                      1.1.0    2019-12-15 [?]
 P dbplyr                   1.4.4    2020-05-27 [?]
 P DelayedArray           * 0.14.1   2020-07-14 [?]
 P DelayedMatrixStats       1.10.1   2020-07-03 [?]
 P digest                   0.6.25   2020-02-23 [?]
 P distill                  0.8      2020-06-04 [?]
 P dplyr                    1.0.0    2020-05-29 [?]
 P dqrng                    0.2.1    2019-05-17 [?]
 P DT                       0.14     2020-06-24 [?]
 P edgeR                  * 3.30.3   2020-06-02 [?]
 P ellipsis                 0.3.1    2020-05-15 [?]
 P ensembldb              * 2.12.1   2020-05-06 [?]
 P evaluate                 0.14     2019-05-28 [?]
 P ExperimentHub            1.14.0   2020-04-27 [?]
 P fansi                    0.4.1    2020-01-08 [?]
 P farver                   2.0.3    2020-01-16 [?]
 P fastmap                  1.0.1    2019-10-08 [?]
 P generics                 0.0.2    2018-11-29 [?]
 P GenomeInfoDb           * 1.24.2   2020-06-15 [?]
 P GenomeInfoDbData         1.2.3    2020-05-18 [?]
 P GenomicAlignments        1.24.0   2020-04-27 [?]
 P GenomicFeatures        * 1.40.1   2020-07-08 [?]
 P GenomicRanges          * 1.40.0   2020-04-27 [?]
 P ggbeeswarm               0.6.0    2017-08-07 [?]
 P ggplot2                * 3.3.2    2020-06-19 [?]
 P Glimma                 * 1.16.0   2020-04-27 [?]
 P glue                     1.4.1    2020-05-13 [?]
 P gridExtra                2.3      2017-09-09 [?]
 P gtable                   0.3.0    2019-03-25 [?]
 P here                   * 0.1      2017-05-28 [?]
 P highr                    0.8      2019-03-20 [?]
 P hms                      0.5.3    2020-01-08 [?]
 P htmltools                0.5.0    2020-06-16 [?]
 P htmlwidgets              1.5.1    2019-10-08 [?]
 P httpuv                   1.5.4    2020-06-06 [?]
 P httr                     1.4.1    2019-08-05 [?]
 P igraph                   1.2.5    2020-03-19 [?]
 P interactiveDisplayBase   1.26.3   2020-06-02 [?]
 P IRanges                * 2.22.2   2020-05-21 [?]
 P irlba                    2.3.3    2019-02-05 [?]
 P janitor                  2.0.1    2020-04-12 [?]
 P jsonlite                 1.7.0    2020-06-25 [?]
 P knitr                    1.29     2020-06-23 [?]
 P labeling                 0.3      2014-08-23 [?]
 P later                    1.1.0.1  2020-06-05 [?]
 P lattice                  0.20-41  2020-04-02 [?]
 P lazyeval                 0.2.2    2019-03-15 [?]
 P lifecycle                0.2.0    2020-03-06 [?]
 P limma                  * 3.44.3   2020-06-12 [?]
 P locfit                   1.5-9.4  2020-03-25 [?]
 P lubridate                1.7.9    2020-06-08 [?]
 P magrittr                 1.5      2014-11-22 [?]
 P Matrix                   1.2-18   2019-11-27 [?]
 P matrixStats            * 0.56.0   2020-03-13 [?]
 P memoise                  1.1.0    2017-04-21 [?]
 P mime                     0.9      2020-02-04 [?]
 P msigdbr                * 7.1.1    2020-05-14 [?]
 P munsell                  0.5.0    2018-06-12 [?]
 P openssl                  1.4.2    2020-06-27 [?]
 P org.Mm.eg.db           * 3.11.4   2020-06-02 [?]
 P patchwork              * 1.0.1    2020-06-22 [?]
 P pheatmap               * 1.0.12   2019-01-04 [?]
 P pillar                   1.4.6    2020-07-10 [?]
 P pkgconfig                2.0.3    2019-09-22 [?]
 P prettyunits              1.1.1    2020-01-24 [?]
 P progress                 1.2.2    2019-05-16 [?]
 P promises                 1.1.1    2020-06-09 [?]
 P ProtGenerics             1.20.0   2020-04-27 [?]
 P purrr                    0.3.4    2020-04-17 [?]
 P R6                       2.4.1    2019-11-12 [?]
 P rappdirs                 0.3.1    2016-03-28 [?]
 P RColorBrewer             1.1-2    2014-12-07 [?]
 P Rcpp                     1.0.5    2020-07-06 [?]
 P RCurl                    1.98-1.2 2020-04-18 [?]
 P readxl                   1.3.1    2019-03-13 [?]
 P rlang                    0.4.7    2020-07-09 [?]
 P rmarkdown                2.3      2020-06-18 [?]
 P rprojroot                1.3-2    2018-01-03 [?]
 P Rsamtools                2.4.0    2020-04-27 [?]
 P RSQLite                  2.2.0    2020-01-07 [?]
 P rsvd                     1.0.3    2020-02-17 [?]
 P rtracklayer              1.48.0   2020-04-27 [?]
 P S4Vectors              * 0.26.1   2020-05-16 [?]
 P scales                   1.1.1    2020-05-11 [?]
 P scater                 * 1.16.2   2020-06-26 [?]
 P scran                  * 1.16.0   2020-04-27 [?]
 P scRNAseq               * 2.2.0    2020-05-07 [?]
 P sessioninfo              1.1.1    2018-11-05 [?]
 P shiny                    1.5.0    2020-06-23 [?]
 P SingleCellExperiment   * 1.10.1   2020-04-28 [?]
 P SingleR                * 1.2.4    2020-05-25 [?]
 P snakecase                0.11.0   2019-05-25 [?]
 P statmod                  1.4.34   2020-02-17 [?]
 P stringi                  1.4.6    2020-02-17 [?]
 P stringr                  1.4.0    2019-02-10 [?]
 P SummarizedExperiment   * 1.18.2   2020-07-09 [?]
 P tibble                   3.0.3    2020-07-10 [?]
 P tidyr                    1.1.0    2020-05-20 [?]
 P tidyselect               1.1.0    2020-05-11 [?]
 P vctrs                    0.3.2    2020-07-15 [?]
 P vipor                    0.4.5    2017-03-22 [?]
 P viridis                  0.5.1    2018-03-29 [?]
 P viridisLite              0.3.0    2018-02-01 [?]
 P withr                    2.2.0    2020-04-20 [?]
 P xfun                     0.15     2020-06-21 [?]
 P XML                      3.99-0.4 2020-07-05 [?]
 P xtable                   1.8-4    2019-04-21 [?]
 P XVector                  0.28.0   2020-04-27 [?]
 P yaml                     2.2.1    2020-02-01 [?]
 P zlibbioc                 1.34.0   2020-04-27 [?]
 source                                 
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Github (clauswilke/colorblindr@1d0d5af)
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 CRAN (R 4.0.0)                         
 Bioconductor                           
 CRAN (R 4.0.0)                         
 Bioconductor                           

[1] /stornext/General/data/user_managed/grpu_mritchie_1/hickey/SCORE/C075_Grant_Coultas/renv/library/R-4.0/x86_64-pc-linux-gnu
[2] /tmp/RtmpwP39zw/renv-system-library
[3] /stornext/System/data/apps/R/R-4.0.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-scialdone2015computational">
<p>Scialdone, A., K. N. Natarajan, L. R. Saraiva, V. Proserpio, S. A. Teichmann, O. Stegle, J. C. Marioni, and F. Buettner. 2015. “Computational Assignment of Cell-Cycle Stage from Single-Cell Transcriptome Data.” <em>Methods</em> 85 (September): 54–61. <a href="https://www.ncbi.nlm.nih.gov/pubmed/26142758">https://www.ncbi.nlm.nih.gov/pubmed/26142758</a>.</p>
</div>
<div id="ref-soneson2018bias">
<p>Soneson, C., and M. D. Robinson. 2018. “Bias, Robustness and Scalability in Single-Cell Differential Expression Analysis.” <em>Nat. Methods</em> 15 (4): 255–61. <a href="https://www.ncbi.nlm.nih.gov/pubmed/29481549">https://www.ncbi.nlm.nih.gov/pubmed/29481549</a>.</p>
</div>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>Email 2020-05-06.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2" role="doc-endnote"><p>E.g., different machines and reagents<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3" role="doc-endnote"><p>Differential expression analyses is performed on the original log-expression values.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4" role="doc-endnote"><p>It must be stressed that the (adjusted) p-values computed here cannot be properly interpreted as measures of significance because the clusters have been empirically identified from the data. Nonetheless, they can still be used as way to rank ‘interesting’ genes.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5" role="doc-endnote"><p>Intra-plate comparisons are robust to differences between libraries but assume that each pair of clusters is present in at least one library. When this is not the case, we cannot commute a logFC for that pair of clusters.<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6" role="doc-endnote"><p>For the purposes of this test, a cell is considered to express a gene simply if it has one or more counts for that gene.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7" role="doc-endnote"><p>Buettner et al. 2015<a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn8" role="doc-endnote"><p>Bertoli, Skotheim, and Bruin 2013; Conboy et al. 2007<a href="#fnref8" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn9" role="doc-endnote"><p>see <a href="https://osca.bioconductor.org/cell-cycle-assignment.html#using-reference-profiles">https://osca.bioconductor.org/cell-cycle-assignment.html#using-reference-profiles</a> for an illustration<a href="#fnref9" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn10" role="doc-endnote"><p>See <a href="https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html">https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html</a> for details.<a href="#fnref10" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<script id="distill-bibliography" type="text/bibtex">
@Article{stegle2015computational,
    Author="Stegle, O.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="Computational and analytical challenges in single-cell transcriptomics",
    Journal="Nat. Rev. Genet.",
    Year="2015",
    Volume="16",
    Number="3",
    Pages="133--145",
    Month="Mar",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25628217}
}

@Article{islam2014quantitative,
    Author="Islam, S.  and Zeisel, A.  and Joost, S.  and La Manno, G.  and Zajac, P.  and Kasper, M.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="Quantitative single-cell RNA-seq with unique molecular identifiers",
    Journal="Nat. Methods",
    Year="2014",
    Volume="11",
    Number="2",
    Pages="163--166",
    Month="Feb"
}

@Article{huber2015orchestrating,
    Author="Huber, W.  and Carey, V. J.  and Gentleman, R.  and Anders, S.  and Carlson, M.  and Carvalho, B. S.  and Bravo, H. C.  and Davis, S.  and Gatto, L.  and Girke, T.  and Gottardo, R.  and Hahne, F.  and Hansen, K. D.  and Irizarry, R. A.  and Lawrence, M.  and Love, M. I.  and MacDonald, J.  and Obenchain, V.  and Oles, A. K.  and Pag\`es, H.  and Reyes, A.  and Shannon, P.  and Smyth, G. K.  and Tenenbaum, D.  and Waldron, L.  and Morgan, M. ",
    Title="{{O}rchestrating high-throughput genomic analysis with {B}ioconductor}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="2",
    Pages="115--121",
    Month="Feb"
}

@Article{brennecke2013accounting,
    Author="Brennecke, P.  and Anders, S.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Zhang, X.  and Proserpio, V.  and Baying, B.  and Benes, V.  and Teichmann, S. A.  and Marioni, J. C.  and Heisler, M. G. ",
    Title="{{A}ccounting for technical noise in single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2013",
    Volume="10",
    Number="11",
    Pages="1093--1095",
    Month="Nov"
}

@Article{marinov2014singlecell,
    Author="Marinov, G. K.  and Williams, B. A.  and McCue, K.  and Schroth, G. P.  and Gertz, J.  and Myers, R. M.  and Wold, B. J. ",
    Title="{{F}rom single-cell to cell-pool transcriptomes: stochasticity in gene expression and {R}{N}{A} splicing}",
    Journal="Genome Res.",
    Year="2014",
    Volume="24",
    Number="3",
    Pages="496--510",
    Month="Mar"
}

@Article{vallejos2015basics,
    Author="Vallejos, C. A.  and Marioni, J. C.  and Richardson, S. ",
    Title="{{B}{A}{S}i{C}{S}: Bayesian analysis of single-cell sequencing data}",
    Journal="PLoS Comput. Biol.",
    Year="2015",
    Volume="11",
    Number="6",
    Pages="e1004333",
    Month="Jun"
}

@Article{trapnell2014dynamics,
    Author="Trapnell, C.  and Cacchiarelli, D.  and Grimsby, J.  and Pokharel, P.  and Li, S.  and Morse, M.  and Lennon, N. J.  and Livak, K. J.  and Mikkelsen, T. S.  and Rinn, J. L. ",
    Title="{{T}he dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="4",
    Pages="381--386",
    Month="Apr"
}

@Article{kolod2015singlecell,
    Author="Ko\l{}odziejczyk, A. A.  and Kim, J. K.  and Tsang, J. C.  and Ilicic, T.  and Henriksson, J.  and Natarajan, K. N.  and Tuck, A. C.  and Gao, X.  and Buhler, M.  and Liu, P.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{S}ingle cell {R}{N}{A}-sequencing of pluripotent states unlocks modular transcriptional variation}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="17",
    Number="4",
    Pages="471--485",
    Month="Oct"
}

@Article{anders2010differential,
    Author="Anders, S.  and Huber, W. ",
    Title="Differential expression analysis for sequence count data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="10",
    Pages="R106",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20979621}
}

@Article{robinson2010scaling,
    Author="Robinson, M. D.  and Oshlack, A. ",
    Title="A scaling normalization method for differential expression analysis of RNA-seq data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="3",
    Pages="R25",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20196867}
}

@Article{islam2011characterization,
    Author="Islam, S.  and Kjallquist, U.  and Moliner, A.  and Zajac, P.  and Fan, J. B.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="{{C}haracterization of the single-cell transcriptional landscape by highly multiplex {R}{N}{A}-seq}",
    Journal="Genome Res.",
    Year="2011",
    Volume="21",
    Number="7",
    Pages="1160--1167",
    Month="Jul"
}

@Article{wilson2015combined,
    Author="Wilson, N. K.  and Kent, D. G.  and Buettner, F.  and Shehata, M.  and Macaulay, I. C.  and Calero-Nieto, F. J.  and Sanchez Castillo, M.  and Oedekoven, C. A.  and Diamanti, E.  and Schulte, R.  and Ponting, C. P.  and Voet, T.  and Caldas, C.  and Stingl, J.  and Green, A. R.  and Theis, F. J.  and Gottgens, B. ",
    Title="{{C}ombined single-cell functional and gene expression analysis resolves heterogeneity within stem cell populations}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="16",
    Number="6",
    Pages="712--724",
    Month="Jun"
}

@Article{picelli2014fulllength,
    Author="Picelli, S.  and Faridani, O. R.  and Bjorklund, A. K.  and Winberg, G.  and Sagasser, S.  and Sandberg, R. ",
    Title="{{F}ull-length {R}{N}{A}-seq from single cells using {S}mart-seq2}",
    Journal="Nat Protoc",
    Year="2014",
    Volume="9",
    Number="1",
    Pages="171--181",
    Month="Jan"
}

@Article{pollen2014lowcoverage,
    Author="Pollen, A. A.  and Nowakowski, T. J.  and Shuga, J.  and Wang, X.  and Leyrat, A. A.  and Lui, J. H.  and Li, N.  and Szpankowski, L.  and Fowler, B.  and Chen, P.  and Ramalingam, N.  and Sun, G.  and Thu, M.  and Norris, M.  and Lebofsky, R.  and Toppani, D.  and Kemp, D. W.  and Wong, M.  and Clerkson, B.  and Jones, B. N.  and Wu, S.  and Knutsson, L.  and Alvarado, B.  and Wang, J.  and Weaver, L. S.  and May, A. P.  and Jones, R. C.  and Unger, M. A.  and Kriegstein, A. R.  and West, J. A. ",
    Title="{{L}ow-coverage single-cell m{R}{N}{A} sequencing reveals cellular heterogeneity and activated signaling pathways in developing cerebral cortex}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="10",
    Pages="1053--1058",
    Month="Oct"
}

@Article{love2015rnaseq,
    Author="Love, M. I.  and Anders, S.  and Kim, V.  and Huber, W. ",
    Title="{{R}{N}{A}-{S}eq workflow: gene-level exploratory analysis and differential expression}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1070"
}

@Article{liao2014featurecounts,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{feature{C}ounts: an efficient general purpose program for assigning sequence reads to genomic features}",
    Journal="Bioinformatics",
    Year="2014",
    Volume="30",
    Number="7",
    Pages="923--930",
    Month="Apr"
}

@Article{liao2013subread,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{{T}he {S}ubread aligner: fast, accurate and scalable read mapping by seed-and-vote}",
    Journal="Nucleic Acids Res.",
    Year="2013",
    Volume="41",
    Number="10",
    Pages="e108",
    Month="May"
}

@Article{law2014voom,
    Author="Law, C. W.  and Chen, Y.  and Shi, W.  and Smyth, G. K. ",
    Title="voom: Precision weights unlock linear model analysis tools for RNA-seq read counts",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="2",
    Pages="R29",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/24485249}
}

@Article{love2014moderated,
    Author="Love, M. I.  and Huber, W.  and Anders, S. ",
    Title="Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="12",
    Pages="550",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25516281}
}

@article{van2008visualizing,
    title={Visualizing data using t-SNE},
    author={Van der Maaten, L. and Hinton, G.},
    journal={J. Mach. Learn. Res.},
    volume={9},
    number={2579-2605},
    pages={85},
    year={2008},
    URL={http://www.jmlr.org/papers/v9/vandermaaten08a.html}
}

@Article{scialdone2015computational,
    Author="Scialdone, A.  and Natarajan, K. N.  and Saraiva, L. R.  and Proserpio, V.  and Teichmann, S. A.  and Stegle, O.  and Marioni, J. C.  and Buettner, F. ",
    Title="Computational assignment of cell-cycle stage from single-cell transcriptome data",
    Journal="Methods",
    Year="2015",
    Volume="85",
    Pages="54--61",
    Month="Sep",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/26142758},
}

@Article{kim2015characterizing,
    Author="Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Illicic, T.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="{{C}haracterizing noise structure in single-cell {R}{N}{A}-seq distinguishes genuine from technical stochastic allelic expression}",
    Journal="Nat. Commun.",
    Year="2015",
    Volume="6",
    Pages="8687"
}

@Article{mccarthy2012differential,
    Author="McCarthy, D. J.  and Chen, Y.  and Smyth, G. K. ",
    Title="{{D}ifferential expression analysis of multifactor {R}{N}{A}-{S}eq experiments with respect to biological variation}",
    Journal="Nucleic Acids Res.",
    Year="2012",
    Volume="40",
    Number="10",
    Pages="4288--4297",
    Month="May"
}

@Article{angel1991role,
    Author="Angel, P.  and Karin, M. ",
    Title="{{T}he role of {J}un, {F}os and the {A}{P}-1 complex in cell-proliferation and transformation}",
    Journal="Biochim. Biophys. Acta",
    Year="1991",
    Volume="1072",
    Number="2-3",
    Pages="129--157",
    Month="Dec"
}

@Article{langfelder2008defining,
    Author="Langfelder, P.  and Zhang, B.  and Horvath, S. ",
    Title="{{D}efining clusters from a hierarchical cluster tree: the {D}ynamic {T}ree {C}ut package for {R}}",
    Journal="Bioinformatics",
    Year="2008",
    Volume="24",
    Number="5",
    Pages="719--720",
    Month="Mar"
}

@Article{ilicic2016classification,
    Author="Ilicic, T.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Bagger, F. O.  and McCarthy, D. J.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{C}lassification of low quality cells from single-cell {R}{N}{A}-seq data}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Number="1",
    Pages="29"
}

@Article{heng2008immunological,
    Author="Heng, T. S.  and Painter, M. W.  and Elpek, K.  and Lukacs-Kornek, V.  and Mauermann, N.  and Turley, S. J.  and Koller, D.  and Kim, F. S.  and Wagers, A. J.  and Asinovski, N.  and Davis, S.  and Fassett, M.  and Feuerer, M.  and Gray, D. H.  and Haxhinasto, S.  and Hill, J. A.  and Hyatt, G.  and Laplace, C.  and Leatherbee, K.  and Mathis, D.  and Benoist, C.  and Jianu, R.  and Laidlaw, D. H.  and Best, J. A.  and Knell, J.  and Goldrath, A. W.  and Jarjoura, J.  and Sun, J. C.  and Zhu, Y.  and Lanier, L. L.  and Ergun, A.  and Li, Z.  and Collins, J. J.  and Shinton, S. A.  and Hardy, R. R.  and Friedline, R.  and Sylvia, K.  and Kang, J. ",
    Title="{{T}he {I}mmunological {G}enome {P}roject: networks of gene expression in immune cells}",
    Journal="Nat. Immunol.",
    Year="2008",
    Volume="9",
    Number="10",
    Pages="1091--1094",
    Month="Oct"
}

@Article{zeisel2015brain,
    Author="Zeisel, A.  and Munoz-Manchado, A. B.  and Codeluppi, S.  and Lonnerberg, P.  and La Manno, G.  and Jureus, A.  and Marques, S.  and Munguba, H.  and He, L.  and Betsholtz, C.  and Rolny, C.  and Castelo-Branco, G.  and Hjerling-Leffler, J.  and Linnarsson, S. ",
    Title="{{B}rain structure. {C}ell types in the mouse cortex and hippocampus revealed by single-cell {R}{N}{A}-seq}",
    Journal="Science",
    Year="2015",
    Volume="347",
    Number="6226",
    Pages="1138--1142",
    Month="Mar"
}

@Article{robinson2010edgeR,
    Author="Robinson, M. D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="edgeR: a Bioconductor package for differential expression analysis of digital gene expression data",
    Journal="Bioinformatics",
    Year="2010",
    Volume="26",
    Number="1",
    Pages="139--140",
    Month="Jan"
}

@Article{ritchie2015limma,
    Author="Ritchie, M. E.  and Phipson, B.  and Wu, D.  and Hu, Y.  and Law, C. W.  and Shi, W.  and Smyth, G. K. ",
    Title="limma powers differential expression analyses for RNA-sequencing and microarray studies",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="7",
    Pages="e47",
    Month="Apr"
}

@Article{julia2015sincell,
    Author="Julia, M.  and Telenti, A.  and Rausell, A. ",
    Title="{{S}incell: an {R}/{B}ioconductor package for statistical assessment of cell-state hierarchies from single-cell {R}{N}{A}-seq}",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="20",
    Pages="3380--3382",
    Month="Oct"
}

@Article{buettner2015computational,
    Author="Buettner, F.  and Natarajan, K. N.  and Casale, F. P.  and Proserpio, V.  and Scialdone, A.  and Theis, F. J.  and Teichmann, S. A.  and Marioni, J. C.  and Stegle, O. ",
    Title="{{C}omputational analysis of cell-to-cell heterogeneity in single-cell {R}{N}{A}-sequencing data reveals hidden subpopulations of cells}",
    Journal="Nat. Biotechnol.",
    Year="2015",
    Volume="33",
    Number="2",
    Pages="155--160",
    Month="Feb"
}

@Article{mahata2014singlecell,
    Author="Mahata, B.  and Zhang, X.  and Ko\l{}odziejczyk, A. A.  and Proserpio, V.  and Haim-Vilmovsky, L.  and Taylor, A. E.  and Hebenstreit, D.  and Dingler, F. A.  and Moignard, V.  and Gottgens, B.  and Arlt, W.  and McKenzie, A. N.  and Teichmann, S. A. ",
    Title="{{S}ingle-cell {R}{N}{A} sequencing reveals {T} helper cells synthesizing steroids de novo to contribute to immune homeostasis}",
    Journal="Cell Rep.",
    Year="2014",
    Volume="7",
    Number="4",
    Pages="1130--1142",
    Month="May"
}

@Article{phipson2010permutation,
    Author="Phipson, B.  and Smyth, G. K. ",
    Title="{{P}ermutation {P}-values should never be zero: calculating exact {P}-values when permutations are randomly drawn}",
    Journal="Stat. Appl. Genet. Mol. Biol.",
    Year="2010",
    Volume="9",
    Pages="Article 39"
}

@incollection{chen2014differential,
    title="Differential expression analysis of complex RNA-seq experiments using edgeR",
    author={Chen, Y. and Lun, A. T. L. and Smyth, G. K.},
    booktitle={Statistical analysis of next generation sequencing data},
    editor={Datta, S. and Nettleton, D.},
    pages={51--74},
    year={2014},
    publisher={Springer}
}

@article{hicks2015widespread,
    author = {Hicks, S. C. and Teng, M. and Irizarry, R. A.},
    title = "On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data",
    year = {2015},
    doi = {10.1101/025528},
    publisher = {Cold Spring Harbor Labs Journals},
    URL = {http://biorxiv.org/content/early/2015/08/25/025528},
    eprint = {http://biorxiv.org/content/early/2015/08/25/025528.full.pdf},
    journal = {bioRxiv}
}

@Article{leng2015oscope,
    Author="Leng, N.  and Chu, L. F.  and Barry, C.  and Li, Y.  and Choi, J.  and Li, X.  and Jiang, P.  and Stewart, R. M.  and Thomson, J. A.  and Kendziorski, C. ",
    Title="{{O}scope identifies oscillatory genes in unsynchronized single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="10",
    Pages="947--950",
    Month="Oct"
}

@Article{fan2016characterizing,
    Author="Fan, J.  and Salathia, N.  and Liu, R.  and Kaeser, G. E.  and Yung, Y. C.  and Herman, J. L.  and Kaper, F.  and Fan, J. B.  and Zhang, K.  and Chun, J.  and Kharchenko, P. V. ",
    Title="{{C}haracterizing transcriptional heterogeneity through pathway and gene set overdispersion analysis}",
    Journal="Nat. Methods",
    Year="2016",
    Volume="13",
    Number="3",
    Pages="241--244",
    Month="Mar"
}

@Article{lund2012detecting,
    Author="Lund, S. P.  and Nettleton, D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{{D}etecting differential expression in {R}{N}{A}-sequence data using quasi-likelihood with shrunken dispersion estimates}",
    Journal="Stat Appl Genet Mol Biol",
    Year="2012",
    Volume="11",
    Number="5",
    Pages=" "
}

@Article{angerer2016destiny,
    Author="Angerer, P.  and Haghverdi, L.  and Buttner, M.  and Theis, F. J.  and Marr, C.  and Buettner, F. ",
    Title="{destiny: diffusion maps for large-scale single-cell data in {R}}",
    Journal="Bioinformatics",
    Year="2016",
    Volume="32",
    Number="8",
    Pages="1241--1243",
}

@Article{lun2016pooling,
    Author="Lun, A. T.  and Bach, K.  and Marioni, J. C. ",
    Title="Pooling across cells to normalize single-cell RNA sequencing data with many zero counts}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Pages="75",
    Month="Apr",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/27122128}
}

@Article{zhu2006gata3,
    Author="Zhu, J. and Yamane, H. and Cote-Sierra, J. and Guo, L. and Paul, W. E.",
    Title="{{G}{A}{T}{A}-3 promotes {T}h2 responses through three different mechanisms: induction of {T}h2 cytokine production, selective growth of {T}h2 cells and inhibition of {T}h1 cell-specific factors}",
    Journal="Cell Res.",
    Year="2006",
    Volume="16",
    Number="1",
    Pages="3--10",
    Month="Jan"
}

@article{berger1996bioequivalence,
    author = "Berger, R. L. and Hsu, J. C.",
    doi = "10.1214/ss/1032280304",
    fjournal = "Statistical Science",
    journal = "Statist. Sci.",
    month = "11",
    number = "4",
    pages = "283--319",
    publisher = "The Institute of Mathematical Statistics",
    title = "Bioequivalence trials, intersection-union tests and equivalence confidence sets",
    url = "http://dx.doi.org/10.1214/ss/1032280304",
    volume = "11",
    year = "1996"
}

@Article{santos2015cyclebase,
    Author="Santos, A.  and Wernersson, R.  and Jensen, L. J. ",
    Title="{{C}yclebase 3.0: a multi-organism database on cell-cycle regulation and phenotypes}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="Database issue",
    Pages="D1140--1144",
    Month="Jan"
}

@article {tung2016batch,
    Author="Tung, P. Y.  and Blischak, J. D.  and Hsiao, C. J.  and Knowles, D. A.  and Burnett, J. E.  and Pritchard, J. K.  and Gilad, Y. ",
    Title="Batch effects and the effective design of single-cell gene expression studies",
    Journal="Sci. Rep.",
    Year="2017",
    Volume="7",
    Pages="39921",
    Month="Jan"
}

@Article{bray2016near,
    Author="Bray, N. L.  and Pimentel, H.  and Melsted, P.  and Pachter, L. ",
    Title="{{N}ear-optimal probabilistic {R}{N}{A}-seq quantification}",
    Journal="Nat. Biotechnol.",
    Year="2016",
    Volume="34",
    Number="5",
    Pages="525--527",
    Month="May"
}

@article{patro2015accurate,
    author = {Patro, Rob and Duggal, Geet and Kingsford, Carl},
    title = {Accurate, fast, and model-aware transcript expression quantification with Salmon},
    year = {2015},
    doi = {10.1101/021592},
    publisher = {Cold Spring Harbor Labs Journals},
    journal = {bioRxiv}
}

@Article{ji2016tscan,
    Author="Ji, Z.  and Ji, H. ",
    Title="{{T}{S}{C}{A}{N}: {P}seudo-time reconstruction and evaluation in single-cell {R}{N}{A}-seq analysis}",
    Journal="Nucleic Acids Res.",
    Year="2016",
    Volume="44",
    Number="13",
    Pages="e117",
    Month="Jul"
}

@Article{klein2015droplet,
    Author="Klein, A. M.  and Mazutis, L.  and Akartuna, I.  and Tallapragada, N.  and Veres, A.  and Li, V.  and Peshkin, L.  and Weitz, D. A.  and Kirschner, M. W. ",
    Title="{{D}roplet barcoding for single-cell transcriptomics applied to embryonic stem cells}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1187--1201",
    Month="May"
}

@Article{macosko2015highly,
    Author="Macosko, E. Z.  and Basu, A.  and Satija, R.  and Nemesh, J.  and Shekhar, K.  and Goldman, M.  and Tirosh, I.  and Bialas, A. R.  and Kamitaki, N.  and Martersteck, E. M.  and Trombetta, J. J.  and Weitz, D. A.  and Sanes, J. R.  and Shalek, A. K.  and Regev, A.  and McCarroll, S. A. ",
    Title="{{H}ighly parallel genome-wide expression profiling of individual cells using nanoliter droplets}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1202--1214",
    Month="May"
}

@Article{bourgon2010independent,
    Author="Bourgon, R.  and Gentleman, R.  and Huber, W. ",
    Title="Independent filtering increases detection power for high-throughput experiments",
    Journal="Proc. Natl. Acad. Sci. U.S.A.",
    Year="2010",
    Volume="107",
    Number="21",
    Pages="9546--9551",
    Month="May",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20460310}
}

@Article{finak2015mast,
    Author="Finak, G.  and McDavid, A.  and Yajima, M.  and Deng, J.  and Gersuk, V.  and Shalek, A. K.  and Slichter, C. K.  and Miller, H. W.  and McElrath, M. J.  and Prlic, M.  and Linsley, P. S.  and Gottardo, R. ",
    Title="MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data",
    Journal="Genome Biol.",
    Year="2015",
    Volume="16",
    Pages="278"
}

@Article{conboy2007cell,
    Author="Conboy, C. M.  and Spyrou, C.  and Thorne, N. P.  and Wade, E. J.  and Barbosa-Morais, N. L.  and Wilson, M. D.  and Bhattacharjee, A.  and Young, R. A.  and Tavare, S.  and Lees, J. A.  and Odom, D. T. ",
    Title="Cell cycle genes are the evolutionarily conserved targets of the E2F4 transcription factor",
    Journal="PLoS ONE",
    Year="2007",
    Volume="2",
    Number="10",
    Pages="e1061",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/17957245}
}

@Article{bertoli2013control,
    Author="Bertoli, C.  and Skotheim, J. M.  and de Bruin, R. A. ",
    Title="Control of cell cycle transcription during G1 and S phases",
    Journal="Nat. Rev. Mol. Cell Biol.",
    Year="2013",
    Volume="14",
    Number="8",
    Pages="518--528",
    Month="Aug",
    URL={{C}ontrol of cell cycle transcription during {G}1 and {S} phases}
}

@Article{chen2016from,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{coller2006new,
    Author="Coller, H. A.  and Sang, L.  and Roberts, J. M. ",
    Title="{{A} new description of cellular quiescence}",
    Journal="PLoS Biol.",
    Year="2006",
    Volume="4",
    Number="3",
    Pages="e83",
    Month="Mar"
}

@Article{mccarthy2017scater,
    Author="McCarthy, D. J.  and Campbell, K. R.  and Lun, A. T.  and Wills, Q. F. ",
    Title="{{S}cater: pre-processing, quality control, normalization and visualization of single-cell {R}{N}{A}-seq data in {R}}",
    Journal="Bioinformatics",
    Year="2017",
    Volume="33",
    Number="8",
    Pages="1179--1186",
    Month="Apr"
}

@article{simes1986improved,
    author = {Simes, R. J.},
    title = {An improved Bonferroni procedure for multiple tests of significance},
    volume = {73},
    number = {3},
    pages = {751--754},
    year = {1986},
    journal = {Biometrika}
}

@Article{xu2015identification,
    Author="Xu, C.  and Su, Z. ",
    Title="Identification of cell types from single-cell transcriptomes using a novel clustering method",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="12",
    Pages="1974--1980",
    Month="Jun"
}

@Article{lun2017assessing,
    Author="Lun, A. T. L.  and Calero-Nieto, F. J.  and Haim-Vilmovsky, L.  and Gottgens, B.  and Marioni, J. C. ",
    Title="Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="11",
    Pages="1795--1806",
    Month="Nov"
}

@Article{yang2016comparative,
    Author="Yang, Z.  and Algesheimer, R.  and Tessone, C. J. ",
    Title="{A comparative analysis of community detection algorithms on artificial networks}",
    Journal="Sci. Rep.",
    Year="2016",
    Volume="6",
    Pages="30750",
    Month="Aug"
}

@Article{smith2017umitools,
    Author="Smith, T.  and Heger, A.  and Sudbery, I. ",
    Title="{{U}{M}{I}-tools: modeling sequencing errors in {U}nique {M}olecular {I}dentifiers to improve quantification accuracy}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="3",
    Pages="491--499",
    Month="Mar"
}

@Article{tasic2016adult,
    Author="Tasic, B.  and Menon, V.  and Nguyen, T. N.  and Kim, T. K.  and Jarsky, T.  and Yao, Z.  and Levi, B.  and Gray, L. T.  and Sorensen, S. A.  and Dolbeare, T.  and Bertagnolli, D.  and Goldy, J.  and Shapovalova, N.  and Parry, S.  and Lee, C.  and Smith, K.  and Bernard, A.  and Madisen, L.  and Sunkin, S. M.  and Hawrylycz, M.  and Koch, C.  and Zeng, H. ",
    Title="{{A}dult mouse cortical cell taxonomy revealed by single cell transcriptomics}",
    Journal="Nat. Neurosci.",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="335--346",
    Month="Feb"
}

@Article{wajapeyee2010senescence,
    Author="Wajapeyee, N.  and Wang, S. Z.  and Serra, R. W.  and Solomon, P. D.  and Nagarajan, A.  and Zhu, X.  and Green, M. R. ",
    Title="{{S}enescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins}",
    Journal="Blood",
    Year="2010",
    Volume="115",
    Number="24",
    Pages="5057--5060",
    Month="Jun"
}

@Article{zeng2012largescale,
    Author="Zeng, H.  and Shen, E. H.  and Hohmann, J. G.  and Oh, S. W.  and Bernard, A.  and Royall, J. J.  and Glattfelder, K. J.  and Sunkin, S. M.  and Morris, J. A.  and Guillozet-Bongaarts, A. L.  and Smith, K. A.  and Ebbert, A. J.  and Swanson, B.  and Kuan, L.  and Page, D. T.  and Overly, C. C.  and Lein, E. S.  and Hawrylycz, M. J.  and Hof, P. R.  and Hyde, T. M.  and Kleinman, J. E.  and Jones, A. R. ",
    Title="{{L}arge-scale cellular-resolution gene profiling in human neocortex reveals species-specific molecular signatures}",
    Journal="Cell",
    Year="2012",
    Volume="149",
    Number="2",
    Pages="483--496",
    Month="Apr"
}

@Article{grun2016denovo,
    Author="Grun, D.  and Muraro, M. J.  and Boisset, J. C.  and Wiebrands, K.  and Lyubimova, A.  and Dharmadhikari, G.  and van den Born, M.  and van Es, J.  and Jansen, E.  and Clevers, H.  and de Koning, E. J. P.  and van Oudenaarden, A. ",
    Title="{{D}e {N}ovo {P}rediction of {S}tem {C}ell {I}dentity using {S}ingle-{C}ell {T}ranscriptome {D}ata}",
    Journal="Cell Stem Cell",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="266--277",
    Month="08"
}

@article{haghverdi2018batch,
    Author="Haghverdi, L.  and Lun, A. T. L.  and Morgan, M. D.  and Marioni, J. C. ",
    Title="Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Month="Apr"
}

@Article{muraro2016singlecell,
    Author="Muraro, M. J.  and Dharmadhikari, G.  and Grun, D.  and Groen, N.  and Dielen, T.  and Jansen, E.  and van Gurp, L.  and Engelse, M. A.  and Carlotti, F.  and de Koning, E. J.  and van Oudenaarden, A. ",
    Title="{{A} {S}ingle-{C}ell {T}ranscriptome {A}tlas of the {H}uman {P}ancreas}",
    Journal="Cell Syst",
    Year="2016",
    Volume="3",
    Number="4",
    Pages="385--394",
    Month="Oct"
}

@Article{segerstolpe2016singlecell,
    Author="Segerstolpe, A.  and Palasantza, A.  and Eliasson, P.  and Andersson, E. M.  and Andreasson, A. C.  and Sun, X.  and Picelli, S.  and Sabirsh, A.  and Clausen, M.  and Bjursell, M. K.  and Smith, D. M.  and Kasper, M.  and Ammala, C.  and Sandberg, R. ",
    Title="{{S}ingle-{C}ell {T}ranscriptome {P}rofiling of {H}uman {P}ancreatic {I}slets in {H}ealth and {T}ype 2 {D}iabetes}",
    Journal="Cell Metab.",
    Year="2016",
    Volume="24",
    Number="4",
    Pages="593--607",
    Month="Oct"
}

@Article{horn1965rationale,
    author="Horn, J. L.",
    title="A rationale and test for the number of factors in factor analysis",
    journal="Psychometrika",
    year="1965",
    month="Jun",
    day="01",
    volume="30",
    number="2",
    pages="179--185",
    issn="1860-0980",
    doi="10.1007/BF02289447",
    url="https://doi.org/10.1007/BF02289447"
}

@Article{buja1992remarks,
    Author="Buja, A.  and Eyuboglu, N. ",
    Title="{{R}emarks on {P}arallel {A}nalysis}",
    Journal="Multivariate Behav Res",
    Year="1992",
    Volume="27",
    Number="4",
    Pages="509--540",
    Month="Oct"
}

@Article{zheng2017massively,
    Author="Zheng, G. X.  and Terry, J. M.  and Belgrader, P.  and Ryvkin, P.  and Bent, Z. W.  and Wilson, R.  and Ziraldo, S. B.  and Wheeler, T. D.  and McDermott, G. P.  and Zhu, J.  and Gregory, M. T.  and Shuga, J.  and Montesclaros, L.  and Underwood, J. G.  and Masquelier, D. A.  and Nishimura, S. Y.  and Schnall-Levin, M.  and Wyatt, P. W.  and Hindson, C. M.  and Bharadwaj, R.  and Wong, A.  and Ness, K. D.  and Beppu, L. W.  and Deeg, H. J.  and McFarland, C.  and Loeb, K. R.  and Valente, W. J.  and Ericson, N. G.  and Stevens, E. A.  and Radich, J. P.  and Mikkelsen, T. S.  and Hindson, B. J.  and Bielas, J. H. ",
    Title="{{M}assively parallel digital transcriptional profiling of single cells}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Pages="14049",
    Month="Jan"
}

@article{lun2018distinguishing,
    author = {Lun, A. and Riesenfeld, S. and Andrews, T. and Dao, T. P. and Gomes, T. and {{participants in the 1\textsuperscript{st} Human Cell Atlas Jamboree}} and Marioni, J.},
    title = {Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data},
    year = {2018},
    doi = {10.1101/234872},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2018/04/04/234872},
    eprint = {https://www.biorxiv.org/content/early/2018/04/04/234872.full.pdf},
    journal = {bioRxiv}
}

@article {wolf2017graph,
    author = {Wolf, F. Alexander and Hamey, Fiona and Plass, Mireya and Solana, Jordi and Dahlin, Joakim S. and Gottgens, Berthold and Rajewsky, Nikolaus and Simon, Lukas and Theis, Fabian J.},
    title = {Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells},
    year = {2017},
    doi = {10.1101/208819},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/10/25/208819},
    eprint = {https://www.biorxiv.org/content/early/2017/10/25/208819.full.pdf},
    journal = {bioRxiv}
}

@Article{lawlor2017singlecell,
    Author="Lawlor, N.  and George, J.  and Bolisetty, M.  and Kursawe, R.  and Sun, L.  and Sivakamasundari, V.  and Kycia, I.  and Robson, P.  and Stitzel, M. L. ",
    Title="{{S}ingle-cell transcriptomes identify human islet cell signatures and reveal cell-type-specific expression changes in type 2 diabetes}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="2",
    Pages="208--222",
    Month="02"
}

@Article{guimaraes2016patterns,
   Author="Guimaraes, J. C.  and Zavolan, M. ",
   Title="{{P}atterns of ribosomal protein expression specify normal and malignant human cells}",
   Journal="Genome Biol.",
   Year="2016",
   Volume="17",
   Number="1",
   Pages="236",
   Month="11"
}

@Article{glare2002betaactin,
   Author="Glare, E. M.  and Divjak, M.  and Bailey, M. J.  and Walters, E. H. ",
   Title="{beta-{A}ctin and {G}{A}{P}{D}{H} housekeeping gene expression in asthmatic airways is variable and not suitable for normalising m{R}{N}{A} levels}",
   Journal="Thorax",
   Year="2002",
   Volume="57",
   Number="9",
   Pages="765--770",
   Month="Sep"
}

@Article{nazari2015gapdh,
   Author="Nazari, F.  and Parham, A.  and Maleki, A. F. ",
   Title="{{G}{A}{P}{D}{H}, \beta-actin and \beta-microglobulin, as three common reference genes, are not reliable for gene expression studies in equine adipose- and marrow-derived mesenchymal stem cells}",
   Journal="J Anim Sci Technol",
   Year="2015",
   Volume="57",
   Pages="18"
}

@Article{soneson2018bias,
    Author="Soneson, C.  and Robinson, M. D. ",
    Title="Bias, robustness and scalability in single-cell differential expression analysis",
    Journal="Nat. Methods",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/29481549},
    Year="2018",
    Volume="15",
    Number="4",
    Pages="255--261",
    Month="Apr"
}

@Article{soneson2015differential,
    Author="Soneson, C.  and Love, M. I.  and Robinson, M. D. ",
    Title="{{D}ifferential analyses for {R}{N}{A}-seq: transcript-level estimates improve gene-level inferences}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1521"
}

@article{stoeckius2017hashing,
    author = {Stoeckius, M. and Zheng, S. and Houck-Loomis, B. and Hao, S. and Yeung, B. and Smibert, P. and Satija, R.},
    title = {Cell "hashing" with barcoded antibodies enables multiplexing and doublet detection for single cell genomics},
    year = {2017},
    doi = {10.1101/237693},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/12/21/237693},
    eprint = {https://www.biorxiv.org/content/early/2017/12/21/237693.full.pdf},
    journal = {bioRxiv}
}

@Article{kang2018multiplexed,
    Author="Kang, H. M.  and Subramaniam, M.  and Targ, S.  and Nguyen, M.  and Maliskova, L.  and McCarthy, E.  and Wan, E.  and Wong, S.  and Byrnes, L.  and Lanata, C. M.  and Gate, R. E.  and Mostafavi, S.  and Marson, A.  and Zaitlen, N.  and Criswell, L. A.  and Ye, C. J. ",
    Title="{{M}ultiplexed droplet single-cell {R}{N}{A}-sequencing using natural genetic variation}",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Volume="36",
    Number="1",
    Pages="89--94",
    Month="Jan"
}

@Article{dahlin2018single,
    Author="Dahlin, J. S.  and Hamey, F. K.  and Pijuan-Sala, B.  and Shepherd, M.  and Lau, W. W. Y.  and Nestorowa, S.  and Weinreb, C.  and Wolock, S.  and Hannah, R.  and Diamanti, E.  and Kent, D. G.  and Gottgens, B.  and Wilson, N. K. ",
    Title="{{A} single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in {K}it mutant mice}",
    Journal="Blood",
    Year="2018",
    Volume="131",
    Number="21",
    Pages="e1-e11",
    Month="May"
}

@Article{bach2017differentiation,
    Author="Bach, K.  and Pensa, S.  and Grzelak, M.  and Hadfield, J.  and Adams, D. J.  and Marioni, J. C.  and Khaled, W. T. ",
    Title="{{D}ifferentiation dynamics of mammary epithelial cells revealed by single-cell {R}{N}{A} sequencing}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Number="1",
    Pages="2128",
    Month="Dec"
}

@article{tian2018scpipe,
  title={scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data},
  author={Tian, Luyi and Su, Shian and Dong, Xueyi and Amann-Zalcenstein, Daniela and Biben, Christine and Seidi, Azadeh and Hilton, Douglas J and Naik, Shalin H and Ritchie, Matthew E},
  journal={PLoS computational biology},
  volume={14},
  number={8},
  pages={e1006361},
  year={2018},
  publisher={Public Library of Science},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/30096152}
}

@article{kiselev2017sc3,
  title={SC3: consensus clustering of single-cell RNA-seq data},
  author={Kiselev, Vladimir Yu and Kirschner, Kristina and Schaub, Michael T and Andrews, Tallulah and Yiu, Andrew and Chandra, Tamir and Natarajan, Kedar N and Reik, Wolf and Barahona, Mauricio and Green, Anthony R and others},
  journal={Nature methods},
  volume={14},
  number={5},
  pages={483},
  year={2017},
  publisher={Nature Publishing Group},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/28346451}
}

@ARTICLE{McInnes2018-hy,
  title         = "UMAP: Uniform Manifold Approximation and Projection for
                   Dimension Reduction",
  author        = "McInnes, Leland and Healy, John and Melville, James",
  abstract      = "UMAP (Uniform Manifold Approximation and Projection) is a
                   novel manifold learning technique for dimension reduction.
                   UMAP is constructed from a theoretical framework based in
                   Riemannian geometry and algebraic topology. The result is a
                   practical scalable algorithm that applies to real world
                   data. The UMAP algorithm is competitive with t-SNE for
                   visualization quality, and arguably preserves more of the
                   global structure with superior run time performance.
                   Furthermore, UMAP has no computational restrictions on
                   embedding dimension, making it viable as a general purpose
                   dimension reduction technique for machine learning.",
  month         =  feb,
  year          =  2018,
  url           = "http://arxiv.org/abs/1802.03426",
  archivePrefix = "arXiv",
  eprint        = "1802.03426",
  primaryClass  = "stat.ML",
  arxivid       = "1802.03426"
}

@article{lun2017overcoming,
  title={Overcoming confounding plate effects in differential expression analyses of single-cell RNA-seq data},
  author={Lun, Aaron TL and Marioni, John C},
  journal={Biostatistics},
  volume={18},
  number={3},
  pages={451--464},
  year={2017},
  publisher={Oxford University Press}
}

@Article{lun2017testing,
    Author="Lun, A. T. L.  and Richard, A. C.  and Marioni, J. C. ",
    Title="Testing for differential abundance in mass cytometry data",
    Journal="Nat. Methods",
    Year="2017",
    Volume="14",
    Number="7",
    Pages="707--709",
    Month="Jul"
}

@Article{chen2016reads,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{phipson2016robust,
    Author="Phipson, B. and Lee, S. and Majewski, I. J. and Alexander, W. S. and Smyth, G. K.",
    year = 2016,
    title = "Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression",
    Journal="Ann. Appl. Stat.",
    Volume=10,
    Number=2,
    Pages="946-963"
}

@Article{aran2019reference,
   Author="Aran, D.  and Looney, A. P.  and Liu, L.  and Wu, E.  and Fong, V.  and Hsu, A.  and Chak, S.  and Naikawadi, R. P.  and Wolters, P. J.  and Abate, A. R.  and Butte, A. J.  and Bhattacharya, M. ",
   Title="Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage",
   Journal="Nat. Immunol.",
   Year="2019",
   Volume="20",
   Number="2",
   Pages="163--172",
   Month="02"
}

@article{Benayoun2019,
author = {Benayoun, B{\'{e}}r{\'{e}}nice A. and Pollina, Elizabeth A. and Singh, Param Priya and Mahmoudi, Salah and Harel, Itamar and Casey, Kerriann M. and Dulken, Ben W. and Kundaje, Anshul and Brunet, Anne},
doi = {10.1101/gr.240093.118},
journal = {Genome Research},
volume = {29},
pages = {697-709},
title = {Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses},
year = {2019}
}

@article{ImmGenRef,
    author = {Heng, Tracy S.P. and Painter, Michio W. and Elpek, Kutlu and Lukacs-Kornek, Veronika and Mauermann, Nora and Turley, Shannon J. and Koller, Daphne and Kim, Francis S. and Wagers, Amy J. and Asinovski, Natasha and Davis, Scott and Fassett, Marlys and Feuerer, Markus and Gray, Daniel H.D. and Haxhinasto, Sokol and Hill, Jonathan A. and Hyatt, Gordon and Laplace, Catherine and Leatherbee, Kristen and Mathis, Diane and Benoist, Christophe and Jianu, Radu and Laidlaw, David H. and Best, J. Adam and Knell, Jamie and Goldrath, Ananda W. and Jarjoura, Jessica and Sun, Joseph C. and Zhu, Yanan and Lanier, Lewis L. and Ergun, Ayla and Li, Zheng and Collins, James J. and Shinton, Susan A. and Hardy, Richard R. and Friedline, Randall and Sylvia, Katelyn and Kang, Joonsoo},
    journal = {Nature Immunology},
    volume = {9},
    issue = {10},
    pages = {1091-4},
    doi = {10.1038/ni1008-1091},
    title = {The immunological genome project: Networks of gene expression in immune cells},
    year = {2008}
}

@article{choi2018haemopedia,
  title={Haemopedia RNA-seq: a database of gene expression during haematopoiesis in mice and humans},
  author={Choi, Jarny and Baldwin, Tracey M and Wong, Mae and Bolden, Jessica E and Fairfax, Kirsten A and Lucas, Erin C and Cole, Rebecca and Biben, Christine and Morgan, Clare and Ramsay, Kerry A and others},
  journal={Nucleic acids research},
  volume={47},
  number={D1},
  pages={D780--D785},
  year={2018},
  publisher={Oxford University Press}
}

@Article{mccarthy2009treat,
    Author="McCarthy, D. J.  and Smyth, G. K. ",
    Title="{Testing significance relative to a fold-change threshold is a TREAT}",
    Journal="Bioinformatics",
    Year="2009",
    Volume="25",
    Number="6",
    Pages="765--771",
    Month="Mar"
}

@article{hpcaRef,
author = {Mabbott, Neil A. and Baillie, J. K. and Brown, Helen and Freeman, Tom C. and Hume, David A.},
doi = {10.1186/1471-2164-14-632},
journal = {BMC Genomics},
volume = {14},
title = {{An expression atlas of human primary cells: Inference of gene function from coexpression networks}},
year = {2013}
}

@article{blueprintRef,
    author = {Martens, Joost H A and Stunnenberg, Hendrik G.},
    doi = {10.3324/haematol.2013.094243},
    journal = {Haematologica},
    title = {{BLUEPRINT: Mapping human blood cell epigenomes}},
    year = {2013},
    volume = {98},
    pages = {1487-1489}
}

@article{encodeRef,
author = {{The ENCODE Project Consortium}},
doi = {10.1038/nature11247},
issn = {14764687},
journal = {Nature},
title = {{An integrated encyclopedia of DNA elements in the human genome}},
year = {2012}
}

@article{lun2016step,
  title={A step-by-step workflow for low-level analysis of single-cell {RNA-seq} data},
  author={Lun, A. T. L. and McCarthy, D. J. and Marioni, J. C.},
  journal ={F1000Res.},
  volume={5},
  month={aug},
  year={2016}
}

@Article{richard2018tcell,
   Author="Richard, A. C.  and Lun, A. T. L.  and Lau, W. W. Y.  and Gottgens, B.  and Marioni, J. C.  and Griffiths, G. M. ",
   Title="T cell cytolytic capacity is independent of initial stimulation strength",
   Journal="Nat. Immunol.",
   Year="2018",
   Volume="19",
   Number="8",
   Pages="849--858",
   Month="08"
}

@Article{leek2012sva,
   Author="Leek, J. T.  and Johnson, W. E.  and Parker, H. S.  and Jaffe, A. E.  and Storey, J. D. ",
   Title="The sva package for removing batch effects and other unwanted variation in high-throughput experiments",
   Journal="Bioinformatics",
   Year="2012",
   Volume="28",
   Number="6",
   Pages="882--883",
   Month="Mar"
}

@Article{butler2018integrating,
   Author="Butler, A.  and Hoffman, P.  and Smibert, P.  and Papalexi, E.  and Satija, R. ",
   Title="Integrating single-cell transcriptomic data across different conditions, technologies, and species",
   Journal="Nat. Biotechnol.",
   Year="2018",
   Volume="36",
   Number="5",
   Pages="411--420",
   Month="06"
}

@Article{lin2019scmerge,
   Author="Lin, Y.  and Ghazanfar, S.  and Wang, K. Y. X.  and Gagnon-Bartsch, J. A.  and Lo, K. K.  and Su, X.  and Han, Z. G.  and Ormerod, J. T.  and Speed, T. P.  and Yang, P.  and Yang, J. Y. H. ",
   Title="scMerge leverages factor analysis, stable expression, and pseudoreplication to merge multiple single-cell RNA-seq datasets",
   Journal="Proc. Natl. Acad. Sci. U.S.A.",
   Year="2019",
   Volume="116",
   Number="20",
   Pages="9775--9784",
   Month="May"
}

@Article{scialdone2016resolving,
   Author="Scialdone, A.  and Tanaka, Y.  and Jawaid, W.  and Moignard, V.  and Wilson, N. K.  and Macaulay, I. C.  and Marioni, J. C.  and Gottgens, B. ",
   Title="Resolving early mesoderm diversification through single-cell expression profiling",
   Journal="Nature",
   Year="2016",
   Volume="535",
   Number="7611",
   Pages="289--293",
   Month="07"
}

@article{crowell2019discovery,
	author = {Crowell, H. L. and Soneson, C. and Germain, P.-L. and Calini, D. and Collin, L. and Raposo, C. and Malhotra, D. and Robinson, M. D.},
	title = {On the discovery of population-specific state transitions from multi-sample multi-condition single-cell RNA sequencing data},
	year = {2019},
	doi = {10.1101/713412},
	journal = {bioRxiv}
}

@article{wu2010roast,
  title={ROAST: rotation gene set tests for complex microarray experiments},
  author={Wu, Di and Lim, Elgene and Vaillant, Fran{\c{c}}ois and Asselin-Labat, Marie-Liesse and Visvader, Jane E and Smyth, Gordon K},
  journal={Bioinformatics},
  volume={26},
  number={17},
  pages={2176--2182},
  year={2010},
  publisher={Oxford University Press}
}

</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
