<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

  <!--radix_placeholder_meta_tags-->
  <title>C075: Multi-sample comparisons with Grant (C075) retinal epithelial cells data set</title>
  
  
  
  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2020-05-19"/>
  <meta property="article:created" itemprop="dateCreated" content="2020-05-19"/>
  <meta name="article:author" content="Peter Hickey"/>
  
  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="C075: Multi-sample comparisons with Grant (C075) retinal epithelial cells data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="C075"/>
  
  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="C075: Multi-sample comparisons with Grant (C075) retinal epithelial cells data set"/>
  
  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Batch effects and the effective design of single-cell gene expression studies;citation_publication_date=2017;citation_volume=7;citation_author=P. Y. Tung;citation_author=J. D. Blischak;citation_author=C. J. Hsiao;citation_author=D. A. Knowles;citation_author=J. E. Burnett;citation_author=J. K. Pritchard;citation_author=Y. Gilad"/>
  <meta name="citation_reference" content="citation_title=&lt;span class=&quot;nocase&quot;&gt;Testing for differential abundance in mass cytometry data&lt;/span&gt;;citation_publication_date=2017;citation_volume=14;citation_author=A. T. L. Lun;citation_author=A. C. Richard;citation_author=J. C. Marioni"/>
  <!--radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","editor_options","bibliography"]}},"value":[{"type":"character","attributes":{},"value":["Multi-sample comparisons with Grant (C075) retinal epithelial cells data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Single Cell Open Research Endeavour (SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/shalin-naik/3310/score"]}]}]},{"type":"character","attributes":{},"value":["2020-05-19"]},{"type":"character","attributes":{},"value":["distill::distill_article"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]},{"type":"character","attributes":{},"value":["ref.bib"]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  
  <script type="application/javascript">
  
    window.headroom_prevent_pin = false;
  
    window.document.addEventListener("DOMContentLoaded", function (event) {
  
      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');
  
      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });
  
      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });
  
      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>
  
  <style type="text/css">
  
  /* Theme (user-documented overrideables for nav appearance) */
  
  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #455a64;
    font-size: 15px;
    font-weight: 300;
  }
  
  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }
  
  .distill-site-nav a:hover {
    color: white;
  }
  
  @media print {
    .distill-site-nav {
      display: none;
    }
  }
  
  .distill-site-header {
  
  }
  
  .distill-site-footer {
  
  }
  
  
  /* Site Header */
  
  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }
  
  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }
  
  
  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }
  
  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }
  
  .distill-site-header .title {
    font-size: 18px;
  }
  
  .distill-site-header .logo {
    padding: 0;
  }
  
  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }
  
  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }
  
  
  
  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }
  
  
  .distill-site-header .nav-toggle {
    display: none;
  }
  
  .nav-dropdown {
    display: inline-block;
    position: relative;
  }
  
  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }
  
  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }
  
  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .nav-dropdown-active {
    display: block;
  }
  
  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }
  
  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }
  
  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }
  
  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }
  
  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }
  
  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }
  
  
  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative;}
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }
  
  /* Site Footer */
  
  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }
  
  /* Headroom */
  
  d-title {
    padding-top: 6rem;
  }
  
  @media print {
    d-title {
      padding-top: 4rem;
    }
  }
  
  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }
  
  .headroom--transition {
    transition: all .4s ease-in-out;
  }
  
  .headroom--unpinned {
    top: -100px;
  }
  
  .headroom--pinned {
    top: 0;
  }
  
  </style>
  
  <link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->
  
  <style type="text/css">
  
  body {
    background-color: white;
  }
  
  .pandoc-table {
    width: 100%;
  }
  
  .pandoc-table>caption {
    margin-bottom: 10px;
  }
  
  .pandoc-table th:not([align]) {
    text-align: left;
  }
  
  .pagedtable-footer {
    font-size: 15px;
  }
  
  .html-widget {
    margin-bottom: 2.0em;
  }
  
  .l-screen-inset {
    padding-right: 16px;
  }
  
  .l-screen .caption {
    margin-left: 10px;
  }
  
  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .shaded .shaded-content {
    background: white;
  }
  
  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }
  
  .hidden {
    display: none !important;
  }
  
  d-article {
    padding-bottom: 30px;
  }
  
  d-appendix {
    padding-top: 30px;
  }
  
  d-article>p>img {
    width: 100%;
  }
  
  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }
  
  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  /* CSS for table of contents */
  
  .d-toc {
    color: rgba(0,0,0,0.8);
    font-size: 0.8em;
    line-height: 1em;
  }
  
  .d-toc-header {
    font-size: 0.6rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.5);
    text-transform: uppercase;
    margin-top: 0;
    margin-bottom: 1.3em;
  }
  
  .d-toc a {
    border-bottom: none;
  }
  
  .d-toc ul {
    padding-left: 0;
  }
  
  .d-toc li>ul {
    padding-top: 0.8em;
    padding-left: 16px;
    margin-bottom: 0.6em;
  }
  
  .d-toc ul,
  .d-toc li {
    list-style-type: none;
  }
  
  .d-toc li {
    margin-bottom: 0.9em;
  }
  
  .d-toc-separator {
    margin-top: 20px;
    margin-bottom: 2em;
  }
  
  .d-article-with-toc {
    border-top: none;
    padding-top: 0;
  }
  
  
  
  /* Tweak code blocks (note that this CSS is repeated above in an injection
     into the d-code shadow dom) */
  
  d-code {
    overflow-x: auto !important;
  }
  
  pre.d-code code.d-code {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }
  
  pre.text-output {
  
    font-size: 12px;
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  @media(min-width: 768px) {
  
  d-code {
    overflow-x: visible !important;
  }
  
  pre.d-code code.d-code  {
      padding-left: 18px;
      font-size: 14px;
  }
  pre.text-output {
    font-size: 14px;
  }
  }
  
  /* Figure */
  
  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }
  
  .figure img {
    width: 100%;
  }
  
  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }
  
  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }
  
  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }
  
  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }
  
  
  
  /* Tweak 1000px media break to show more text */
  
  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }
  
    .grid {
      grid-column-gap: 16px;
    }
  
    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }
  
  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }
  
    .grid {
      grid-column-gap: 32px;
    }
  }
  
  
  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */
  
  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }
  
  
  /* Social footer */
  
  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }
  
  .disqus-comments {
    margin-right: 30px;
  }
  
  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }
  
  #disqus_thread {
    margin-top: 30px;
  }
  
  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }
  
  .article-sharing a:hover {
    border-bottom: none;
  }
  
  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }
  
  .subscribe p {
    margin-bottom: 0.5em;
  }
  
  
  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }
  
  
  /* Improve display for browsers without grid (IE/Edge <= 15) */
  
  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }
  
  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }
  
  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }
  
  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }
  
  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }
  
  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }
  
  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }
  
  
  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }
  
  .downlevel .footnotes ol {
    padding-left: 13px;
  }
  
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }
  
  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }
  
  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }
  
  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;
  
    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;
  
    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }
  
  </style>
  
  <script type="application/javascript">
  
  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }
  
  // show body when load is complete
  function on_load_complete() {
  
    // set body to visible
    document.body.style.visibility = 'visible';
  
    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }
  
    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }
  
  function init_distill() {
  
    init_common();
  
    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);
  
    // create d-title
    $('.d-title').changeElementType('d-title');
  
    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);
  
    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();
  
    // move posts container into article
    $('.posts-container').appendTo($('d-article'));
  
    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');
  
    // create d-bibliography
    var bibliography = $('<d-bibliography></d-bibliography>');
    $('#distill-bibliography').wrap(bibliography);
  
    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;
  
    // replace citations with <d-cite>
    $('.citation').each(function(i, val) {
      appendix = true;
      var cites = $(this).attr('data-cites').split(" ");
      var dt_cite = $('<d-cite></d-cite>');
      dt_cite.attr('key', cites.join());
      $(this).replaceWith(dt_cite);
    });
    // remove refs
    $('#refs').remove();
  
    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();
  
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-toc a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });
  
    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');
  
    // replace code blocks with d-code
    $('pre>code').each(function(i, val) {
      var code = $(this);
      var pre = code.parent();
      var clz = "";
      var language = pre.attr('class');
      if (language) {
        // map unknown languages to "clike" (without this they just dissapear)
        if ($.inArray(language, ["bash", "clike", "css", "go", "html",
                                 "javascript", "js", "julia", "lua", "markdown",
                                 "markup", "mathml", "python", "svg", "xml"]) == -1)
          language = "clike";
        language = ' language="' + language + '"';
        var dt_code = $('<d-code block' + language + clz + '></d-code>');
        dt_code.text(code.text());
        pre.replaceWith(dt_code);
      } else {
        code.addClass('text-output').unwrap().changeElementType('pre');
      }
    });
  
    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {
  
      // capture layout
      var layout = $(this).attr('data-layout');
  
      // apply layout to markdown level block elements
      var elements = $(this).children().not('d-code, pre.text-output, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });
  
  
      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });
  
    // load distill framework
    load_distill_framework();
  
    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {
  
      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;
  
      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');
  
      // table of contents
      if (have_authors) // adjust border if we are in authors
        $('.d-toc').parent().addClass('d-article-with-toc');
  
      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');
  
      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }
  
      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');
  
      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");
  
      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }
  
       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }
  
      // inject pre code styles (can't do this with a global stylesheet b/c a shadow root is used)
      $('d-code').each(function(i, val) {
        var style = document.createElement('style');
        style.innerHTML = 'pre code { padding-left: 10px; font-size: 12px; border-left: 2px solid rgba(0,0,0,0.1); } ' +
                          '@media(min-width: 768px) { pre code { padding-left: 18px; font-size: 14px; } }';
        if (this.shadowRoot)
          this.shadowRoot.appendChild(style);
      });
  
      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();
  
      // clear polling timer
      clearInterval(tid);
  
      // show body now that everything is ready
      on_load_complete();
    }
  
    var tid = setInterval(distill_post_process, 50);
    distill_post_process();
  
  }
  
  function init_downlevel() {
  
    init_common();
  
     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));
  
    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
  
    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();
  
    // remove toc
    $('.d-toc-header').remove();
    $('.d-toc').remove();
    $('.d-toc-separator').remove();
  
    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });
  
  
    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);
  
    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);
  
    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();
  
    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();
  
    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));
  
    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });
  
    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));
  
    $('body').addClass('downlevel');
  
    on_load_complete();
  }
  
  
  function init_common() {
  
    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};
  
        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });
  
        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);
  
    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});
  
    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });
  
      }
    });
  
    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });
  
    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }
  
    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');
  
    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");
  
    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();
  
    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }
  
  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });
  
  </script>
  
  <!--/radix_placeholder_distill-->
  <link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
  <script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
  <script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Multi-sample comparisons with Grant (C075) retinal epithelial cells data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Single Cell Open Research Endeavour (SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/shalin-naik/3310/score"}],"publishedDate":"2020-05-19T00:00:00.000+10:00","citationText":"Hickey, 2020"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">C075</a>
<a href="index.html">Introduction</a>
<a href="C075_Grant_Coultas.preprocess.html">Preprocessing</a>
<a href="C075_Grant_Coultas.merge.html">Merging batches</a>
<a href="C075_Grant_Coultas.annotate.html">Annotating cells</a>
<a href="C075_Grant_Coultas.multi-sample_comparisons.html">Differential analyses</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/FastQC/">FastQC reports</a>
<a href="../output/scPipe/">scPipe reports</a>
<a href="../output/marker_genes/">Marker gene tables</a>
<a href="../output/DEGs/">Differentially expressed genes</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/C075_Grant_Coultas">
<i class="fa fa-github"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Multi-sample comparisons with Grant (C075) retinal epithelial cells data set</h1>

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Single Cell Open Research Endeavour (SCORE), WEHI)<a href="https://www.wehi.edu.au/people/shalin-naik/3310/score" class="uri">https://www.wehi.edu.au/people/shalin-naik/3310/score</a>
  
<br/>2020-05-19
</div>

<div class="d-article">
<h3 class="d-toc-header">Table of Contents</h3>
<nav class="d-toc" id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#differential-expression-vs.differential-abundance">Differential expression vs. differential abundance</a></li>
<li><a href="#differential-expression-between-conditions">Differential expression between conditions</a></li>
<li><a href="#differential-abundance-between-conditions">Differential abundance between conditions</a></li>
<li><a href="#concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information">Additional information</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
<hr class="d-toc-separator"/>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="introduction">Introduction</h1>
<p>Differential analyses of multi-condition scRNA-seq experiments can be broadly split into two categories: differential expression (DE) and differential abundance (DA) analyses. The former tests for changes in expression between conditions for cells of the same type that are present in both conditions, while the latter tests for changes in the composition of cell types (or states, etc.) between conditions.</p>
<p>The aim of this study is to identify differentially expressed genes between mutant and control samples.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<h1 id="differential-expression-vs.differential-abundance">Differential expression vs. differential abundance</h1>
<p>While useful, the distinction between DA and DE analyses is inherently artificial for scRNA-seq data. This is because the labels used in the former are defined based on the genes to be tested in the latter. To illustrate, consider a scRNA-seq experiment involving two biological conditions with several shared cell types. We focus on a cell type <span class="math inline">\(X\)</span> that is present in both conditions but contains some DEGs between conditions. This leads to two possible outcomes:</p>
<ol type="1">
<li>The DE between conditions causes <span class="math inline">\(X\)</span> to form two separate clusters (say, <span class="math inline">\(X_1\)</span> and <span class="math inline">\(X_2\)</span>) in expression space. This manifests as DA where <span class="math inline">\(X_1\)</span> is enriched in one condition and <span class="math inline">\(X_2\)</span> is enriched in the other condition.</li>
<li>The DE between conditions is not sufficient to split <span class="math inline">\(X\)</span> into two separate clusters, e.g., because the data integration procedure identifies them as corresponding cell types and merges them together. This means that the differences between conditions manifest as DE within the single cluster corresponding to <span class="math inline">\(X\)</span>.</li>
</ol>
<p>We have described the example above in terms of clustering, but the same arguments apply for any labelling strategy based on the expression profiles, e.g., automated cell type assignment. Moreover, the choice between outcomes 1 and 2 is made implicitly by the combined effect of the data merging, clustering and label assignment procedures. For example, differences between conditions are more likely to manifest as DE for coarser clusters and as DA for finer clusters, but this is difficult to predict reliably.</p>
<p>The moral of the story is that DA and DE analyses are simply two different perspectives on the same phenomena. For any comprehensive characterization of differences between populations, it is usually necessary to consider both analyses. Indeed, they complement each other almost by definition, e.g., clustering parameters that reduce DE will increase DA and vice versa.</p>
<h1 id="differential-expression-between-conditions">Differential expression between conditions</h1>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We perform the DE analysis separately for all cells with the same ‘label’. The actual DE testing is performed on “pseudobulk” expression profiles <span class="citation" data-cites="tung2016batch">(Tung et al. <a href="#ref-tung2016batch">2017</a>)</span>, generated by summing counts together for all cells with the same combination of label and sample.</p>
<p>This leverages the resolution offered by single-cell technologies to define the labels, and combines it with the statistical rigor of existing methods for DE analyses involving a small number of samples.</p>
<p>Here we use the clusters as the cell labels, i.e. all cells from the same mouse assigned to the same cluster will have the same label. We also perform DE testing without regard for the cluster labels.</p>
<p>The <a href="../output/DEGs/"><code>output/DEGs/</code></a> directory contains CSV files summarising the pseudobulk results and PDFs showing the single-cell expression data for each DEG.</p>
<p>For the differential expression results reported I have used an FDR cutoff of 0.05.</p>
<h2 id="cluster-labels">Cluster labels</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Certain cluster/sample combinations we have few cells, and so these comparisons may be underpowered. None of the clusters is strongly genotype-specific, as shown in Figure <a href="#fig:umap">1</a> and the table below.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:umap"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/umap-1.png" alt="UMAP visualization colored by subpopulation ID (left) and group ID (right)." width="624" />
<p class="caption">
Figure 1: UMAP visualization colored by subpopulation ID (left) and group ID (right).
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-5">Table 1: </span>Number of cells per combination of cluster label and sample.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">plate_number.mouse</th>
<th style="text-align: right;">cluster_1</th>
<th style="text-align: right;">cluster_2</th>
<th style="text-align: right;">cluster_3</th>
<th style="text-align: right;">cluster_4</th>
<th style="text-align: right;">cluster_5</th>
<th style="text-align: right;">cluster_6</th>
<th style="text-align: right;">cluster_7</th>
<th style="text-align: right;">cluster_8</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">LC279.#1554</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">69</td>
<td style="text-align: right;">43</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC280.#1576</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC280.#1577</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">39</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">17</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC392.#1743</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">24</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC392.#1747</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">39</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">21</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC396.#1819</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">27</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC396.#1824</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">28</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC398.#1823</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">13</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC398.#1824</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">5</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Following aggregation of counts, we can visualise the pseudobulk samples using an multidimensional scaling (MDS) plot (Figure <a href="#fig:mds-with-clusters">2</a>). This can be used to check that the pseudobulk samples still largely cluster by our cluster labels and see if there is some treatment-specific differences within the clusters.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:mds-with-clusters"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/mds-with-clusters-1.png" alt="Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by cluster and shaped by group ID." width="624" />
<p class="caption">
Figure 2: Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by cluster and shaped by group ID.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:mds-with-plate-number"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/mds-with-plate-number-1.png" alt="Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by plate number and shaped by group ID." width="624" />
<p class="caption">
Figure 3: Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by plate number and shaped by group ID.
</p>
</div>
</div>
<p>We then run a differential expression analysis separately for each cluster.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="mutant_vs_control"><code>mutant_vs_control</code></h3>
<p>The number of DEGs per label is given below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cluster_2</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">5838</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td>cluster_3</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">6286</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td>cluster_4</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">7886</td>
<td style="text-align: right;">22</td>
</tr>
<tr class="even">
<td>cluster_5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">5039</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td>cluster_6</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">2966</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td>cluster_7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4243</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td>cluster_8</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">4789</td>
<td style="text-align: right;">1</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>A survey of the DEGs from all clusters identifies a small number of genes that are differentially expressed across a number of clusters.</p>
<div class="layout-chunk" data-layout="l-body">
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Gene"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["# Clusters gene is DE"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"Vim","2":"4"},{"1":"Bsg","2":"3"},{"1":"Igfbp3","2":"3"},{"1":"Slc6a6","2":"3"},{"1":"Spock2","2":"3"},{"1":"Col15a1","2":"2"},{"1":"Marcks","2":"2"},{"1":"Marcksl1","2":"2"},{"1":"Mest","2":"2"},{"1":"Pltp","2":"2"},{"1":"Sepp1","2":"2"},{"1":"Xist","2":"2"},{"1":"Actb","2":"1"},{"1":"Anxa1","2":"1"},{"1":"Birc5","2":"1"},{"1":"Btg1","2":"1"},{"1":"Calm2","2":"1"},{"1":"Cd34","2":"1"},{"1":"Ctla2a","2":"1"},{"1":"Cxcl12","2":"1"},{"1":"Egr1","2":"1"},{"1":"Emcn","2":"1"},{"1":"Emp1","2":"1"},{"1":"Erdr1","2":"1"},{"1":"Flna","2":"1"},{"1":"Fth1","2":"1"},{"1":"Ftl1","2":"1"},{"1":"Gm10073","2":"1"},{"1":"Gm10925","2":"1"},{"1":"Gm13192","2":"1"},{"1":"Gm13339","2":"1"},{"1":"Gm21887","2":"1"},{"1":"Gm24270","2":"1"},{"1":"Gm28438","2":"1"},{"1":"Gm6977","2":"1"},{"1":"Gm8730","2":"1"},{"1":"Gnl3","2":"1"},{"1":"H2-K1","2":"1"},{"1":"Ifitm3","2":"1"},{"1":"Itm2a","2":"1"},{"1":"Loxl2","2":"1"},{"1":"Lrp8","2":"1"},{"1":"mt-Atp6","2":"1"},{"1":"mt-Cytb","2":"1"},{"1":"mt-Nd1","2":"1"},{"1":"mt-Nd2","2":"1"},{"1":"mt-Rnr1","2":"1"},{"1":"Myh9","2":"1"},{"1":"Nampt","2":"1"},{"1":"Odc1","2":"1"},{"1":"Phgdh","2":"1"},{"1":"Rplp0","2":"1"},{"1":"Rps28","2":"1"},{"1":"S100a11","2":"1"},{"1":"Sat1","2":"1"},{"1":"Serpinb6a","2":"1"},{"1":"Slc1a4","2":"1"},{"1":"Slc38a5","2":"1"},{"1":"Slc7a5","2":"1"},{"1":"Snap23","2":"1"},{"1":"Sparcl1","2":"1"},{"1":"Stmn2","2":"1"},{"1":"Tfrc","2":"1"},{"1":"Timp3","2":"1"},{"1":"Tmsb10","2":"1"},{"1":"Tspo","2":"1"},{"1":"Ucp2","2":"1"},{"1":"Vwa1","2":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<p>Figure <a href="#fig:degs-heatmap-with-clusters">4</a> is a heatmap of the pseudobulk-level log-expression values of the differentially expressed genes between mutant and control cells.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:degs-heatmap-with-clusters"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/degs-heatmap-with-clusters-1.png" alt="Heatmap of pseudobulk-level log-expression values normalized to the mean of control samples; rows correspond to genes, columns to cluster-sample combinations. Included is the union of DEGs across all clusters. Data is split vertically by cluster (of cells); top and bottom 1% logFC quantiles were truncated for visualization." width="624" />
<p class="caption">
Figure 4: Heatmap of pseudobulk-level log-expression values normalized to the mean of control samples; rows correspond to genes, columns to cluster-sample combinations. Included is the union of DEGs across all clusters. Data is split vertically by cluster (of cells); top and bottom 1% logFC quantiles were truncated for visualization.
</p>
</div>
</div>
<p>The <a href="../output/DEGs/cluster_labels/"><code>output/DEGs/cluster_labels/</code></a> directory contains cluster-specific heatmaps of differentially expressed genes (<span class="math inline">\(FDR &lt; 0.05\)</span>; up to 100 genes shown).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The <a href="../output/DEGs/cluster_labels/"><code>output/DEGs/cluster_labels/</code></a> directory also contains plots of expression values for all cells in a given cluster for the cluster-specific differentially expressed genes (<span class="math inline">\(FDR &lt; 0.05\)</span>; up to 100 genes shown).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="gene-set-analysis">Gene set analysis</h3>
<p><strong>TODO:</strong> Need to decide of DE analysis makes any sense before going further.</p>
<h2 id="ignoring-cluster-labels">Ignoring cluster labels</h2>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>In this analysis we ignore the cluster labels and perform a DE analysis. Care must be taken when interpreting these results as a ‘differentially expressed gene’ may really be due to compositional differences between the treatments.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-14">Table 2: </span>Number of cells per combination of cluster label and sample.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">plate_number.mouse</th>
<th style="text-align: right;">ignoring_cluster_labels</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">LC279.#1554</td>
<td style="text-align: right;">297</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC280.#1576</td>
<td style="text-align: right;">166</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC280.#1577</td>
<td style="text-align: right;">163</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC392.#1743</td>
<td style="text-align: right;">166</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC392.#1747</td>
<td style="text-align: right;">173</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC396.#1819</td>
<td style="text-align: right;">162</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC396.#1824</td>
<td style="text-align: right;">96</td>
</tr>
<tr class="even">
<td style="text-align: left;">LC398.#1823</td>
<td style="text-align: right;">88</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LC398.#1824</td>
<td style="text-align: right;">43</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Following aggregation of counts, we can visualise the pseudobulk samples using an multidimensional scaling (MDS) plot (Figure <a href="#fig:mds-with-clusters">2</a>). This can be used to check that the pseudobulk samples still largely cluster by our cluster labels and see if there is some treatment-specific differences within the clusters.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:mds-ignoring-clusters"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/mds-ignoring-clusters-1.png" alt="Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by cluster and shaped by group ID." width="624" />
<p class="caption">
Figure 5: Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by cluster and shaped by group ID.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:mds-ignoring-plate-number"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/mds-ignoring-plate-number-1.png" alt="Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by plate number and shaped by group ID." width="624" />
<p class="caption">
Figure 6: Pseudobulk-level multidimensional scaling (MDS) plot. Each point represents one cluster-sample instance; points are colored by plate number and shaped by group ID.
</p>
</div>
</div>
<p>We then run a differential expression analysis separately for each cluster.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="mutant_vs_control-1"><code>mutant_vs_control</code></h3>
<p>The number of DEGs per label is given below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th></th>
<th style="text-align: right;">Down</th>
<th style="text-align: right;">NotSig</th>
<th style="text-align: right;">Up</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ignoring_cluster_labels</td>
<td style="text-align: right;">133</td>
<td style="text-align: right;">11877</td>
<td style="text-align: right;">140</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Figure <a href="#fig:degs-heatmap-ignoring-clustesrs">7</a> is a heatmap of the pseudobulk-level log-expression values the top-100 differentially expressed genes between mutant and control cells.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="figure"><span id="fig:degs-heatmap-ignoring-clustesrs"></span>
<img src="C075_Grant_Coultas.multi-sample_comparisons_files/degs-heatmap-ignoring-clustesrs-1.png" alt="Heatmap of pseudobulk-level log-expression values normalized to the mean of control samples; rows correspond to genes, columns to samples. Genes are DEGs. Top and bottom 1% logFC quantiles were truncated for visualization." width="624" />
<p class="caption">
Figure 7: Heatmap of pseudobulk-level log-expression values normalized to the mean of control samples; rows correspond to genes, columns to samples. Genes are DEGs. Top and bottom 1% logFC quantiles were truncated for visualization.
</p>
</div>
</div>
<p>The <a href="../output/DEGs/ignoring_cluster_labels/"><code>output/DEGs/ignoring_cluster_labels/</code></a> directory also contains plots of expression values for all cells differentially expressed genes (<span class="math inline">\(FDR &lt; 0.05\)</span>; up to 100 genes shown).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<h3 id="gene-set-analysis-1">Gene set analysis</h3>
<p><strong>TODO:</strong> Need to decide of DE analysis makes any sense before going further.</p>
<h1 id="differential-abundance-between-conditions">Differential abundance between conditions</h1>
<p>In a differential abundance analysis, we test for significant changes in per-label cell abundance across conditions. This will reveal which cell types are depleted or enriched due to genotype differences, which is arguably just as interesting as changes in expression within each cell type. We effectively treat scRNA-seq as a “super-FACS” technology for defining relevant subpopulations using the entire transcriptome.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p><strong>For the differential abundance results reported I have used an FDR cutoff of 0.05.</strong></p>
<p>We prepare for the DA analysis by quantifying the number of cells assigned to each cluster. In this case, we are aiming to identify clusters that change in abundance among the infected cells compared to the uninfected cells.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-22">Table 3: </span>Number of cells per combination of cluster label and sample.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">label</th>
<th style="text-align: right;">LC279.#1554</th>
<th style="text-align: right;">LC280.#1576</th>
<th style="text-align: right;">LC280.#1577</th>
<th style="text-align: right;">LC392.#1743</th>
<th style="text-align: right;">LC392.#1747</th>
<th style="text-align: right;">LC396.#1819</th>
<th style="text-align: right;">LC396.#1824</th>
<th style="text-align: right;">LC398.#1823</th>
<th style="text-align: right;">LC398.#1824</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">cluster_1</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">cluster_2</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">39</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cluster_3</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">cluster_4</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">39</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cluster_5</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">cluster_6</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cluster_7</td>
<td style="text-align: right;">69</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">13</td>
</tr>
<tr class="even">
<td style="text-align: left;">cluster_8</td>
<td style="text-align: right;">43</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">5</td>
</tr>
</tbody>
</table>
</div>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>Our DA analysis will again be performed with the <em><a href="https://bioconductor.org/packages/3.10/edgeR">edgeR</a></em> package. This allows us to take advantage of the negative binomial generalized linear model methods to model overdispersed count data in the presence of limited replication - except that the counts are not of reads per gene, but of cells per label <span class="citation" data-cites="lun2017testing">(Lun, Richard, and Marioni <a href="#ref-lun2017testing">2017</a>)</span>. The aim is to share information across labels to improve our estimates of the biological variability in cell abundance between replicates.</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>We filter out low-abundance labels as previously described. This avoids cluttering the result table with very rare subpopulations that contain only a handful of cells. For a DA analysis of cluster abundances, filtering is generally not required as most clusters will not be of low-abundance (otherwise there would not have been enough evidence to define the cluster in the first place).</p>
<div class="layout-chunk" data-layout="l-body">

</div>
<p>The number of differentially abundance clusters is given below. The log-fold change in our model refers to the change in cell abundance after infection, rather than the change in gene expression. A positive <code>logFC</code> means an enrichment of that cluster in the <code>infected</code> vs. the <code>uninfected</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<caption><span id="tab:unnamed-chunk-26">Table 4: </span>The number of clusters that are differentially abundant.</caption>
<thead>
<tr class="header">
<th></th>
<th style="text-align: right;">Mutant</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Down</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td>NotSig</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td>Up</td>
<td style="text-align: right;">2</td>
</tr>
</tbody>
</table>
</div>
<p>The full results are given below.</p>
<div class="layout-chunk" data-layout="l-body">
<table>
<thead>
<tr class="header">
<th></th>
<th style="text-align: right;">logFC</th>
<th style="text-align: right;">logCPM</th>
<th style="text-align: right;">F</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cluster_5</td>
<td style="text-align: right;">2.5319969</td>
<td style="text-align: right;">16.24552</td>
<td style="text-align: right;">35.8084738</td>
<td style="text-align: right;">0.0000396</td>
<td style="text-align: right;">0.0003165</td>
</tr>
<tr class="even">
<td>cluster_7</td>
<td style="text-align: right;">-1.0372038</td>
<td style="text-align: right;">17.43891</td>
<td style="text-align: right;">23.1330362</td>
<td style="text-align: right;">0.0003100</td>
<td style="text-align: right;">0.0012400</td>
</tr>
<tr class="odd">
<td>cluster_2</td>
<td style="text-align: right;">0.6912108</td>
<td style="text-align: right;">17.23912</td>
<td style="text-align: right;">12.7306484</td>
<td style="text-align: right;">0.0032703</td>
<td style="text-align: right;">0.0087209</td>
</tr>
<tr class="even">
<td>cluster_8</td>
<td style="text-align: right;">-0.4720918</td>
<td style="text-align: right;">17.18427</td>
<td style="text-align: right;">4.5148252</td>
<td style="text-align: right;">0.0526653</td>
<td style="text-align: right;">0.1053306</td>
</tr>
<tr class="odd">
<td>cluster_4</td>
<td style="text-align: right;">0.3649496</td>
<td style="text-align: right;">17.56231</td>
<td style="text-align: right;">3.1104360</td>
<td style="text-align: right;">0.1004800</td>
<td style="text-align: right;">0.1607681</td>
</tr>
<tr class="even">
<td>cluster_3</td>
<td style="text-align: right;">-0.3624476</td>
<td style="text-align: right;">16.91702</td>
<td style="text-align: right;">2.7422207</td>
<td style="text-align: right;">0.1593070</td>
<td style="text-align: right;">0.2124093</td>
</tr>
<tr class="odd">
<td>cluster_6</td>
<td style="text-align: right;">-0.3454095</td>
<td style="text-align: right;">16.96639</td>
<td style="text-align: right;">2.2331024</td>
<td style="text-align: right;">0.1875447</td>
<td style="text-align: right;">0.2143368</td>
</tr>
<tr class="even">
<td>cluster_1</td>
<td style="text-align: right;">-0.1151770</td>
<td style="text-align: right;">16.16289</td>
<td style="text-align: right;">0.0655397</td>
<td style="text-align: right;">0.8018188</td>
<td style="text-align: right;">0.8018188</td>
</tr>
</tbody>
</table>
</div>
<h3 id="handling-composition-effects">Handling composition effects</h3>
<p>Our default approach for normalization in a differential abundance analysis is to only normalize based on the total number of cells in each sample, which means that we are effectively testing for differential proportions between conditions.</p>
<p>Unfortunately, the use of the total number of cells leaves us susceptible to composition effects. For example, a large increase in abundance for one cell subpopulation will introduce decreases in proportion for all other subpopulations - which is technically correct, but may be misleading if one concludes that those other subpopulations are decreasing in abundance of their own volition. If composition biases are proving problematic for interpretation of DA results, we have several avenues for removing them or mitigating their impact by leveraging a priori biological knowledge.</p>
<h1 id="concluding-remarks">Concluding remarks</h1>
<p>The <a href="../output/DEGs/"><code>output/DEGs</code></a> directory contains CSV files summarising the differential expressoin results and PDFs showing the single-cell expression data for each DEG.</p>
<h1 id="additional-information" class="appendix">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 id="session-info" class="appendix">Session info</h1>
<p>The analysis and this document were prepared using the following software:</p>
<div class="layout-chunk" data-layout="l-body">
<pre><code>
─ Session info ─────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       CentOS Linux 7 (Core)       
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2020-05-19                  

─ Packages ─────────────────────────────────────────────────────────
 ! package              * version  date       lib
 P acepack                1.4.1    2016-10-29 [?]
 P assertthat             0.2.1    2019-03-21 [?]
 P backports              1.1.5    2019-10-02 [?]
 P base64enc              0.1-3    2015-07-28 [?]
 P beeswarm               0.2.3    2016-04-25 [?]
 P Biobase              * 2.46.0   2019-10-29 [?]
 P BiocGenerics         * 0.32.0   2019-10-29 [?]
 P BiocManager            1.30.10  2019-11-16 [?]
 P BiocNeighbors          1.4.2    2020-02-29 [?]
 P BiocParallel         * 1.20.1   2019-12-21 [?]
 P BiocSingular           1.2.2    2020-02-14 [?]
 P BiocStyle            * 2.14.4   2020-01-09 [?]
 P bitops                 1.0-6    2013-08-17 [?]
 P checkmate              2.0.0    2020-02-06 [?]
 P cli                    2.0.2    2020-02-28 [?]
 P cluster                2.1.0    2019-06-19 [3]
 P colorblindr            0.1.0    2019-12-11 [?]
 P colorspace             1.4-1    2019-03-18 [?]
 P cowplot                1.0.0    2019-07-11 [?]
 P crayon                 1.3.4    2017-09-16 [?]
 P data.table             1.12.8   2019-12-09 [?]
 P DelayedArray         * 0.12.2   2020-01-06 [?]
 P DelayedMatrixStats     1.8.0    2019-10-29 [?]
 P digest                 0.6.25   2020-02-23 [?]
 P distill                0.7      2019-05-03 [?]
 P dplyr                * 0.8.4    2020-01-31 [?]
 P edgeR                * 3.28.1   2020-02-26 [?]
 P evaluate               0.14     2019-05-28 [?]
 P fansi                  0.4.1    2020-01-08 [?]
 P farver                 2.0.3    2020-01-16 [?]
 P foreign                0.8-76   2020-03-03 [?]
 P Formula              * 1.2-3    2018-05-03 [?]
 P GenomeInfoDb         * 1.22.0   2019-10-29 [?]
 P GenomeInfoDbData       1.2.2    2020-01-01 [?]
 P GenomicRanges        * 1.38.0   2019-10-29 [?]
 P ggbeeswarm             0.6.0    2017-08-07 [?]
 P ggplot2              * 3.2.1    2019-08-10 [?]
 P glue                   1.3.1    2019-03-12 [?]
 P gridExtra              2.3      2017-09-09 [?]
 P gtable                 0.3.0    2019-03-25 [?]
 P here                 * 0.1      2017-05-28 [?]
 P highr                  0.8      2019-03-20 [?]
 P Hmisc                * 4.3-1    2020-02-07 [?]
 P htmlTable              1.13.3   2019-12-04 [?]
 P htmltools              0.4.0    2019-10-04 [?]
 P htmlwidgets            1.5.1    2019-10-08 [?]
 P IRanges              * 2.20.2   2020-01-13 [?]
 P irlba                  2.3.3    2019-02-05 [?]
 P janitor              * 1.2.1    2020-01-22 [?]
 P jpeg                   0.1-8.1  2019-10-24 [?]
 P jsonlite               1.6.1    2020-02-02 [?]
 P knitr                  1.28     2020-02-06 [?]
 P labeling               0.3      2014-08-23 [?]
 P lattice              * 0.20-40  2020-02-19 [?]
 P latticeExtra           0.6-29   2019-12-19 [?]
 P lazyeval               0.2.2    2019-03-15 [?]
 P lifecycle              0.1.0    2019-08-01 [?]
 P limma                * 3.42.2   2020-02-03 [?]
 P locfit                 1.5-9.1  2013-04-20 [?]
 P magrittr               1.5      2014-11-22 [?]
 P Matrix                 1.2-18   2019-11-27 [3]
 P matrixStats          * 0.55.0   2019-09-07 [?]
 P munsell                0.5.0    2018-06-12 [?]
 P nnet                   7.3-13   2020-02-25 [?]
 P pheatmap             * 1.0.12   2019-01-04 [?]
 P pillar                 1.4.3    2019-12-20 [?]
 P pkgconfig              2.0.3    2019-09-22 [?]
 P png                    0.1-7    2013-12-03 [?]
 P Polychrome             1.2.4    2020-02-03 [?]
 P purrr                  0.3.3    2019-10-18 [?]
 P R6                     2.4.1    2019-11-12 [?]
 P RColorBrewer           1.1-2    2014-12-07 [?]
 P Rcpp                   1.0.3    2019-11-08 [?]
 P RCurl                  1.98-1.1 2020-01-19 [?]
 P rlang                  0.4.5    2020-03-01 [?]
 P rmarkdown              2.1      2020-01-20 [?]
 P rpart                  4.1-15   2019-04-12 [3]
 P rprojroot              1.3-2    2018-01-03 [?]
 P rstudioapi             0.11     2020-02-07 [?]
 P rsvd                   1.0.3    2020-02-17 [?]
 P S4Vectors            * 0.24.3   2020-01-18 [?]
 P scales                 1.1.0    2019-11-18 [?]
 P scater               * 1.14.6   2019-12-16 [?]
 P scatterplot3d          0.3-41   2018-03-14 [?]
 P sessioninfo            1.1.1    2018-11-05 [?]
 P SingleCellExperiment * 1.8.0    2019-10-29 [?]
 P statmod                1.4.34   2020-02-17 [?]
 P stringi                1.4.6    2020-02-17 [?]
 P stringr                1.4.0    2019-02-10 [?]
 P SummarizedExperiment * 1.16.1   2019-12-19 [?]
 P survival             * 3.1-8    2019-12-03 [3]
 P tibble                 2.1.3    2019-06-06 [?]
 P tidyr                  1.0.2    2020-01-24 [?]
 P tidyselect             1.0.0    2020-01-27 [?]
 P vctrs                  0.2.3    2020-02-20 [?]
 P vipor                  0.4.5    2017-03-22 [?]
 P viridis                0.5.1    2018-03-29 [?]
 P viridisLite            0.3.0    2018-02-01 [?]
 P withr                  2.1.2    2018-03-15 [?]
 P xfun                   0.12     2020-01-13 [?]
 P XVector                0.26.0   2019-10-29 [?]
 P yaml                   2.2.1    2020-02-01 [?]
 P zlibbioc               1.32.0   2019-10-29 [?]
 source                                 
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 Bioconductor                           
 Bioconductor                           
 CRAN (R 3.6.1)                         
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.3)                         
 github (clauswilke/colorblindr@1ac3d4d)
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 Bioconductor                           
 Bioconductor                           
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 Bioconductor                           
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 Bioconductor                           
 Bioconductor                           
 Bioconductor                           
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 Bioconductor                           
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 Bioconductor                           
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.3)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.3)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 Bioconductor                           
 CRAN (R 3.6.1)                         
 Bioconductor                           
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 Bioconductor                           
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 Bioconductor                           
 CRAN (R 3.6.3)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.2)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.1)                         
 CRAN (R 3.6.2)                         
 Bioconductor                           
 CRAN (R 3.6.2)                         
 Bioconductor                           

[1] /stornext/General/data/user_managed/grpu_mritchie_1/hickey/SCORE/C075_Grant_Coultas/renv/library/R-3.6/x86_64-pc-linux-gnu
[2] /tmp/RtmpOBVEfI/renv-system-library
[3] /stornext/System/data/apps/R/R-3.6.3/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
<div id="refs" class="references">
<div id="ref-lun2017testing">
<p>Lun, A. T. L., A. C. Richard, and J. C. Marioni. 2017. “Testing for differential abundance in mass cytometry data.” <em>Nat. Methods</em> 14 (7): 707–9.</p>
</div>
<div id="ref-tung2016batch">
<p>Tung, P. Y., J. D. Blischak, C. J. Hsiao, D. A. Knowles, J. E. Burnett, J. K. Pritchard, and Y. Gilad. 2017. “Batch Effects and the Effective Design of Single-Cell Gene Expression Studies.” <em>Sci. Rep.</em> 7 (January): 39921.</p>
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<script id="distill-bibliography" type="text/bibtex">
@Article{stegle2015computational,
    Author="Stegle, O.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="Computational and analytical challenges in single-cell transcriptomics",
    Journal="Nat. Rev. Genet.",
    Year="2015",
    Volume="16",
    Number="3",
    Pages="133--145",
    Month="Mar",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25628217}
}

@Article{islam2014quantitative,
    Author="Islam, S.  and Zeisel, A.  and Joost, S.  and La Manno, G.  and Zajac, P.  and Kasper, M.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="Quantitative single-cell RNA-seq with unique molecular identifiers",
    Journal="Nat. Methods",
    Year="2014",
    Volume="11",
    Number="2",
    Pages="163--166",
    Month="Feb"
}

@Article{huber2015orchestrating,
    Author="Huber, W.  and Carey, V. J.  and Gentleman, R.  and Anders, S.  and Carlson, M.  and Carvalho, B. S.  and Bravo, H. C.  and Davis, S.  and Gatto, L.  and Girke, T.  and Gottardo, R.  and Hahne, F.  and Hansen, K. D.  and Irizarry, R. A.  and Lawrence, M.  and Love, M. I.  and MacDonald, J.  and Obenchain, V.  and Oles, A. K.  and Pag\`es, H.  and Reyes, A.  and Shannon, P.  and Smyth, G. K.  and Tenenbaum, D.  and Waldron, L.  and Morgan, M. ",
    Title="{{O}rchestrating high-throughput genomic analysis with {B}ioconductor}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="2",
    Pages="115--121",
    Month="Feb"
}

@Article{brennecke2013accounting,
    Author="Brennecke, P.  and Anders, S.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Zhang, X.  and Proserpio, V.  and Baying, B.  and Benes, V.  and Teichmann, S. A.  and Marioni, J. C.  and Heisler, M. G. ",
    Title="{{A}ccounting for technical noise in single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2013",
    Volume="10",
    Number="11",
    Pages="1093--1095",
    Month="Nov"
}

@Article{marinov2014singlecell,
    Author="Marinov, G. K.  and Williams, B. A.  and McCue, K.  and Schroth, G. P.  and Gertz, J.  and Myers, R. M.  and Wold, B. J. ",
    Title="{{F}rom single-cell to cell-pool transcriptomes: stochasticity in gene expression and {R}{N}{A} splicing}",
    Journal="Genome Res.",
    Year="2014",
    Volume="24",
    Number="3",
    Pages="496--510",
    Month="Mar"
}

@Article{vallejos2015basics,
    Author="Vallejos, C. A.  and Marioni, J. C.  and Richardson, S. ",
    Title="{{B}{A}{S}i{C}{S}: Bayesian analysis of single-cell sequencing data}",
    Journal="PLoS Comput. Biol.",
    Year="2015",
    Volume="11",
    Number="6",
    Pages="e1004333",
    Month="Jun"
}

@Article{trapnell2014dynamics,
    Author="Trapnell, C.  and Cacchiarelli, D.  and Grimsby, J.  and Pokharel, P.  and Li, S.  and Morse, M.  and Lennon, N. J.  and Livak, K. J.  and Mikkelsen, T. S.  and Rinn, J. L. ",
    Title="{{T}he dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="4",
    Pages="381--386",
    Month="Apr"
}

@Article{kolod2015singlecell,
    Author="Ko\l{}odziejczyk, A. A.  and Kim, J. K.  and Tsang, J. C.  and Ilicic, T.  and Henriksson, J.  and Natarajan, K. N.  and Tuck, A. C.  and Gao, X.  and Buhler, M.  and Liu, P.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{S}ingle cell {R}{N}{A}-sequencing of pluripotent states unlocks modular transcriptional variation}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="17",
    Number="4",
    Pages="471--485",
    Month="Oct"
}

@Article{anders2010differential,
    Author="Anders, S.  and Huber, W. ",
    Title="Differential expression analysis for sequence count data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="10",
    Pages="R106",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20979621}
}

@Article{robinson2010scaling,
    Author="Robinson, M. D.  and Oshlack, A. ",
    Title="A scaling normalization method for differential expression analysis of RNA-seq data",
    Journal="Genome Biol.",
    Year="2010",
    Volume="11",
    Number="3",
    Pages="R25",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20196867}
}

@Article{islam2011characterization,
    Author="Islam, S.  and Kjallquist, U.  and Moliner, A.  and Zajac, P.  and Fan, J. B.  and Lonnerberg, P.  and Linnarsson, S. ",
    Title="{{C}haracterization of the single-cell transcriptional landscape by highly multiplex {R}{N}{A}-seq}",
    Journal="Genome Res.",
    Year="2011",
    Volume="21",
    Number="7",
    Pages="1160--1167",
    Month="Jul"
}

@Article{wilson2015combined,
    Author="Wilson, N. K.  and Kent, D. G.  and Buettner, F.  and Shehata, M.  and Macaulay, I. C.  and Calero-Nieto, F. J.  and Sanchez Castillo, M.  and Oedekoven, C. A.  and Diamanti, E.  and Schulte, R.  and Ponting, C. P.  and Voet, T.  and Caldas, C.  and Stingl, J.  and Green, A. R.  and Theis, F. J.  and Gottgens, B. ",
    Title="{{C}ombined single-cell functional and gene expression analysis resolves heterogeneity within stem cell populations}",
    Journal="Cell Stem Cell",
    Year="2015",
    Volume="16",
    Number="6",
    Pages="712--724",
    Month="Jun"
}

@Article{picelli2014fulllength,
    Author="Picelli, S.  and Faridani, O. R.  and Bjorklund, A. K.  and Winberg, G.  and Sagasser, S.  and Sandberg, R. ",
    Title="{{F}ull-length {R}{N}{A}-seq from single cells using {S}mart-seq2}",
    Journal="Nat Protoc",
    Year="2014",
    Volume="9",
    Number="1",
    Pages="171--181",
    Month="Jan"
}

@Article{pollen2014lowcoverage,
    Author="Pollen, A. A.  and Nowakowski, T. J.  and Shuga, J.  and Wang, X.  and Leyrat, A. A.  and Lui, J. H.  and Li, N.  and Szpankowski, L.  and Fowler, B.  and Chen, P.  and Ramalingam, N.  and Sun, G.  and Thu, M.  and Norris, M.  and Lebofsky, R.  and Toppani, D.  and Kemp, D. W.  and Wong, M.  and Clerkson, B.  and Jones, B. N.  and Wu, S.  and Knutsson, L.  and Alvarado, B.  and Wang, J.  and Weaver, L. S.  and May, A. P.  and Jones, R. C.  and Unger, M. A.  and Kriegstein, A. R.  and West, J. A. ",
    Title="{{L}ow-coverage single-cell m{R}{N}{A} sequencing reveals cellular heterogeneity and activated signaling pathways in developing cerebral cortex}",
    Journal="Nat. Biotechnol.",
    Year="2014",
    Volume="32",
    Number="10",
    Pages="1053--1058",
    Month="Oct"
}

@Article{love2015rnaseq,
    Author="Love, M. I.  and Anders, S.  and Kim, V.  and Huber, W. ",
    Title="{{R}{N}{A}-{S}eq workflow: gene-level exploratory analysis and differential expression}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1070"
}

@Article{liao2014featurecounts,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{feature{C}ounts: an efficient general purpose program for assigning sequence reads to genomic features}",
    Journal="Bioinformatics",
    Year="2014",
    Volume="30",
    Number="7",
    Pages="923--930",
    Month="Apr"
}

@Article{liao2013subread,
    Author="Liao, Y.  and Smyth, G. K.  and Shi, W. ",
    Title="{{T}he {S}ubread aligner: fast, accurate and scalable read mapping by seed-and-vote}",
    Journal="Nucleic Acids Res.",
    Year="2013",
    Volume="41",
    Number="10",
    Pages="e108",
    Month="May"
}

@Article{law2014voom,
    Author="Law, C. W.  and Chen, Y.  and Shi, W.  and Smyth, G. K. ",
    Title="voom: Precision weights unlock linear model analysis tools for RNA-seq read counts",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="2",
    Pages="R29",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/24485249}
}

@Article{love2014moderated,
    Author="Love, M. I.  and Huber, W.  and Anders, S. ",
    Title="Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2",
    Journal="Genome Biol.",
    Year="2014",
    Volume="15",
    Number="12",
    Pages="550",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/25516281}
}

@article{van2008visualizing,
    title={Visualizing data using t-SNE},
    author={Van der Maaten, L. and Hinton, G.},
    journal={J. Mach. Learn. Res.},
    volume={9},
    number={2579-2605},
    pages={85},
    year={2008},
    URL={http://www.jmlr.org/papers/v9/vandermaaten08a.html}
}

@Article{scialdone2015computational,
    Author="Scialdone, A.  and Natarajan, K. N.  and Saraiva, L. R.  and Proserpio, V.  and Teichmann, S. A.  and Stegle, O.  and Marioni, J. C.  and Buettner, F. ",
    Title="Computational assignment of cell-cycle stage from single-cell transcriptome data",
    Journal="Methods",
    Year="2015",
    Volume="85",
    Pages="54--61",
    Month="Sep",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/26142758},
}

@Article{kim2015characterizing,
    Author="Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Illicic, T.  and Teichmann, S. A.  and Marioni, J. C. ",
    Title="{{C}haracterizing noise structure in single-cell {R}{N}{A}-seq distinguishes genuine from technical stochastic allelic expression}",
    Journal="Nat. Commun.",
    Year="2015",
    Volume="6",
    Pages="8687"
}

@Article{mccarthy2012differential,
    Author="McCarthy, D. J.  and Chen, Y.  and Smyth, G. K. ",
    Title="{{D}ifferential expression analysis of multifactor {R}{N}{A}-{S}eq experiments with respect to biological variation}",
    Journal="Nucleic Acids Res.",
    Year="2012",
    Volume="40",
    Number="10",
    Pages="4288--4297",
    Month="May"
}

@Article{angel1991role,
    Author="Angel, P.  and Karin, M. ",
    Title="{{T}he role of {J}un, {F}os and the {A}{P}-1 complex in cell-proliferation and transformation}",
    Journal="Biochim. Biophys. Acta",
    Year="1991",
    Volume="1072",
    Number="2-3",
    Pages="129--157",
    Month="Dec"
}

@Article{langfelder2008defining,
    Author="Langfelder, P.  and Zhang, B.  and Horvath, S. ",
    Title="{{D}efining clusters from a hierarchical cluster tree: the {D}ynamic {T}ree {C}ut package for {R}}",
    Journal="Bioinformatics",
    Year="2008",
    Volume="24",
    Number="5",
    Pages="719--720",
    Month="Mar"
}

@Article{ilicic2016classification,
    Author="Ilicic, T.  and Kim, J. K.  and Ko\l{}odziejczyk, A. A.  and Bagger, F. O.  and McCarthy, D. J.  and Marioni, J. C.  and Teichmann, S. A. ",
    Title="{{C}lassification of low quality cells from single-cell {R}{N}{A}-seq data}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Number="1",
    Pages="29"
}

@Article{heng2008immunological,
    Author="Heng, T. S.  and Painter, M. W.  and Elpek, K.  and Lukacs-Kornek, V.  and Mauermann, N.  and Turley, S. J.  and Koller, D.  and Kim, F. S.  and Wagers, A. J.  and Asinovski, N.  and Davis, S.  and Fassett, M.  and Feuerer, M.  and Gray, D. H.  and Haxhinasto, S.  and Hill, J. A.  and Hyatt, G.  and Laplace, C.  and Leatherbee, K.  and Mathis, D.  and Benoist, C.  and Jianu, R.  and Laidlaw, D. H.  and Best, J. A.  and Knell, J.  and Goldrath, A. W.  and Jarjoura, J.  and Sun, J. C.  and Zhu, Y.  and Lanier, L. L.  and Ergun, A.  and Li, Z.  and Collins, J. J.  and Shinton, S. A.  and Hardy, R. R.  and Friedline, R.  and Sylvia, K.  and Kang, J. ",
    Title="{{T}he {I}mmunological {G}enome {P}roject: networks of gene expression in immune cells}",
    Journal="Nat. Immunol.",
    Year="2008",
    Volume="9",
    Number="10",
    Pages="1091--1094",
    Month="Oct"
}

@Article{zeisel2015brain,
    Author="Zeisel, A.  and Munoz-Manchado, A. B.  and Codeluppi, S.  and Lonnerberg, P.  and La Manno, G.  and Jureus, A.  and Marques, S.  and Munguba, H.  and He, L.  and Betsholtz, C.  and Rolny, C.  and Castelo-Branco, G.  and Hjerling-Leffler, J.  and Linnarsson, S. ",
    Title="{{B}rain structure. {C}ell types in the mouse cortex and hippocampus revealed by single-cell {R}{N}{A}-seq}",
    Journal="Science",
    Year="2015",
    Volume="347",
    Number="6226",
    Pages="1138--1142",
    Month="Mar"
}

@Article{robinson2010edgeR,
    Author="Robinson, M. D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{edge{R}: a {B}ioconductor package for differential expression analysis of digital gene expression data}",
    Journal="Bioinformatics",
    Year="2010",
    Volume="26",
    Number="1",
    Pages="139--140",
    Month="Jan"
}

@Article{ritchie2015limma,
    Author="Ritchie, M. E.  and Phipson, B.  and Wu, D.  and Hu, Y.  and Law, C. W.  and Shi, W.  and Smyth, G. K. ",
    Title="{limma powers differential expression analyses for {R}{N}{A}-sequencing and microarray studies}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="7",
    Pages="e47",
    Month="Apr"
}

@Article{julia2015sincell,
    Author="Julia, M.  and Telenti, A.  and Rausell, A. ",
    Title="{{S}incell: an {R}/{B}ioconductor package for statistical assessment of cell-state hierarchies from single-cell {R}{N}{A}-seq}",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="20",
    Pages="3380--3382",
    Month="Oct"
}

@Article{buettner2015computational,
    Author="Buettner, F.  and Natarajan, K. N.  and Casale, F. P.  and Proserpio, V.  and Scialdone, A.  and Theis, F. J.  and Teichmann, S. A.  and Marioni, J. C.  and Stegle, O. ",
    Title="{{C}omputational analysis of cell-to-cell heterogeneity in single-cell {R}{N}{A}-sequencing data reveals hidden subpopulations of cells}",
    Journal="Nat. Biotechnol.",
    Year="2015",
    Volume="33",
    Number="2",
    Pages="155--160",
    Month="Feb"
}

@Article{mahata2014singlecell,
    Author="Mahata, B.  and Zhang, X.  and Ko\l{}odziejczyk, A. A.  and Proserpio, V.  and Haim-Vilmovsky, L.  and Taylor, A. E.  and Hebenstreit, D.  and Dingler, F. A.  and Moignard, V.  and Gottgens, B.  and Arlt, W.  and McKenzie, A. N.  and Teichmann, S. A. ",
    Title="{{S}ingle-cell {R}{N}{A} sequencing reveals {T} helper cells synthesizing steroids de novo to contribute to immune homeostasis}",
    Journal="Cell Rep.",
    Year="2014",
    Volume="7",
    Number="4",
    Pages="1130--1142",
    Month="May"
}

@Article{phipson2010permutation,
    Author="Phipson, B.  and Smyth, G. K. ",
    Title="{{P}ermutation {P}-values should never be zero: calculating exact {P}-values when permutations are randomly drawn}",
    Journal="Stat. Appl. Genet. Mol. Biol.",
    Year="2010",
    Volume="9",
    Pages="Article 39"
}

@incollection{chen2014differential,
    title="{Differential expression analysis of complex RNA-seq experiments using edgeR}",
    author={Chen, Y. and Lun, A. T. L. and Smyth, G. K.},
    booktitle={Statistical analysis of next generation sequencing data},
    editor={Datta, S. and Nettleton, D.},
    pages={51--74},
    year={2014},
    publisher={Springer}
}

@article{hicks2015widespread,
    author = {Hicks, S. C. and Teng, M. and Irizarry, R. A.},
    title = "On the widespread and critical impact of systematic bias and batch effects in single-cell RNA-Seq data",
    year = {2015},
    doi = {10.1101/025528},
    publisher = {Cold Spring Harbor Labs Journals},
    URL = {http://biorxiv.org/content/early/2015/08/25/025528},
    eprint = {http://biorxiv.org/content/early/2015/08/25/025528.full.pdf},
    journal = {bioRxiv}
}

@Article{leng2015oscope,
    Author="Leng, N.  and Chu, L. F.  and Barry, C.  and Li, Y.  and Choi, J.  and Li, X.  and Jiang, P.  and Stewart, R. M.  and Thomson, J. A.  and Kendziorski, C. ",
    Title="{{O}scope identifies oscillatory genes in unsynchronized single-cell {R}{N}{A}-seq experiments}",
    Journal="Nat. Methods",
    Year="2015",
    Volume="12",
    Number="10",
    Pages="947--950",
    Month="Oct"
}

@Article{fan2016characterizing,
    Author="Fan, J.  and Salathia, N.  and Liu, R.  and Kaeser, G. E.  and Yung, Y. C.  and Herman, J. L.  and Kaper, F.  and Fan, J. B.  and Zhang, K.  and Chun, J.  and Kharchenko, P. V. ",
    Title="{{C}haracterizing transcriptional heterogeneity through pathway and gene set overdispersion analysis}",
    Journal="Nat. Methods",
    Year="2016",
    Volume="13",
    Number="3",
    Pages="241--244",
    Month="Mar"
}

@Article{lund2012detecting,
    Author="Lund, S. P.  and Nettleton, D.  and McCarthy, D. J.  and Smyth, G. K. ",
    Title="{{D}etecting differential expression in {R}{N}{A}-sequence data using quasi-likelihood with shrunken dispersion estimates}",
    Journal="Stat Appl Genet Mol Biol",
    Year="2012",
    Volume="11",
    Number="5",
    Pages=" "
}

@Article{angerer2016destiny,
    Author="Angerer, P.  and Haghverdi, L.  and Buttner, M.  and Theis, F. J.  and Marr, C.  and Buettner, F. ",
    Title="{destiny: diffusion maps for large-scale single-cell data in {R}}",
    Journal="Bioinformatics",
    Year="2016",
    Volume="32",
    Number="8",
    Pages="1241--1243",
}

@Article{lun2016pooling,
    Author="Lun, A. T.  and Bach, K.  and Marioni, J. C. ",
    Title="Pooling across cells to normalize single-cell RNA sequencing data with many zero counts}",
    Journal="Genome Biol.",
    Year="2016",
    Volume="17",
    Pages="75",
    Month="Apr",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/27122128}
}

@Article{zhu2006gata3,
    Author="Zhu, J. and Yamane, H. and Cote-Sierra, J. and Guo, L. and Paul, W. E.",
    Title="{{G}{A}{T}{A}-3 promotes {T}h2 responses through three different mechanisms: induction of {T}h2 cytokine production, selective growth of {T}h2 cells and inhibition of {T}h1 cell-specific factors}",
    Journal="Cell Res.",
    Year="2006",
    Volume="16",
    Number="1",
    Pages="3--10",
    Month="Jan"
}

@article{berger1996bioequivalence,
    author = "Berger, R. L. and Hsu, J. C.",
    doi = "10.1214/ss/1032280304",
    fjournal = "Statistical Science",
    journal = "Statist. Sci.",
    month = "11",
    number = "4",
    pages = "283--319",
    publisher = "The Institute of Mathematical Statistics",
    title = "Bioequivalence trials, intersection-union tests and equivalence confidence sets",
    url = "http://dx.doi.org/10.1214/ss/1032280304",
    volume = "11",
    year = "1996"
}

@Article{santos2015cyclebase,
    Author="Santos, A.  and Wernersson, R.  and Jensen, L. J. ",
    Title="{{C}yclebase 3.0: a multi-organism database on cell-cycle regulation and phenotypes}",
    Journal="Nucleic Acids Res.",
    Year="2015",
    Volume="43",
    Number="Database issue",
    Pages="D1140--1144",
    Month="Jan"
}

@article {tung2016batch,
    Author="Tung, P. Y.  and Blischak, J. D.  and Hsiao, C. J.  and Knowles, D. A.  and Burnett, J. E.  and Pritchard, J. K.  and Gilad, Y. ",
    Title="Batch effects and the effective design of single-cell gene expression studies",
    Journal="Sci. Rep.",
    Year="2017",
    Volume="7",
    Pages="39921",
    Month="Jan"
}

@Article{bray2016near,
    Author="Bray, N. L.  and Pimentel, H.  and Melsted, P.  and Pachter, L. ",
    Title="{{N}ear-optimal probabilistic {R}{N}{A}-seq quantification}",
    Journal="Nat. Biotechnol.",
    Year="2016",
    Volume="34",
    Number="5",
    Pages="525--527",
    Month="May"
}

@article{patro2015accurate,
    author = {Patro, Rob and Duggal, Geet and Kingsford, Carl},
    title = {Accurate, fast, and model-aware transcript expression quantification with Salmon},
    year = {2015},
    doi = {10.1101/021592},
    publisher = {Cold Spring Harbor Labs Journals},
    journal = {bioRxiv}
}

@Article{ji2016tscan,
    Author="Ji, Z.  and Ji, H. ",
    Title="{{T}{S}{C}{A}{N}: {P}seudo-time reconstruction and evaluation in single-cell {R}{N}{A}-seq analysis}",
    Journal="Nucleic Acids Res.",
    Year="2016",
    Volume="44",
    Number="13",
    Pages="e117",
    Month="Jul"
}

@Article{klein2015droplet,
    Author="Klein, A. M.  and Mazutis, L.  and Akartuna, I.  and Tallapragada, N.  and Veres, A.  and Li, V.  and Peshkin, L.  and Weitz, D. A.  and Kirschner, M. W. ",
    Title="{{D}roplet barcoding for single-cell transcriptomics applied to embryonic stem cells}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1187--1201",
    Month="May"
}

@Article{macosko2015highly,
    Author="Macosko, E. Z.  and Basu, A.  and Satija, R.  and Nemesh, J.  and Shekhar, K.  and Goldman, M.  and Tirosh, I.  and Bialas, A. R.  and Kamitaki, N.  and Martersteck, E. M.  and Trombetta, J. J.  and Weitz, D. A.  and Sanes, J. R.  and Shalek, A. K.  and Regev, A.  and McCarroll, S. A. ",
    Title="{{H}ighly parallel genome-wide expression profiling of individual cells using nanoliter droplets}",
    Journal="Cell",
    Year="2015",
    Volume="161",
    Number="5",
    Pages="1202--1214",
    Month="May"
}

@Article{bourgon2010independent,
    Author="Bourgon, R.  and Gentleman, R.  and Huber, W. ",
    Title="Independent filtering increases detection power for high-throughput experiments",
    Journal="Proc. Natl. Acad. Sci. U.S.A.",
    Year="2010",
    Volume="107",
    Number="21",
    Pages="9546--9551",
    Month="May",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/20460310}
}

@Article{finak2015mast,
    Author="Finak, G.  and McDavid, A.  and Yajima, M.  and Deng, J.  and Gersuk, V.  and Shalek, A. K.  and Slichter, C. K.  and Miller, H. W.  and McElrath, M. J.  and Prlic, M.  and Linsley, P. S.  and Gottardo, R. ",
    Title="{{M}{A}{S}{T}: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell {R}{N}{A} sequencing data}",
    Journal="Genome Biol.",
    Year="2015",
    Volume="16",
    Pages="278"
}

@Article{conboy2007cell,
    Author="Conboy, C. M.  and Spyrou, C.  and Thorne, N. P.  and Wade, E. J.  and Barbosa-Morais, N. L.  and Wilson, M. D.  and Bhattacharjee, A.  and Young, R. A.  and Tavare, S.  and Lees, J. A.  and Odom, D. T. ",
    Title="Cell cycle genes are the evolutionarily conserved targets of the E2F4 transcription factor",
    Journal="PLoS ONE",
    Year="2007",
    Volume="2",
    Number="10",
    Pages="e1061",
    URL={https://www.ncbi.nlm.nih.gov/pubmed/17957245}
}

@Article{bertoli2013control,
    Author="Bertoli, C.  and Skotheim, J. M.  and de Bruin, R. A. ",
    Title="Control of cell cycle transcription during G1 and S phases",
    Journal="Nat. Rev. Mol. Cell Biol.",
    Year="2013",
    Volume="14",
    Number="8",
    Pages="518--528",
    Month="Aug",
    URL={{C}ontrol of cell cycle transcription during {G}1 and {S} phases}
}

@Article{chen2016from,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{coller2006new,
    Author="Coller, H. A.  and Sang, L.  and Roberts, J. M. ",
    Title="{{A} new description of cellular quiescence}",
    Journal="PLoS Biol.",
    Year="2006",
    Volume="4",
    Number="3",
    Pages="e83",
    Month="Mar"
}

@Article{mccarthy2017scater,
    Author="McCarthy, D. J.  and Campbell, K. R.  and Lun, A. T.  and Wills, Q. F. ",
    Title="{{S}cater: pre-processing, quality control, normalization and visualization of single-cell {R}{N}{A}-seq data in {R}}",
    Journal="Bioinformatics",
    Year="2017",
    Volume="33",
    Number="8",
    Pages="1179--1186",
    Month="Apr"
}

@article{simes1986improved,
    author = {Simes, R. J.},
    title = {An improved Bonferroni procedure for multiple tests of significance},
    volume = {73},
    number = {3},
    pages = {751--754},
    year = {1986},
    journal = {Biometrika}
}

@Article{xu2015identification,
    Author="Xu, C.  and Su, Z. ",
    Title="Identification of cell types from single-cell transcriptomes using a novel clustering method",
    Journal="Bioinformatics",
    Year="2015",
    Volume="31",
    Number="12",
    Pages="1974--1980",
    Month="Jun"
}

@Article{lun2017assessing,
    Author="Lun, A. T. L.  and Calero-Nieto, F. J.  and Haim-Vilmovsky, L.  and Gottgens, B.  and Marioni, J. C. ",
    Title="Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="11",
    Pages="1795--1806",
    Month="Nov"
}

@Article{yang2016comparative,
    Author="Yang, Z.  and Algesheimer, R.  and Tessone, C. J. ",
    Title="{A comparative analysis of community detection algorithms on artificial networks}",
    Journal="Sci. Rep.",
    Year="2016",
    Volume="6",
    Pages="30750",
    Month="Aug"
}

@Article{smith2017umitools,
    Author="Smith, T.  and Heger, A.  and Sudbery, I. ",
    Title="{{U}{M}{I}-tools: modeling sequencing errors in {U}nique {M}olecular {I}dentifiers to improve quantification accuracy}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="3",
    Pages="491--499",
    Month="Mar"
}

@Article{tasic2016adult,
    Author="Tasic, B.  and Menon, V.  and Nguyen, T. N.  and Kim, T. K.  and Jarsky, T.  and Yao, Z.  and Levi, B.  and Gray, L. T.  and Sorensen, S. A.  and Dolbeare, T.  and Bertagnolli, D.  and Goldy, J.  and Shapovalova, N.  and Parry, S.  and Lee, C.  and Smith, K.  and Bernard, A.  and Madisen, L.  and Sunkin, S. M.  and Hawrylycz, M.  and Koch, C.  and Zeng, H. ",
    Title="{{A}dult mouse cortical cell taxonomy revealed by single cell transcriptomics}",
    Journal="Nat. Neurosci.",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="335--346",
    Month="Feb"
}

@Article{wajapeyee2010senescence,
    Author="Wajapeyee, N.  and Wang, S. Z.  and Serra, R. W.  and Solomon, P. D.  and Nagarajan, A.  and Zhu, X.  and Green, M. R. ",
    Title="{{S}enescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins}",
    Journal="Blood",
    Year="2010",
    Volume="115",
    Number="24",
    Pages="5057--5060",
    Month="Jun"
}

@Article{zeng2012largescale,
    Author="Zeng, H.  and Shen, E. H.  and Hohmann, J. G.  and Oh, S. W.  and Bernard, A.  and Royall, J. J.  and Glattfelder, K. J.  and Sunkin, S. M.  and Morris, J. A.  and Guillozet-Bongaarts, A. L.  and Smith, K. A.  and Ebbert, A. J.  and Swanson, B.  and Kuan, L.  and Page, D. T.  and Overly, C. C.  and Lein, E. S.  and Hawrylycz, M. J.  and Hof, P. R.  and Hyde, T. M.  and Kleinman, J. E.  and Jones, A. R. ",
    Title="{{L}arge-scale cellular-resolution gene profiling in human neocortex reveals species-specific molecular signatures}",
    Journal="Cell",
    Year="2012",
    Volume="149",
    Number="2",
    Pages="483--496",
    Month="Apr"
}

@Article{grun2016denovo,
    Author="Grun, D.  and Muraro, M. J.  and Boisset, J. C.  and Wiebrands, K.  and Lyubimova, A.  and Dharmadhikari, G.  and van den Born, M.  and van Es, J.  and Jansen, E.  and Clevers, H.  and de Koning, E. J. P.  and van Oudenaarden, A. ",
    Title="{{D}e {N}ovo {P}rediction of {S}tem {C}ell {I}dentity using {S}ingle-{C}ell {T}ranscriptome {D}ata}",
    Journal="Cell Stem Cell",
    Year="2016",
    Volume="19",
    Number="2",
    Pages="266--277",
    Month="08"
}

@article{haghverdi2018batch,
    Author="Haghverdi, L.  and Lun, A. T. L.  and Morgan, M. D.  and Marioni, J. C. ",
    Title="Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Month="Apr"
}

@Article{muraro2016singlecell,
    Author="Muraro, M. J.  and Dharmadhikari, G.  and Grun, D.  and Groen, N.  and Dielen, T.  and Jansen, E.  and van Gurp, L.  and Engelse, M. A.  and Carlotti, F.  and de Koning, E. J.  and van Oudenaarden, A. ",
    Title="{{A} {S}ingle-{C}ell {T}ranscriptome {A}tlas of the {H}uman {P}ancreas}",
    Journal="Cell Syst",
    Year="2016",
    Volume="3",
    Number="4",
    Pages="385--394",
    Month="Oct"
}

@Article{segerstolpe2016singlecell,
    Author="Segerstolpe, A.  and Palasantza, A.  and Eliasson, P.  and Andersson, E. M.  and Andreasson, A. C.  and Sun, X.  and Picelli, S.  and Sabirsh, A.  and Clausen, M.  and Bjursell, M. K.  and Smith, D. M.  and Kasper, M.  and Ammala, C.  and Sandberg, R. ",
    Title="{{S}ingle-{C}ell {T}ranscriptome {P}rofiling of {H}uman {P}ancreatic {I}slets in {H}ealth and {T}ype 2 {D}iabetes}",
    Journal="Cell Metab.",
    Year="2016",
    Volume="24",
    Number="4",
    Pages="593--607",
    Month="Oct"
}

@Article{horn1965rationale,
    author="Horn, J. L.",
    title="A rationale and test for the number of factors in factor analysis",
    journal="Psychometrika",
    year="1965",
    month="Jun",
    day="01",
    volume="30",
    number="2",
    pages="179--185",
    issn="1860-0980",
    doi="10.1007/BF02289447",
    url="https://doi.org/10.1007/BF02289447"
}

@Article{buja1992remarks,
    Author="Buja, A.  and Eyuboglu, N. ",
    Title="{{R}emarks on {P}arallel {A}nalysis}",
    Journal="Multivariate Behav Res",
    Year="1992",
    Volume="27",
    Number="4",
    Pages="509--540",
    Month="Oct"
}

@Article{zheng2017massively,
    Author="Zheng, G. X.  and Terry, J. M.  and Belgrader, P.  and Ryvkin, P.  and Bent, Z. W.  and Wilson, R.  and Ziraldo, S. B.  and Wheeler, T. D.  and McDermott, G. P.  and Zhu, J.  and Gregory, M. T.  and Shuga, J.  and Montesclaros, L.  and Underwood, J. G.  and Masquelier, D. A.  and Nishimura, S. Y.  and Schnall-Levin, M.  and Wyatt, P. W.  and Hindson, C. M.  and Bharadwaj, R.  and Wong, A.  and Ness, K. D.  and Beppu, L. W.  and Deeg, H. J.  and McFarland, C.  and Loeb, K. R.  and Valente, W. J.  and Ericson, N. G.  and Stevens, E. A.  and Radich, J. P.  and Mikkelsen, T. S.  and Hindson, B. J.  and Bielas, J. H. ",
    Title="{{M}assively parallel digital transcriptional profiling of single cells}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Pages="14049",
    Month="Jan"
}

@article{lun2018distinguishing,
    author = {Lun, A. and Riesenfeld, S. and Andrews, T. and Dao, T. P. and Gomes, T. and {{participants in the 1\textsuperscript{st} Human Cell Atlas Jamboree}} and Marioni, J.},
    title = {Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data},
    year = {2018},
    doi = {10.1101/234872},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2018/04/04/234872},
    eprint = {https://www.biorxiv.org/content/early/2018/04/04/234872.full.pdf},
    journal = {bioRxiv}
}

@article {wolf2017graph,
    author = {Wolf, F. Alexander and Hamey, Fiona and Plass, Mireya and Solana, Jordi and Dahlin, Joakim S. and Gottgens, Berthold and Rajewsky, Nikolaus and Simon, Lukas and Theis, Fabian J.},
    title = {Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells},
    year = {2017},
    doi = {10.1101/208819},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/10/25/208819},
    eprint = {https://www.biorxiv.org/content/early/2017/10/25/208819.full.pdf},
    journal = {bioRxiv}
}

@Article{lawlor2017singlecell,
    Author="Lawlor, N.  and George, J.  and Bolisetty, M.  and Kursawe, R.  and Sun, L.  and Sivakamasundari, V.  and Kycia, I.  and Robson, P.  and Stitzel, M. L. ",
    Title="{{S}ingle-cell transcriptomes identify human islet cell signatures and reveal cell-type-specific expression changes in type 2 diabetes}",
    Journal="Genome Res.",
    Year="2017",
    Volume="27",
    Number="2",
    Pages="208--222",
    Month="02"
}

@Article{guimaraes2016patterns,
   Author="Guimaraes, J. C.  and Zavolan, M. ",
   Title="{{P}atterns of ribosomal protein expression specify normal and malignant human cells}",
   Journal="Genome Biol.",
   Year="2016",
   Volume="17",
   Number="1",
   Pages="236",
   Month="11"
}

@Article{glare2002betaactin,
   Author="Glare, E. M.  and Divjak, M.  and Bailey, M. J.  and Walters, E. H. ",
   Title="{beta-{A}ctin and {G}{A}{P}{D}{H} housekeeping gene expression in asthmatic airways is variable and not suitable for normalising m{R}{N}{A} levels}",
   Journal="Thorax",
   Year="2002",
   Volume="57",
   Number="9",
   Pages="765--770",
   Month="Sep"
}

@Article{nazari2015gapdh,
   Author="Nazari, F.  and Parham, A.  and Maleki, A. F. ",
   Title="{{G}{A}{P}{D}{H}, \beta-actin and \beta-microglobulin, as three common reference genes, are not reliable for gene expression studies in equine adipose- and marrow-derived mesenchymal stem cells}",
   Journal="J Anim Sci Technol",
   Year="2015",
   Volume="57",
   Pages="18"
}

@Article{soneson2018bias,
    Author="Soneson, C.  and Robinson, M. D. ",
    Title="Bias, robustness and scalability in single-cell differential expression analysis",
    Journal="Nat. Methods",
    URL = {https://www.ncbi.nlm.nih.gov/pubmed/29481549},
    Year="2018",
    Volume="15",
    Number="4",
    Pages="255--261",
    Month="Apr"
}

@Article{soneson2015differential,
    Author="Soneson, C.  and Love, M. I.  and Robinson, M. D. ",
    Title="{{D}ifferential analyses for {R}{N}{A}-seq: transcript-level estimates improve gene-level inferences}",
    Journal="F1000Res",
    Year="2015",
    Volume="4",
    Pages="1521"
}

@article{stoeckius2017hashing,
    author = {Stoeckius, M. and Zheng, S. and Houck-Loomis, B. and Hao, S. and Yeung, B. and Smibert, P. and Satija, R.},
    title = {Cell "hashing" with barcoded antibodies enables multiplexing and doublet detection for single cell genomics},
    year = {2017},
    doi = {10.1101/237693},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2017/12/21/237693},
    eprint = {https://www.biorxiv.org/content/early/2017/12/21/237693.full.pdf},
    journal = {bioRxiv}
}

@Article{kang2018multiplexed,
    Author="Kang, H. M.  and Subramaniam, M.  and Targ, S.  and Nguyen, M.  and Maliskova, L.  and McCarthy, E.  and Wan, E.  and Wong, S.  and Byrnes, L.  and Lanata, C. M.  and Gate, R. E.  and Mostafavi, S.  and Marson, A.  and Zaitlen, N.  and Criswell, L. A.  and Ye, C. J. ",
    Title="{{M}ultiplexed droplet single-cell {R}{N}{A}-sequencing using natural genetic variation}",
    Journal="Nat. Biotechnol.",
    Year="2018",
    Volume="36",
    Number="1",
    Pages="89--94",
    Month="Jan"
}

@Article{dahlin2018single,
    Author="Dahlin, J. S.  and Hamey, F. K.  and Pijuan-Sala, B.  and Shepherd, M.  and Lau, W. W. Y.  and Nestorowa, S.  and Weinreb, C.  and Wolock, S.  and Hannah, R.  and Diamanti, E.  and Kent, D. G.  and Gottgens, B.  and Wilson, N. K. ",
    Title="{{A} single-cell hematopoietic landscape resolves 8 lineage trajectories and defects in {K}it mutant mice}",
    Journal="Blood",
    Year="2018",
    Volume="131",
    Number="21",
    Pages="e1-e11",
    Month="May"
}

@Article{bach2017differentiation,
    Author="Bach, K.  and Pensa, S.  and Grzelak, M.  and Hadfield, J.  and Adams, D. J.  and Marioni, J. C.  and Khaled, W. T. ",
    Title="{{D}ifferentiation dynamics of mammary epithelial cells revealed by single-cell {R}{N}{A} sequencing}",
    Journal="Nat Commun",
    Year="2017",
    Volume="8",
    Number="1",
    Pages="2128",
    Month="Dec"
}

@article{tian2018scpipe,
  title={scPipe: A flexible R/Bioconductor preprocessing pipeline for single-cell RNA-sequencing data},
  author={Tian, Luyi and Su, Shian and Dong, Xueyi and Amann-Zalcenstein, Daniela and Biben, Christine and Seidi, Azadeh and Hilton, Douglas J and Naik, Shalin H and Ritchie, Matthew E},
  journal={PLoS computational biology},
  volume={14},
  number={8},
  pages={e1006361},
  year={2018},
  publisher={Public Library of Science},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/30096152}
}

@article{kiselev2017sc3,
  title={SC3: consensus clustering of single-cell RNA-seq data},
  author={Kiselev, Vladimir Yu and Kirschner, Kristina and Schaub, Michael T and Andrews, Tallulah and Yiu, Andrew and Chandra, Tamir and Natarajan, Kedar N and Reik, Wolf and Barahona, Mauricio and Green, Anthony R and others},
  journal={Nature methods},
  volume={14},
  number={5},
  pages={483},
  year={2017},
  publisher={Nature Publishing Group},
  URL={https://www.ncbi.nlm.nih.gov/pubmed/28346451}
}

@ARTICLE{McInnes2018-hy,
  title         = "UMAP: Uniform Manifold Approximation and Projection for
                   Dimension Reduction",
  author        = "McInnes, Leland and Healy, John and Melville, James",
  abstract      = "UMAP (Uniform Manifold Approximation and Projection) is a
                   novel manifold learning technique for dimension reduction.
                   UMAP is constructed from a theoretical framework based in
                   Riemannian geometry and algebraic topology. The result is a
                   practical scalable algorithm that applies to real world
                   data. The UMAP algorithm is competitive with t-SNE for
                   visualization quality, and arguably preserves more of the
                   global structure with superior run time performance.
                   Furthermore, UMAP has no computational restrictions on
                   embedding dimension, making it viable as a general purpose
                   dimension reduction technique for machine learning.",
  month         =  feb,
  year          =  2018,
  url           = "http://arxiv.org/abs/1802.03426",
  archivePrefix = "arXiv",
  eprint        = "1802.03426",
  primaryClass  = "stat.ML",
  arxivid       = "1802.03426"
}

@article{lun2017overcoming,
  title={Overcoming confounding plate effects in differential expression analyses of single-cell RNA-seq data},
  author={Lun, Aaron TL and Marioni, John C},
  journal={Biostatistics},
  volume={18},
  number={3},
  pages={451--464},
  year={2017},
  publisher={Oxford University Press}
}

@Article{lun2017testing,
    Author="Lun, A. T. L.  and Richard, A. C.  and Marioni, J. C. ",
    Title="{{T}esting for differential abundance in mass cytometry data}",
    Journal="Nat. Methods",
    Year="2017",
    Volume="14",
    Number="7",
    Pages="707--709",
    Month="Jul"
}

@Article{chen2016reads,
    Author="Chen, Y.  and Lun, A. T.  and Smyth, G. K. ",
    Title="{{F}rom reads to genes to pathways: differential expression analysis of {R}{N}{A}-{S}eq experiments using {R}subread and the edge{R} quasi-likelihood pipeline}",
    Journal="F1000Res",
    Year="2016",
    Volume="5",
    Pages="1438"
}

@Article{phipson2016robust,
    Author="Phipson, B. and Lee, S. and Majewski, I. J. and Alexander, W. S. and Smyth, G. K.",
    year = 2016,
    title = "Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression",
    Journal="Ann. Appl. Stat.",
    Volume=10,
    Number=2,
    Pages="946-963"
}

@Article{aran2019reference,
   Author="Aran, D.  and Looney, A. P.  and Liu, L.  and Wu, E.  and Fong, V.  and Hsu, A.  and Chak, S.  and Naikawadi, R. P.  and Wolters, P. J.  and Abate, A. R.  and Butte, A. J.  and Bhattacharya, M. ",
   Title="Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage",
   Journal="Nat. Immunol.",
   Year="2019",
   Volume="20",
   Number="2",
   Pages="163--172",
   Month="02"
}

@article{Benayoun2019,
author = {Benayoun, B{\'{e}}r{\'{e}}nice A. and Pollina, Elizabeth A. and Singh, Param Priya and Mahmoudi, Salah and Harel, Itamar and Casey, Kerriann M. and Dulken, Ben W. and Kundaje, Anshul and Brunet, Anne},
doi = {10.1101/gr.240093.118},
journal = {Genome Research},
volume = {29},
pages = {697-709},
title = {Remodeling of epigenome and transcriptome landscapes with aging in mice reveals widespread induction of inflammatory responses},
year = {2019}
}

@article{ImmGenRef,
    author = {Heng, Tracy S.P. and Painter, Michio W. and Elpek, Kutlu and Lukacs-Kornek, Veronika and Mauermann, Nora and Turley, Shannon J. and Koller, Daphne and Kim, Francis S. and Wagers, Amy J. and Asinovski, Natasha and Davis, Scott and Fassett, Marlys and Feuerer, Markus and Gray, Daniel H.D. and Haxhinasto, Sokol and Hill, Jonathan A. and Hyatt, Gordon and Laplace, Catherine and Leatherbee, Kristen and Mathis, Diane and Benoist, Christophe and Jianu, Radu and Laidlaw, David H. and Best, J. Adam and Knell, Jamie and Goldrath, Ananda W. and Jarjoura, Jessica and Sun, Joseph C. and Zhu, Yanan and Lanier, Lewis L. and Ergun, Ayla and Li, Zheng and Collins, James J. and Shinton, Susan A. and Hardy, Richard R. and Friedline, Randall and Sylvia, Katelyn and Kang, Joonsoo},
    journal = {Nature Immunology},
    volume = {9},
    issue = {10},
    pages = {1091-4},
    doi = {10.1038/ni1008-1091},
    title = {The immunological genome project: Networks of gene expression in immune cells},
    year = {2008}
}

@article{choi2018haemopedia,
  title={Haemopedia RNA-seq: a database of gene expression during haematopoiesis in mice and humans},
  author={Choi, Jarny and Baldwin, Tracey M and Wong, Mae and Bolden, Jessica E and Fairfax, Kirsten A and Lucas, Erin C and Cole, Rebecca and Biben, Christine and Morgan, Clare and Ramsay, Kerry A and others},
  journal={Nucleic acids research},
  volume={47},
  number={D1},
  pages={D780--D785},
  year={2018},
  publisher={Oxford University Press}
}

@Article{mccarthy2009treat,
    Author="McCarthy, D. J.  and Smyth, G. K. ",
    Title="{Testing significance relative to a fold-change threshold is a TREAT}",
    Journal="Bioinformatics",
    Year="2009",
    Volume="25",
    Number="6",
    Pages="765--771",
    Month="Mar"
}

@article{hpcaRef,
author = {Mabbott, Neil A. and Baillie, J. K. and Brown, Helen and Freeman, Tom C. and Hume, David A.},
doi = {10.1186/1471-2164-14-632},
journal = {BMC Genomics},
volume = {14},
title = {{An expression atlas of human primary cells: Inference of gene function from coexpression networks}},
year = {2013}
}

@article{blueprintRef,
    author = {Martens, Joost H A and Stunnenberg, Hendrik G.},
    doi = {10.3324/haematol.2013.094243},
    journal = {Haematologica},
    title = {{BLUEPRINT: Mapping human blood cell epigenomes}},
    year = {2013},
    volume = {98},
    pages = {1487-1489}
}

@article{encodeRef,
author = {{The ENCODE Project Consortium}},
doi = {10.1038/nature11247},
issn = {14764687},
journal = {Nature},
title = {{An integrated encyclopedia of DNA elements in the human genome}},
year = {2012}
}

@article{lun2016step,
  title={A step-by-step workflow for low-level analysis of single-cell {RNA-seq} data},
  author={Lun, A. T. L. and McCarthy, D. J. and Marioni, J. C.},
  journal ={F1000Res.},
  volume={5},
  month={aug},
  year={2016}
}
</script>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
