---
title: "Multi-sample comparisons with Grant (C075) retinal epithelial cells data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: Single Cell Open Research Endeavour (SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(SingleCellExperiment)
library(here)
library(scater)
library(cowplot)
library(patchwork)
library(BiocParallel)
library(pheatmap)

source(here("code", "helper_functions.R"))

# NOTE: Using multiple cores seizes up my laptop. Can use more on unix box.
options("mc.cores" = ifelse(Sys.info()[["nodename"]] == "PC1331", 2L, 8L))
register(MulticoreParam(workers = getOption("mc.cores")))

knitr::opts_chunk$set(
  fig.path = "C075_Grant_Coultas.multi-sample_comparisons_files/")
```

# Motivation

A powerful use of scRNA-seq technology lies in the design of replicated multi-condition experiments to detect changes in composition or expression between conditions.
For example, a researcher could use this strategy to detect changes in cell type abundance after drug treatment [@richard2018tcell] or genetic modifications [@scialdone2016resolving].
This provides more biological insight than conventional scRNA-seq experiments involving only one biological condition, especially if we can relate population changes to specific experimental perturbations.

Differential analyses of multi-condition scRNA-seq experiments can be broadly split into two categories - differential expression (DE) and differential abundance (DA) analyses.
The former tests for changes in expression between conditions for cells of the same type that are present in both conditions, while the latter tests for changes in the composition of cell types (or states, etc.) between conditions.

For this experiment, we are interested in identifying differences between `genotype`.
We are not interested in the differences due to `plate_number`, `sex`, and `mouse`, so we will block on these variables where possible.

# Setting up the data

We start from the preprocessed *SingleCellExperiment* object created in ['Annotating the Grant (C075) retinal epithelial cells data set'](C075_Grant_Coultas.annotate.html).

```{r}
sce <- readRDS(here("data", "SCEs", "C075_Grant_Coultas.annotated.SCE.rds"))

# Define what constitutes a sample (i.e. replicate) in this experiment.
sce$sample <- paste0(sce$plate_number, ".", sce$mouse)

# Some useful colours
plate_number_colours <- setNames(
  unique(sce$plate_number_colours),
  unique(names(sce$plate_number_colours)))
sample_type_colours <- setNames(
  unique(sce$sample_type_colours),
  unique(names(sce$sample_type_colours)))
genotype_colours <- setNames(
  unique(sce$genotype_colours),
  unique(names(sce$genotype_colours)))
mouse_colours <- setNames(
  unique(sce$mouse_colours),
  unique(names(sce$mouse_colours)))
sex_colours <- setNames(
  unique(sce$sex_colours),
  unique(names(sce$sex_colours)))
sequencing_run_colours <- setNames(
  unique(sce$sequencing_run_colours),
  unique(names(sce$sequencing_run_colours)))
cluster_colours <- setNames(
  unique(sce$cluster_colours),
  unique(names(sce$cluster_colours)))
label_colours <- setNames(
  unique(sce$label_colours),
  unique(names(sce$label_colours)))

# Some useful gene sets
mito_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME == "MT")]

ribo_set <- grep("^Rp(s|l)", rownames(sce), value = TRUE)
# NOTE: A more curated approach for identifying ribosomal protein genes 
#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)
library(msigdbr)
c2_sets <- msigdbr(species = "Mus musculus", category = "C2")
ribo_set <- union(
  ribo_set,
  c2_sets[c2_sets$gs_name == "KEGG_RIBOSOME", ]$gene_symbol)

sex_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME %in% c("X", "Y"))]

pseudogene_set <- rownames(sce)[
  any(grepl("pseudogene", rowData(sce)$ENSEMBL.GENEBIOTYPE))]
```

As a reminder, Zoe requested that clusters 3, 4, and 6 be aggregated to create cell labels for the purposes of differential analyses.
Figure \@ref(fig:collabels-umap) shows the cluster labels and the cell labels (i.e. Zoe's aggregated cluster labels) on the UMAP plot.

```{r collabels-umap, fig.cap = "UMAP plot, where each point represents a cell and is coloured according to the legend.", fig.asp = 2 / 5}
p1 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = cluster), size = 0.5) +
  scale_colour_manual(values = cluster_colours) + 
  theme_cowplot(font_size = 8)
p2 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = label), size = 0.5) +
  scale_colour_manual(values = label_colours) + 
  theme_cowplot(font_size = 8)
p1 | p2
```

Figure \@ref(fig:collabels-barplot) breaks down the cell labels by key experimental factors.

```{r collabels-barplot, fig.asp = 1, fig.cap = "Breakdown of cell labels by experimental factors."}
p1 <- ggcells(sce) + 
  geom_bar(
    aes(x = label, fill = genotype),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggcells(sce) + 
  geom_bar(
    aes(x = label, fill = sex),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sex_colours) +
  theme_cowplot(font_size = 8)
p3 <- ggcells(sce) + 
  geom_bar(
    aes(x = label, fill = plate_number),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = plate_number_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggcells(sce) + 
  geom_bar(
    aes(x = label, fill = mouse),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = mouse_colours) +
  theme_cowplot(font_size = 8)
p5 <- ggcells(sce) + 
  geom_bar(
    aes(x = label, fill = plate_number),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = plate_number_colours) +
  theme_cowplot(font_size = 8)
p6 <- ggcells(sce) + 
  geom_bar(aes(x = label, fill = label)) +
  coord_flip() +
  ylab("Number of cells") +
  scale_fill_manual(values = label_colours) +
  theme_cowplot(font_size = 8) + 
  guides(fill = FALSE)

(p1 | p2) / (p3 | p4) / (p5 | p6)
```

Figure \@ref(fig:plate-number-barplot) is a reminder that the mice are partially matched by genotype (and sibship) within each plate.

```{r plate-number-barplot, fig.asp = 1 / 2, fig.cap = "Breakdown of plates by experimental factors."}
p1 <- ggcells(sce) + 
  geom_bar(
    aes(x = plate_number, fill = genotype),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 8)

p2 <- ggcells(sce) + 
  geom_bar(
    aes(x = plate_number, fill = mouse),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = mouse_colours) +
  theme_cowplot(font_size = 8)

p1 | p2
```

# Differential abundance between conditions

## Overview

In a DA analysis, we test for significant changes in per-label cell abundance across conditions.
This will reveal which cell types are depleted or enriched with differences in genotype, which is arguably just as interesting as changes in expression within each cell type.
The DA analysis has a long history in flow cytometry [@finak2015mast; @lun2017testing] where it is routinely used to examine the effects of different conditions on the composition of complex cell populations.
By performing it here, we effectively treat scRNA-seq as a "super-FACS" technology for defining relevant subpopulations using the entire transcriptome.

We prepare for the DA analysis by quantifying the number of cells assigned to each label.
The table below and Figure \@ref(fig:cluster-barplot-by-plate) summarise the number of cells assigned to each label per plate.
In this case, we are aiming to identify labels that change in abundance among the `Mutant` cells compared to the `Control` cells.
Visually, `cluster_5` has a strong and consistent change in abundance between genotypes.

```{r}
abundances <- table(sce$label, sce$sample)
abundances <- unclass(abundances)

abundances_df <- as.data.frame(colData(sce)) %>%
  dplyr::count(label, sample, genotype) %>%
  tidyr::pivot_wider(
    names_from = label,
    id_cols = c(genotype, sample), 
    values_from = n,
    values_fill = 0L) %>%
  dplyr::arrange(genotype)
knitr::kable(
  abundances_df,
  fig.cap = "Number of cells assigned to each label from each sample.")
```

```{r cluster-barplot-by-plate, fig.asp = 1 / 2, fig.cap = "Breakdown of labels by experimental factors.", layout = "l-page"}
p1 <- ggcells(sce, aes(x = label, fill = genotype)) + 
  geom_bar(position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 5) + 
  facet_grid(~plate_number)

p2 <- ggcells(sce, aes(x = label, fill = genotype)) + 
  geom_bar() +
  coord_flip() +
  ylab("Number of cells") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 5) + 
  facet_grid(~plate_number)

p1 / p2
```

## Performing the DA anlaysis

### Overview

Our DA analysis will be performed with the `r BiocStyle::Biocpkg("edgeR")` package. 
This allows us to take advantage of the negative binomial generalized linear model (NB GLM) methods to model overdispersed count data in the presence of limited replication - except that the counts are not of reads per gene, but of cells per label [@lun2017testing]. 
The aim is to share information across labels to improve our estimates of the biological variability in cell abundance between replicates.

```{r}
library(edgeR)

# Attaching some column metadata.
extra_info <- colData(sce)[match(colnames(abundances), sce$sample),]
y_ab <- DGEList(abundances, samples = extra_info)
```

### Statistical methodology

#### Pre-processing

We filter out low-abundance labels to avoid cluttering the result table with very rare subpopulations that contain only a handful of cells.

```{r}
# Constructing the design matrix.
design <- model.matrix(~plate_number + genotype, y_ab$samples)
colnames(design) <- sub("plate\\_number|genotype", "", colnames(design))

keep <- filterByExpr(y_ab, design = design)
y_ab <- y_ab[keep, ]
```

For a DA analysis of cluster abundances, filtering is generally not required as most clusters will not be of low-abundance (otherwise there would not have been enough evidence to define the cluster in the first place), but in this case we filter out `r names(which(!keep))`.

#### Statistical methodology

Unlike a differential expression analyses, we do not perform an additional normalization step.
This means that we are only normalizing based on the "library size", i.e., the total number of cells in each sample.
Any changes we detect between conditions will subsequently represent differences in the proportion of cells in each cluster.
The motivation behind this decision is discussed in more detail in [Handling composition effects].

We set up the design matrix to block on the sample-to-sample differences across different mice and plates, while retaining an additive term that represents the effect of genotype.
Here, the log-fold change in our model refers to the change in cell abundance associated with genotype, rather than the change in gene expression.

<aside>
Plate `LC279` effectively does not contribute to this analysis because it only contains `Control` cells.
</aside>

```{r}
# NOTE: Design matrix actually constructed earlier.
```

We estimate the NB dispersion and QL dispersion for each label^[In estimating both the NB and QL dispersion we turn off the trend estimation as we do not have enough points for its stable estimation].

```{r}
y_ab <- estimateDisp(y_ab, design, trend = "none")
fit_ab <- glmQLFit(y_ab, design, robust = TRUE, abundance.trend = FALSE)
```

## Analysis using cell labels

```{r}
fdr <- 0.05
```

We test for differences in abundance between `Mutant` and `Control` samples using
`glmQLFTest()` function from the `r BiocStyle::Biocpkg("edgeR")` package to obtain a list of genotype-associated labels.

```{r}
res <- glmQLFTest(fit_ab, coef = "Mutant")
```

We find one label, `cluster_5`, is strongly associated with genotype at a false discovery rate of `r scales::percent(fdr)`, as shown in the table below.

```{r}
knitr::kable(
  as.data.frame(topTags(res, n = Inf)), 
  caption = "Results of DA analysis applied to each label.")
```

## Handling composition effects

Our default approach for normalization in a differential abundance analysis is to only normalize based on the total number of cells in each sample, which means that we are effectively testing for differential proportions between conditions.

Unfortunately, the use of the total number of cells leaves us susceptible to composition effects.
For example, a large increase in abundance for one cell subpopulation will introduce decreases in proportion for all other subpopulations - which is technically correct, but may be misleading if one concludes that those other subpopulations are decreasing in abundance of their own volition.
If composition biases are proving problematic for interpretation of DA results, we have [several avenues for removing them or mitigating their impact](https://osca.bioconductor.org/multi-sample-comparisons.html#composition-effects).

# Differential expression between conditions

## Creating pseudo-bulk samples

The most obvious differential analysis is to look for changes in expression between conditions.
We perform the DE analysis separately for each label to identify cell type-specific transcriptional effects of genotype.
The actual DE testing is performed on "pseudo-bulk" expression profiles [@tung2016batch], generated by summing counts together for all cells with the same combination of label and sample.
This leverages the resolution offered by single-cell technologies to define the labels, and combines it with the statistical rigor of existing methods for DE analyses involving a small number of samples.

```{r}
# Define colData columns to be kept for pseudo-bulked data.
columns_to_keep <- c("mouse", "plate_number", "genotype", "sex")

summed <- aggregateAcrossCells(
  sce,
  id = colData(sce)[, c("label", "sample", columns_to_keep)],
  coldata_merge = FALSE)

summed <- logNormCounts(summed)
colnames(summed) <- paste0(summed$label, ".", summed$sample)
```

At this point, it is worth reflecting on the motivations behind the use of pseudo-bulking:

- Larger counts are more amenable to standard DE analysis pipelines designed for bulk RNA-seq data. Normalization is more straightforward and certain statistical approximations are more accurate^[E.g., the saddlepoint approximation for quasi-likelihood methods or normality for linear models.].
- Collapsing cells into samples reflects the fact that our biological replication occurs at the sample level [@lun2017overcoming]. Each sample is represented no more than once for each condition, avoiding problems from unmodelled correlations between samples. Supplying the per-cell counts directly to a DE analysis pipeline would imply that each cell is an independent biological replicate, which is not true from an experimental perspective^[A mixed effects model can handle this variance structure but involves extra statistical and computational complexity for little benefit, see Crowell *et al*.].
- Variance between cells within each sample is masked, provided it does not affect variance across (replicate) samples. This avoids penalizing DEGs that are not uniformly up- or down-regulated for all cells in all samples of one condition. Masking is generally desirable as DEGs - unlike marker genes - do not need to have low within-sample variance to be interesting, e.g., if the treatment effect is consistent across replicate populations but heterogeneous on a per-cell basis^[Of course, high per-cell variability will still result in weaker DE if it affects the variability across populations, while homogeneous per-cell responses will result in stronger DE due to a larger population-level log-fold change. These effects are also largely desirable.].

## Performing the DE analysis

### Overview

The DE analysis will be performed using quasi-likelihood (QL) methods from the `r BiocStyle::Biocpkg("edgeR")` package [@robinson2010edgeR;@chen2014differential].
This uses a negative binomial generalized linear model (NB GLM) to handle overdispersed count data in experiments with limited replication.
In our case, we have biological variation with four replicates per condition, with the replicates for each condition partially paired as genotype-discordant siblings within each plate, so `r BiocStyle::Biocpkg("edgeR")` (or its contemporaries) is a natural choice for the analysis.

We do not use all labels for GLM fitting as the strong DE between labels makes it difficult to compute a sensible average abundance to model the mean-dispersion trend.
Moreover, label-specific batch effects would not be easily handled with a single additive term in the design matrix for the batch.

### Statistical methodology and considerations

#### Pre-processing

```{r}
min_ncells <- 10
```

A typical step in bulk RNA-seq data analyses is to remove samples with very low library sizes due to failed library preparation or sequencing.
The very low counts in these samples can be troublesome in downstream steps such as normalization or for some statistical approximations used in the DE analysis.
In our situation, this is equivalent to removing label-sample combinations that have very few or lowly-sequenced cells.
The exact definition of "very low" will vary, but in this case we remove combinations containing fewer than `r min_ncells` cells.
This is lower than generally recommended^[E.g., Crowell *et al*. note that "$\approx100$ cells is sufficient to reach decent performance" with pseudobulk DE methods and that "there is a sizable gain in performance in going from 50 to 100 cells" per label per sample.] but here we are limited by the small numbers of cells/sample that were processed.

Another typical step in bulk RNA-seq analyses is to remove genes that are lowly expressed.
This reduces computational work, improves the accuracy of mean-variance trend modelling and decreases the severity of the multiple testing correction.
Genes are discarded if they are not expressed above a log-CPM threshold in a minimum number of samples (determined from the size of the smallest treatment group in the experimental design).

Finally, we correct for composition biases by computing normalization factors with the trimmed mean of M-values method [@robinson2010scaling].
We do not need the bespoke single-cell methods used earlier in this analysis, as the counts for our pseudo-bulk samples are large enough to apply bulk normalization methods.

#### Statistical modelling

We set up the design matrix to block on the sample-to-sample differences across different mice and plates, while retaining an additive term that represents the effect of genotype.
The latter is represented in our model as the log-fold change in gene expression in `Mutant` cells over their `Control` counterparts within the same label.
Our aim is to test whether this log-fold change is significantly different from zero.

We estimate the negative binomial (NB) dispersions to model the mean-variance trend, which is not easily accommodated by QL dispersions alone due to the quadratic nature of the NB mean-variance trend.
We also estimate the quasi-likelihood dispersions to model the uncertainty and variability of the per-gene variance, which is not well handled by the NB dispersions.
So the two dispersion types complement each other in the final analysis.

```{r}
fdr <- 0.05
```

We test for differences in expression due to genotype.
DEGs are defined as those with non-zero log-fold changes at a false discovery rate of `r scales::percent(fdr)`.

## Analysis using cell labels

As noted in [Setting up the data], we start by using the aggregated cluster labels as the cell labels.

```{r}
# Pulling out a sample-level 'targets' data.frame.
targets <- colData(summed)[!duplicated(summed$sample),]

# Constructing the design matrix.
design <- model.matrix(~plate_number + genotype, targets)
colnames(design) <- sub("plate\\_number|genotype", "", colnames(design))
rownames(design) <- targets$sample
```

Now that we have laid out the theory underlying the DE analysis, we repeat this process for each of the labels to identify genotype-associated DE in each cell type.
To prepare for this, we filter out all sample-label combinations with insufficient cells.

```{r}
summed_filt <- summed[, summed$ncells >= min_ncells]
```

We then apply the `pseudoBulkDGE()` function from the `r BiocStyle::Biocpkg("scran")` package to obtain a list of genotype-associated DE genes for each label.
This function puts some additional effort into automatically dealing with labels that are not represented in both `Mutant` and `Control` cells, for which a DE analysis between conditions is meaningless or are not represented in a sufficient number of replicate samples to enable modelling of biological variability.

```{r}
library(scran)
de_results <- pseudoBulkDGE(
  summed_filt, 
  sample = summed_filt$sample,
  label = summed_filt$label,
  design = design,
  coef = "Mutant",
  condition = summed_filt$genotype)
```

Figure \@ref(fig:degs-heatmap) summarises the DEGs using a heatmap of pseudobulk-level log-expression values.
In general, there seems to be little differential expression that is associated with genotype.

```{r degs-heatmap, fig.cap = "Heatmap of pseudobulk-level log-expression values normalized to the mean of all samples in the filtered data; rows correspond to genes, columns to label-sample combinations. Included is the union of DEGs detected (FDR < `r fdr`) across all labels. Data is split horizontally by label and rows are annotated to show in which label(s) each gene is DE.", fig.asp = 1.5}
o <- order(summed_filt$label, summed_filt$genotype, summed_filt$sample)

all_features <- lapply(
  de_results, 
  function(x) rownames(x[which(x$FDR < fdr), ]))
features <- unique(unlist(all_features))
mat <- logcounts(summed_filt)[features, ]
mat <- mat - rowMeans(mat)
pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_row = data.frame(
    cluster_7 = ifelse(features %in% all_features$cluster_7, "DE", "not DE"),
    cluster_5 = ifelse(features %in% all_features$cluster_5, "DE", "not DE"),
    clusters_3_4_6 = ifelse(features %in% all_features$clusters_3_4_6, "DE", "not DE"),
    cluster_2 = ifelse(features %in% all_features$cluster_2, "DE", "not DE"),
    cluster_1 = ifelse(features %in% all_features$cluster_1, "DE", "not DE"),
    row.names = features),
  annotation_col = data.frame(
    label = summed_filt$label[o],
    genotype = summed_filt$genotype[o],
    plate_number = summed_filt$plate_number[o],
    sex = summed_filt$sex[o],
    row.names = colnames(summed_filt)[o]),
  annotation_colors = list(
    label = label_colours[levels(summed_filt$label)],
    genotype = genotype_colours,
    plate_number = plate_number_colours,
    sex = sex_colours,
    cluster_1 = c("not DE" = "white", "DE" = label_colours[["cluster_1"]]),
    cluster_2 = c("not DE" = "white", "DE" = label_colours[["cluster_2"]]),
    clusters_3_4_6 = c("not DE" = "white", "DE" = label_colours[["clusters_3_4_6"]]),
    cluster_5 = c("not DE" = "white", "DE" = label_colours[["cluster_5"]]),
    cluster_7 = c("not DE" = "white", "DE" = label_colours[["cluster_7"]])),
  breaks = seq(-max(abs(mat)), max(abs(mat)), length.out = 101),
  fontsize = 8,
  gaps_col = cumsum(runLength(Rle(summed_filt$label[o]))))
```

The following table gives the number of DEGs at a FDR of `r scales::percent(fdr)` for each label.

```{r}
is_de <- decideTestsPerLabel(de_results, threshold = fdr)
summarizeTestsPerLabel(is_de) %>%
  knitr::kable(
    caption = "Number of DEGs per label. Each row corresponds to a label and each column corresponds to the number of downregulated genes (`-1`), the number of non-differentially expressed genes (`0`), the number of upregulated genes (`1`), and the number of genes not tested (`NA`).")
```

```{r, message = FALSE}
# Create outputs
deg_dir <- here("output", "DEGs")
dir.create(deg_dir)
createDEGOutputs(outdir = deg_dir, de_results = de_results)
```

Note that genes listed as `NA` in the above table were either filtered out as low-abundance genes for a given label’s analysis, or the comparison of interest was not possible for a particular label, e.g., due to lack of residual degrees of freedom or an absence of samples from both conditions.
If it is necessary to extract statistics in the absence of replicates, several strategies can be applied such as reducing the complexity of the model or using a predefined value for the NB dispersion. We could also consider reducing the minimum number of cells per sample, although we are already using a rather low minimum.

Please see [`output/DEGs/`](../output/DEGs/) for spreadsheets and interactive `r BiocStyle::Biocpkg("Glimma")` plots of these DEG lists.
The `r BiocStyle::Biocpkg("Glimma")` plots show the pseudobulk data for that label on the lefthand side and the single-cell data for that label on the righthand side.
The single-cell data are plotted per sample using abbreviated sample labels, e.g., `C.LC280` means "control mouse on plate LC280", `M.LC280` means "mutate mouse on plate LC280", etc.

<aside>
Note that the spreadsheets contain all labels and genes whereas the Glimma plots only contain labels and genes actually tested for DE. The column names in the Glimma tables may be abbreviated.
</aside>

## Analysis using 'coarser' labels

As noted in [Pre-processing], by only requiring `r min_ncells` cells/label/mouse we are using fewer cells/label/sample than the generally recommended minimum of 100.
If we had more cells/label/sample then we would expect to increase our true positive rate, i.e. we would have greater statistical power to identify DEGs [@crowell2019discovery].
Fortunately, @crowell2019discovery also found that the false discovery rate is not greatly affected by the choice of minimum number of cells, so we are probably not at risk of making more false discoveries by using a low minimum number of cells/label/sample.

To increase the number of cells/label/sample would require one or more of:

1. The acquisition of additional cells (i.e. more experiments)
2. The use of 'coarser' labels (i.e. aggregation of the existing labels)
3. The use of 'coarser' samples (i.e. aggregation of the existing samples)

The first is probably not feasible while the latter ignores the known biological (e.g., mouse) and technical (e.g., plate) variability between replicates samples.
We are therefore left with using 'coarser' labels.

To generate 'coarser' labels, we can simply ignore the clustering results entirely (i.e. the label is `ignoring_clusters`).
In doing so we are implicitly accepting that any DEGs identified in the subsequent analysis are at increased chance of reflecting differences in 'cell type' between genotypes (i.e. differential abundance) rather than differential gene expression between genotypes *per se*.

```{r}
colLabels(sce) <- "ignoring_clusters"

# Define colData columns to be kept for pseudo-bulked data.
columns_to_keep <- c("mouse", "plate_number", "genotype", "sex")

summed <- aggregateAcrossCells(
  sce,
  id = colData(sce)[, c("label", "sample", columns_to_keep)],
  coldata_merge = FALSE)

summed <- logNormCounts(summed)
colnames(summed) <- paste0(summed$label, ".", summed$sample)

targets <- colData(summed)[!duplicated(summed$sample),]

# Constructing the design matrix.
design <- model.matrix(~plate_number + genotype, targets)
colnames(design) <- sub("plate\\_number|genotype", "", colnames(design))
rownames(design) <- targets$sample

summed_filt <- summed[, summed$ncells >= min_ncells]

de_results <- pseudoBulkDGE(
  summed_filt, 
  sample = summed_filt$sample,
  label = summed_filt$label,
  design = design,
  coef = "Mutant",
  condition = summed_filt$genotype)
```

Figure \@ref(fig:degs-ignoring-clusters-heatmap) summarises the DEGs using a heatmap of pseudobulk-level log-expression values.
There are more DEGs detected when we use the 'coarser' labels.

```{r degs-ignoring-clusters-heatmap, fig.cap = "Heatmap of pseudobulk-level log-expression values normalized to the mean of all samples in the filtered data; rows correspond to genes, columns to label-sample combinations. Included is the union of DEGs detected (FDR < `r fdr`).", fig.asp = 1.8}
o <- order(summed_filt$label, summed_filt$genotype, summed_filt$sample)

all_features <- lapply(
  de_results, 
  function(x) rownames(x[which(x$FDR < fdr), ]))
features <- unique(unlist(all_features))
mat <- logcounts(summed_filt)[features, ]
mat <- mat - rowMeans(mat)
pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_col = data.frame(
    genotype = summed_filt$genotype[o],
    plate_number = summed_filt$plate_number[o],
    sex = summed_filt$sex[o],
    row.names = colnames(summed_filt)[o]),
  annotation_colors = list(
    genotype = genotype_colours,
    plate_number = plate_number_colours,
    sex = sex_colours),
  breaks = seq(-max(abs(mat)), max(abs(mat)), length.out = 101),
  fontsize = 6)
```

The following table gives the number of DEGs at a FDR of `r scales::percent(fdr)`.

```{r}
is_de <- decideTestsPerLabel(de_results, threshold = fdr)
summarizeTestsPerLabel(is_de) %>%
  knitr::kable(
    caption = "Number of DEGs per label. Each row corresponds to a label and each column corresponds to the number of downregulated genes (`-1`), the number of non-differentially expressed genes (`0`), the number of upregulated genes (`1`), and the number of genes not tested (`NA`).")
```

```{r, message = FALSE}
# Create outputs
createDEGOutputs(outdir = deg_dir, de_results = de_results)
```

Please see [`output/DEGs/`](../output/DEGs/) for spreadsheets and interactive `r BiocStyle::Biocpkg("Glimma")` plots of these DEG lists.

# Comments on interpretation

## DA or DE? Two sides of the same coin

While useful, the distinction between DA and DE analyses is inherently artificial for scRNA-seq data.
This is because the labels used in the former are defined based on the genes to be tested in the latter.
To illustrate, consider a scRNA-seq experiment involving two biological conditions with several shared cell types.
We focus on a cell type $X$ that is present in both conditions but contains some DEGs between conditions.
This leads to two possible outcomes:

1. The DE between conditions causes $X$ to form two separate clusters (say, $X_1$ and $X_2$ in expression space. This manifests as DA where $X_1$ is enriched in one condition and $X_2$ is enriched in the other condition.
2. The DE between conditions is not sufficient to split $X$ into two separate clusters, e.g., because the data integration procedure identifies them as corresponding cell types and merges them together. This means that the differences between conditions manifest as DE within the single cluster corresponding to $X$.

We have described the example above in terms of clustering, but the same arguments apply for any labelling strategy based on the expression profiles, e.g., automated cell type assignment.
Moreover, the choice between outcomes 1 and 2 is made implicitly by the combined effect of the data merging, clustering and label assignment procedures.
For example, differences between conditions are more likely to manifest as DE for coarser clusters and as DA for finer clusters, but this is difficult to predict reliably.

The moral of the story is that DA and DE analyses are simply two different perspectives on the same phenomena.
For any comprehensive characterization of differences between populations, it is usually necessary to consider both analyses.
Indeed, they complement each other almost by definition, e.g., clustering parameters that reduce DE will increase DA and vice versa.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```
