---
title: "Selection of biologically relevant cells for the Grant (C075) retinal epithelial cells data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: Single Cell Open Research Endeavour (SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(SingleCellExperiment)
library(here)
library(scater)
library(scran)
library(ggplot2)
library(cowplot)
library(edgeR)
library(Glimma)
library(BiocParallel)
library(patchwork)
library(pheatmap)

source(here("code", "helper_functions.R"))

# NOTE: Using multiple cores seizes up my laptop. Can use more on unix box.
options("mc.cores" = ifelse(Sys.info()[["nodename"]] == "PC1331", 2L, 8L))
register(MulticoreParam(workers = getOption("mc.cores")))

knitr::opts_chunk$set(fig.path = "C075_Grant_Coultas.cell_selection_files/")
```

# Motivation

scRNA-seq datasets may include cells that are not relevant to the study, even after the initial quality control, which we don't want to include in downstream analyses.
In this section aim to filter out these 'unwanted' cells and retain only the 'biologically relevant' cells.
Examples of unwanted cells include:

- Cells with 'reasonable' QC metrics, but that are transcriptomically distinct from the majority of cells in the dataset
- Cells of unwanted cell types, such as those that might sneak through a FACS or magnetic bead enrichment sample preparation

Once we are confident that we have selected the biologically relevant cells, we will perform data integration (if necessary) and a further round of clustering in preparation for downstream analysis.

The removal of unwanted cells is an iterative process where at each step we:

1. Identify cluster(s) enriched for unwanted cells. The exact criteria used to define 'unwanted' will depend on the type of cells we are trying to identify at each step.
    a. Perform diagnostic checks to ensure we aren't discarding biologically relevant cells.
    b. Remove the unwanted cells.
2. Re-process the remaining cells.
    a. Identify HVGs.
    b. Perform dimensionality reduction (PCA and UMAP).
    c. Cluster cells.
    
<aside>
Throughout, we will exclude mitochondrial genes, ribosomal protein genes, sex chromosome genes, and pseudogenes from the HVGs.
</aside>

Clustering is a critical component of this process, so we discuss it in further detail in the next subsection.

## Clustering

Clustering is an unsupervised learning procedure that is used in scRNA-seq data analysis to empirically define groups of cells with similar expression profiles.
Its primary purpose is to summarize the data in a digestible format for human interpretation. 
This allows us to describe population heterogeneity in terms of discrete labels that are easily understood, rather than attempting to comprehend the high-dimensional manifold on which the cells truly reside.
Clustering is thus a critical step for extracting biological insights from scRNA-seq data.

Clustering calculations are usually performed using the top PCs to take advantage of data compression and denoising^[But see the 'Data integration' section of this report for an exception to the rule.].

### Clusters vs. cell types

It is worth stressing the distinction between clusters and cell types.
The former is an empirical construct while the latter is a biological truth (albeit a vaguely defined one).
For this reason, questions like "what is the true number of clusters?" are usually meaningless.
We can define as many clusters as we like, with whatever algorithm we like - each clustering will represent its own partitioning of the high-dimensional expression space, and is as "real" as any other clustering.

A more relevant question is "how well do the clusters approximate the cell types?"
Unfortunately, this is difficult to answer given the context-dependent interpretation of biological truth.
Some analysts will be satisfied with resolution of the major cell types; other analysts may want resolution of subtypes; and others still may require resolution of different states (e.g., metabolic activity, stress) within those subtypes.
Two clusterings can also be highly inconsistent yet both valid, simply partitioning the cells based on different aspects of biology.
Indeed, asking for an unqualified "best" clustering is akin to asking for the best magnification on a microscope without any context.

It is helpful to realize that clustering, like a microscope, is simply a tool to explore the data.
We can zoom in and out by changing the resolution of the clustering parameters, and we can experiment with different clustering algorithms to obtain alternative perspectives of the data.
This iterative approach is entirely permissible for data exploration, which constitutes the majority of all scRNA-seq data analysis.

## Graph-based clustering

We build a shared nearest neighbour graph [@xu2015identification] and use the Louvain algorithm to identify clusters.
We would build the graph using the principal components.

# Preparing the data

We start from the preprocessed *SingleCellExperiment* object created in ['Preprocessing the Grant (C075) retinal epithelial cells data set'](C075_Grant_Coultas.preprocess.html).

```{r}
sce <- readRDS(here("data", "SCEs", "C075_Grant_Coultas.preprocessed.SCE.rds"))

umap_df <- makePerCellDF(sce)

# Some useful colours
plate_number_colours <- setNames(
  unique(sce$plate_number_colours),
  levels(sce$plate_number))
sample_type_colours <- setNames(
  unique(sce$sample_type_colours),
  levels(sce$sample_type))
genotype_colours <- setNames(
  unique(sce$genotype_colours),
  levels(sce$genotype))
mouse_colours <- setNames(
  unique(sce$mouse_colours),
  levels(sce$mouse))
sex_colours <- setNames(
  unique(sce$sex_colours),
  levels(sce$sex))
sequencing_run_colours <- setNames(
  unique(sce$sequencing_run_colours),
  levels(sce$sequencing_run))

# Some useful gene sets
mito_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME == "MT")]

ribo_set <- grep("^Rp(s|l)", rownames(sce), value = TRUE)
# NOTE: A more curated approach for identifying ribosomal protein genes 
#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)
library(msigdbr)
c2_sets <- msigdbr(species = "Mus musculus", category = "C2")
ribo_set <- union(
  ribo_set,
  c2_sets[c2_sets$gs_name == "KEGG_RIBOSOME", ]$gene_symbol)

sex_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME %in% c("X", "Y"))]

pseudogene_set <- rownames(sce)[
  any(grepl("pseudogene", rowData(sce)$ENSEMBL.GENEBIOTYPE))]
```

## Initial clustering

```{r}
set.seed(4759)
snn_gr <- buildSNNGraph(sce, use.dimred = "PCA")
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster <- factor(clusters$membership)
stopifnot(nlevels(sce$cluster) == 8)
cluster_colours <- setNames(
  palette("Okabe-Ito"),
  levels(sce$cluster))
```

There are `r nlevels(sce$cluster)` clusters detected, shown on the UMAP plot Figure \@ref(fig:clusterplot-umap) and broken down by experimental factors in Figure \@ref(fig:cluster-barplot).

```{r clusterplot-umap, fig.cap = "UMAP plot, where each point represents a droplet and is coloured according to the legend.", fig.asp = 0.7}
p1 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = cluster), size = 0.25) +
  scale_colour_manual(values = cluster_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p2 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = genotype), size = 0.25) +
  scale_colour_manual(values = genotype_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p3 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = sex), size = 0.25) +
  scale_colour_manual(values = sex_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p4 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = plate_number), size = 0.25) +
  scale_colour_manual(values = plate_number_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")

(p1 | p2) / (p3 | p4)
```

```{r cluster-barplot, fig.asp = 1 / 2, fig.cap = "Breakdown of clusters by experimental factors."}
p1 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = genotype),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = sex),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sex_colours) +
  theme_cowplot(font_size = 8)
p3 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = plate_number),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = plate_number_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggplot(as.data.frame(colData(sce)[, "cluster", drop = FALSE])) + 
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of droplets") +
  scale_fill_manual(values = cluster_colours) +
  theme_cowplot(font_size = 8) + 
  guides(fill = FALSE)

(p1 | p2) / (p3 | p4)
```

Notably:

- Some of the clusters are highly plate-specific.

# Plate-specific clusters

## Motivation

We have seen that cells from plates `LC294` and `LC358` cluster separately from the remaining plates.
To try to understand what is driving the differences between cells on these two sets of plates, we perform a marker gene analysis.
The results of this analysis will help us decide how to handle this issue.

## Analysis

We use two sets of marker genes for this analysis:

1. [Pre-defined marker genes]
2. [*De novo* marker genes]

```{r}
sce$cluster_batches <- factor(
  ifelse(
    sce$plate_number %in% c("LC294", "LC358"),
    "LC294 and LC358", 
    "rest"))
```

### Pre-defined marker genes

Zoe sent me a list of marker genes to help characterise the cells^[Email 2020-05-06].

```{r}
zoes_markers <- readxl::read_excel(
  here(
    "data", 
    "marker_genes",
    "Genes for violin plots Hbo1 retina ECs.xlsx")) %>%
  # NOTE: Zoe forgot to label the cell cycle genes
  dplyr::mutate(
    `Endothelial identity` = ifelse(
      is.na(`Endothelial identity`),
      "Cell cycle",
      `Endothelial identity`),
    Expressed = Gene %in% rownames(sce))
DT::datatable(
  zoes_markers,
  caption = "Zoe's supplied list of arker genes, their 'Endothelial identity', and whether they are expressed in the current dataset.")
```

Figure \@ref(fig:cluster-batches-zoes-markers) is a heatmap of Zoe's marker genes.
There is perhaps some evidence that some of these genes have lower expression in cells from plates `LC294` and `LC358` compared to cells from the rest of the plates, but it's not especially compelling

<aside>
It's certainly not as striking as the result below in [Marker gene discovery].
</aside>

```{r cluster-batches-zoes-markers, fig.wide = TRUE, fig.cap = "Heatmap of row-normalized log-expression values for Zoe's marker genes. Each column is a sample, each row a gene. Samples have been separetely clustered within each `cluster_batches`.", fig.asp = 1.1}
features <- dplyr::filter(zoes_markers, Expressed) %>% 
  dplyr::pull(Gene)
gene_set <- dplyr::filter(zoes_markers, Expressed) %>% 
  dplyr::pull(`Endothelial identity`)

# Samples are hierarchically ordered within each level of `cluster_batches`.
mat <- logcounts(sce)[features, ]
mat1 <- logcounts(sce)[features, sce$cluster_batches == "LC294 and LC358"]
mat1 <- mat1 - rowMeans(mat)
h1 <- hclust(dist(t(mat1)))
o1 <- h1$order
mat2 <- logcounts(sce)[features, sce$cluster_batches == "rest"]
mat2 <- mat2 - rowMeans(mat)
h2 <- hclust(dist(t(mat2)))
o2 <- h2$order

z <- cbind(
  sce[, sce$cluster_batches == "LC294 and LC358"][, o1],
  sce[, sce$cluster_batches == "rest"][, o2])
mat <- logcounts(z)[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-4, 4)
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
pheatmap(
  mat = mat,
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_row = data.frame(
    gene_set = gene_set,
    row.names = features),
  annotation_col = data.frame(
    cluster_batches = z$cluster_batches,
    cluster = z$cluster,
    plate_number = z$plate_number,
    genotype = z$genotype,
    row.names = colnames(z)),
  annotation_colors = list(
    cluster = cluster_colours[levels(z$cluster)],
    plate_number = plate_number_colours[levels(z$plate_number)],
    genotype = genotype_colours[levels(z$genotype)]),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  annotation_names_row = FALSE,
  fontsize_row = 6)
```

### *De novo* marker genes

We use a binomial test to look for genes that differ in the proportion of expressing cells from plates `LC294` and `LC358` compared to cells from the rest of the plates.
The premise is that we want to see if there are genes that were active in one set of plates and silent in another, a potentially very serious difference between cells on the two sets of plates.

<aside>
For the purposes of this section, a cell is considered to express a gene simply if it has non-zero expression for that gene.
</aside>

```{r}
markers <- findMarkers(
  sce,
  groups = sce$cluster_batches,
  direction = "up",
  test.type = "binom", 
  row.data = rowData(sce))
```

```{r}
dir.create(here("output", "marker_genes", "binom"), recursive = TRUE)

write.csv(
  as.data.frame(flattenDF(markers[["LC294 and LC358"]])),
  gzfile(
    here(
      "output", 
      "marker_genes", 
      "binom", 
      "binom_markers.LC294_and_LC358.plate-specific_clusters.csv.gz")),
  row.names = FALSE,
  # NOTE: quote = TRUE needed because some fields contain commas.
  quote = TRUE)

write.csv(
  as.data.frame(flattenDF(markers[["rest"]])),
  gzfile(
    here(
      "output", 
      "marker_genes", 
      "binom", 
      "binom_markers.rest_of_plates.plate-specific_clusters.csv.gz")),
  row.names = FALSE,
    # NOTE: quote = TRUE needed because some fields contain commas.
  quote = TRUE)
```

There are `r sum(markers[["LC294 and LC358"]][["FDR"]] < 0.05)` genes (`r round(100 * sum(markers[["LC294 and LC358"]][["FDR"]] < 0.05) / nrow(sce), 1)`%) more frequently expressed in cells from plates `LC294` and `LC358` and `r sum(markers[["rest"]][["FDR"]] < 0.05)` genes (`r round(100 * sum(markers[["rest"]][["FDR"]] < 0.05) / nrow(sce), 1)`%) more frequently expressed in cells from the rest of the plates (FDR $< 0.05$).

<aside>
The complete gene lists from the differential expression analysis^[Effect sizes for each comparison are reported as log2-fold changes in the proportion of expressing cells in one group over the proportion in another group.] are available in [`output/marker_genes/binom/`](../output/marker_genes/binom/) (see those with files with `plate-specific_clusters` in the filename).
</aside>

As an example of the information available in these gene lists, Figure \@ref(fig:cluster-batches-markers) highlights the top-50 genes for each group of plates.
What is remarkable about this figure is that it shows that there is a set of genes that are only detected in cells from plates `LC294` and `LC358` and that these genes are seemingly biologically relevant genes (e.g., *Epcam*, *Cdhr5*, *Cdhr2*).
In contrast, the genes detected more frequently in the remaining plates are at least detected to some degree in plates `LC294` and `LC358`.

```{r cluster-batches-markers, fig.wide = TRUE, fig.cap = "Heatmap of row-normalized log-expression values for selected marker genes between plates `LC294` and `LC358` and the rest of the plates. Each column is a sample, each row a gene.", fig.asp = 1.1}
n <- 50
features <- unlist(
  lapply(markers, function(x) head(rownames(x), n)), use.names = TRUE)

mat <- logcounts(sce)[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-4, 4)
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
o <- order(sce$cluster_batches, sce$cluster, sce$plate_number, sce$genotype)
pheatmap(
  mat = mat[, o],
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_row = data.frame(
    cluster_batches = rep(levels(sce$cluster_batches), each = n),
    row.names = features),
  annotation_col = data.frame(
    cluster_batches = sce$cluster_batches[o],
    cluster = sce$cluster[o],
    plate_number = sce$plate_number[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    plate_number = plate_number_colours[levels(sce$plate_number)],
    genotype = genotype_colours[levels(sce$genotype)]),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  annotation_names_row = FALSE,
  fontsize_row = 6)
```

## Summary

This result was discussed with Zoe^[Meeting between Daniela, Zoe and I on 2020-03-06 and in email on 2020-07-19].
Zoe characterised the genes expressed exclusively by cells on plates LC294 and LC358 as being "abnormal for the cells I collected (i.e. ... epithelial cell markers)" and noted that "there was nothing obviously different about the phenotype or the way that [she] prepared these samples that would explain these differences".

```{r}
keep <- !sce$plate_number %in% c("LC294", "LC358")
```

It was therefore decided to exclude cells from plates on LC294 and LC358 from further analysis.

```{r}
sce <- sce[, keep]
colData(sce) <- droplevels(colData(sce))
```

Figure \@ref(fig:breakdown-by-plate-number) shows that most of the remaining plates have cells from a knockout and a wildtype mouse with each pair of mice on a plate are from the same sibship^[Mouse `#1824` acts as a control sample on two plates.].
The exception is the initial pilot plate, LC279, which has cells from a single control mouse.

```{r breakdown-by-plate-number, fig.asp = 1 / 3, fig.cap = "Breakdown of of the samples by plate."}
plot_grid(
  ggcells(sce) + 
    geom_bar(
      aes(x = plate_number, fill = genotype),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = genotype_colours) +
    theme_cowplot(font_size = 8),
  ggcells(sce) + 
    geom_bar(
      aes(x = plate_number, fill = sex),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sex_colours) +
    theme_cowplot(font_size = 8),
  ggcells(sce) + 
    geom_bar(
      aes(x = plate_number, fill = mouse),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = mouse_colours) +
    theme_cowplot(font_size = 8),
  ggcells(sce) + 
    geom_bar(
      aes(x = plate_number, fill = sequencing_run),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = sequencing_run_colours) +
    theme_cowplot(font_size = 8),
  ggcells(sce) + 
    geom_bar(aes(x = plate_number, fill = plate_number)) +
    coord_flip() +
    ylab("Number of wells") +
    scale_fill_manual(values = plate_number_colours) +
    theme_cowplot(font_size = 8),
  align = "v",
  ncol = 3)
```

## Re-processing

```{r}
var_fit <- modelGeneVarWithSpikes(sce, "ERCC", block = sce$batch)
hvg <- getTopHVGs(var_fit, var.threshold = 0)
is_mito <- hvg %in% mito_set
is_ribo <- hvg %in% ribo_set
is_pseudogene <- hvg %in% pseudogene_set
is_sex <- hvg %in% sex_set
hvg <- hvg[!(is_mito | is_ribo | is_sex | is_pseudogene)]

set.seed(67726)
sce <- denoisePCA(
  sce, 
  var_fit, 
  subset.row = hvg,
  BSPARAM = BiocSingular::IrlbaParam(deferred = TRUE))

set.seed(853)
sce <- runUMAP(sce, dimred = "PCA")
umap_df <- makePerCellDF(sce)

set.seed(4759)
snn_gr <- buildSNNGraph(sce, use.dimred = "PCA")
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster <- factor(clusters$membership)
cluster_colours <- setNames(
  Polychrome::glasbey.colors(nlevels(sce$cluster) + 1)[-1],
  levels(sce$cluster))
sce$cluster_colours <- cluster_colours[sce$cluster]
```

There are `r nlevels(sce$cluster)` clusters detected, shown on the UMAP plot Figure \@ref(fig:no-plate-specific-clusters-clusterplot-umap) and broken down by experimental factors in Figure \@ref(fig:no-plate-specific-clusters-cluster-barplot).

```{r no-plate-specific-clusters-clusterplot-umap, fig.cap = "UMAP plot, where each point represents a droplet and is coloured according to the legend.", fig.asp = 0.7}
p1 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = cluster), size = 0.25) +
  scale_colour_manual(values = cluster_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p2 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = genotype), size = 0.25) +
  scale_colour_manual(values = genotype_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p3 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = sex), size = 0.25) +
  scale_colour_manual(values = sex_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p4 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = plate_number), size = 0.25) +
  scale_colour_manual(values = plate_number_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")

(p1 | p2) / (p3 | p4)
```

```{r no-plate-specific-clusters-cluster-barplot, fig.asp = 1 / 2, fig.cap = "Breakdown of clusters by experimental factors."}
p1 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = genotype),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = sex),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sex_colours) +
  theme_cowplot(font_size = 8)
p3 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = plate_number),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = plate_number_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggplot(as.data.frame(colData(sce)[, "cluster", drop = FALSE])) + 
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of droplets") +
  scale_fill_manual(values = cluster_colours) +
  theme_cowplot(font_size = 8) + 
  guides(fill = FALSE)

(p1 | p2) / (p3 | p4)
```

Notably:

- Still, some of the clusters are highly plate-specific (i.e. potential batch effects). 

We therefore proceed to [Data integration] to remove these plate-specific differences

# Data integration

## Motivation

Large single-cell RNA sequencing (scRNA-seq) projects usually need to generate data across multiple batches due to logistical constraints.
However, the processing of different batches is often subject to uncontrollable differences, e.g., changes in operator, differences in reagent quality.
This results in systematic differences in the observed expression in cells from different batches, which we refer to as "batch effects".
Batch effects are problematic as they can be major drivers of heterogeneity in the data, masking the relevant biological differences and complicating interpretation of the results.

Computational correction of these effects is critical for eliminating batch-to-batch variation, allowing data across multiple batches to be combined for common downstream analysis.
However, existing methods based on linear models [@ritchie2015limma;@leek2012sva] assume that the composition of cell populations are either known or the same across batches.
To overcome these limitations, bespoke methods have been developed for batch correction of single-cell data [@haghverdi2018batch;@butler2018integrating;@lin2019scmerge] that do not require a priori knowledge about the composition of the population.
This allows them to be used in workflows for exploratory analyses of scRNA-seq data where such knowledge is usually unavailable.

We will use the Mutual Nearest Neighbours (MNN) approach of @haghverdi2018batch, as implemented in the `r BiocStyle::Biocpkg("batchelor")` package, to perform data integration.
The MNN approach does not rely on pre-defined or equal population compositions across batches, only requiring that a subset of the population be shared between batches.

### Algorithm overview

Consider a cell $a$ in batch $A$, and identify the cells in batch $B$ that are nearest neighbours to $a$ in the expression space defined by the selected features (genes).
Repeat this for a cell $b$ in batch $B$, identifying its nearest neighbours in  
$A$.
Mutual nearest neighbours are pairs of cells from different batches that belong in each other’s set of nearest neighbours.
The reasoning is that MNN pairs represent cells from the same biological state prior to the application of a batch effect^[See @haghverdi2018batch for full theoretical details.].
Thus, the difference between cells in MNN pairs can be used as an estimate of the batch effect, the subtraction of which yields batch-corrected values.

Compared to linear regression, MNN correction does not assume that the population composition is the same or known beforehand.
This is because it learns the shared population structure via identification of MNN pairs and uses this information to obtain an appropriate estimate of the batch effect.
Instead, the key assumption of MNN-based approaches is that the batch effect is orthogonal to the biology in high-dimensional expression space.
Violations reduce the effectiveness and accuracy of the correction, with the most common case arising from variations in the direction of the batch effect between clusters^[Nonetheless, the assumption is usually reasonable as a random vector is very likely to be orthogonal in high-dimensional space.].

## Analysis

We treat each plate as a batch and let the MNN algorithm select the 'best' order for merging the plates.

```{r, results = "hide"}
library(batchelor)

mnn_out <- fastMNN(
  multiBatchNorm(sce, batch = factor(sce$plate_number)),
  batch = factor(sce$plate_number),
  cos.norm = FALSE,
  d = ncol(reducedDim(sce, "PCA")),
  auto.merge = TRUE,
  # NOTE: Found that auto-merging resulted in less lost variation than when 
  #       using the specified merge. Furthermore, the auto-merge order is very 
  #       similar to the manually specified order.
  # merge.order = list(
  #   list("LC392", "LC280"),
  #   list("LC396", "LC398"),
  #   list("LC279")),
  subset.row = hvg,
  # NOTE: Normally use the default of `correct.all = FALSE`.
  correct.all = TRUE)

reducedDim(sce, "corrected") <- reducedDim(mnn_out, "corrected")
# NOTE: Don't normally keep the 'reconstructed' data, but it might be useful 
#       for visualisation in this project.
assay(sce, "reconstructed") <- assay(mnn_out, "reconstructed")
```

One useful diagnostic of the MNN algorithm is the proportion of variance within each batch that is lost during MNN correction^[Specifically, this refers to the within-batch variance that is removed during orthogonalization with respect to the average correction vector at each merge step.].
Large proportions of lost variance (>10%) suggest that correction is removing genuine biological heterogeneity.
This would occur due to violations of the assumption of orthogonality between the batch effect and the biological subspace [@haghverdi2018batch].
In this case, the proportion of lost variance is small, indicating that non-orthogonality is not a major concern.

```{r}
var_loss <- metadata(mnn_out)$merge.info$lost.var
rownames(var_loss) <- sprintf("merge_%d", seq_len(nrow(var_loss)))
knitr::kable(
  100 * var_loss,
  digits = 1,
  caption = "Percentage of estimated biological variation lost within each plate at each step of the auto merge. Ideally, all these values should be small (e.g., < 10%).")
```

### Summary

The MNN algorithm returns two sets of corrected values for use in downstream analyses like clustering or visualization:

1. `corrected`: 'Batch-corrected principal components'
2. `reconstructed`: 'Batch-correcteed expression values'

The `corrected` data contains the low-dimensional corrected coordinates for all cells, which we will use in place of the PCs in our downstream analyses.
The `reconstructed` data contains the corrected expression values for each gene in each cell, obtained by projecting the low-dimensional coordinates in corrected back into gene expression space.
**We do not recommend using the `reconstructed` data for anything other than visualization^[See [here](https://osca.bioconductor.org/integrating-datasets.html#using-corrected-values) for a detailed explanation of this recommendation.].

## Re-processing

We will use the MNN-corrected values in place of the PCs in our downstream analyses, such as UMAP and clustering^[At this point, it is also tempting to use the corrected expression values for gene-based analyses like DE-based marker gene detection. This is not generally recommended as an arbitrary correction algorithm is not obliged to preserve the magnitude (or even direction) of differences in per-gene expression when attempting to align multiple batches.].

```{r}
set.seed(853)
sce <- runUMAP(sce, dimred = "corrected")
umap_df <- makePerCellDF(sce)

set.seed(4759)
snn_gr <- buildSNNGraph(sce, use.dimred = "corrected")
clusters <- igraph::cluster_louvain(snn_gr)
sce$cluster <- factor(clusters$membership)
stopifnot(nlevels(sce$cluster) == 7)
cluster_colours <- setNames(
  palette("Okabe-Ito"),
  levels(sce$cluster))
```

There are `r nlevels(sce$cluster)` clusters detected, shown on the UMAP plot Figure \@ref(fig:mnn-corrected-clusterplot-umap) and broken down by experimental factors in Figure \@ref(fig:mnn-corrected-cluster-barplot).

```{r mnn-corrected-clusterplot-umap, fig.cap = "UMAP plot, where each point represents a droplet and is coloured according to the legend.", fig.asp = 0.7}
p1 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = cluster), size = 0.25) +
  scale_colour_manual(values = cluster_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p2 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = genotype), size = 0.25) +
  scale_colour_manual(values = genotype_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p3 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = sex), size = 0.25) +
  scale_colour_manual(values = sex_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p4 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = plate_number), size = 0.25) +
  scale_colour_manual(values = plate_number_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")

(p1 | p2) / (p3 | p4)
```

```{r mnn-corrected-cluster-barplot, fig.asp = 1 / 2, fig.cap = "Breakdown of clusters by experimental factors."}
p1 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = genotype),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = sex),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sex_colours) +
  theme_cowplot(font_size = 8)
p3 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = plate_number),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = plate_number_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggplot(as.data.frame(colData(sce)[, "cluster", drop = FALSE])) + 
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of droplets") +
  scale_fill_manual(values = cluster_colours) +
  theme_cowplot(font_size = 8) + 
  guides(fill = FALSE)

(p1 | p2) / (p3 | p4)
```

Notably:

- **Now, none of the clusters are highly plate-specific.**
- Now, there are clusters that are somewhat genotype-specific.

# Concluding remarks

```{r}
saveRDS(
  sce,
  here("data", "SCEs", "C075_Grant_Coultas.cells_selected.SCE.rds"),
  compress = "xz")
```
The processed *SingleCellExperiment* object is available (see [`data/SCEs/C075_Grant_Coultas.cells_selected.SCE.rds`](../data/SCEs/C075_Grant_Coultas.cells_selected.SCE.rds)).
This will be used in downstream analyses, e.g., identifying cluster marker genes and refining the cell labels.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```

</details>
