---
title: "Annotating the Grant (C075) retinal epithelial cells data set"
description: |
author:
  - name: Peter Hickey 
    url: https://peterhickey.org
    affiliation: Single Cell Open Research Endeavour (SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/shalin-naik/3310/score
date: "`r Sys.Date()`"
output: distill::distill_article
editor_options: 
  chunk_output_type: console
bibliography: ref.bib
---

```{r setup}
library(SingleCellExperiment)
library(here)
library(scater)
library(scran)
library(ggplot2)
library(cowplot)
library(edgeR)
library(Glimma)
library(BiocParallel)
library(patchwork)
library(pheatmap)

source(here("code", "helper_functions.R"))

# NOTE: Using multiple cores seizes up my laptop. Can use more on unix box.
options("mc.cores" = ifelse(Sys.info()[["nodename"]] == "PC1331", 2L, 8L))
register(MulticoreParam(workers = getOption("mc.cores")))

knitr::opts_chunk$set(fig.path = "C075_Grant_Coultas.annotate_files/")
```

# Preparing the data

We start from the preprocessed *SingleCellExperiment* object created in ['Selection of biologically relevant cells for the Grant (C075) retinal epithelial cells data set'](C075_Grant_Coultas.cell_selection.html).

```{r}
sce <- readRDS(here("data", "SCEs", "C075_Grant_Coultas.cells_selected.SCE.rds"))

# Some useful colours
plate_number_colours <- setNames(
  unique(sce$plate_number_colours),
  unique(names(sce$plate_number_colours)))
sample_type_colours <- setNames(
  unique(sce$sample_type_colours),
  unique(names(sce$sample_type_colours)))
genotype_colours <- setNames(
  unique(sce$genotype_colours),
  unique(names(sce$genotype_colours)))
mouse_colours <- setNames(
  unique(sce$mouse_colours),
  unique(names(sce$mouse_colours)))
sex_colours <- setNames(
  unique(sce$sex_colours),
  unique(names(sce$sex_colours)))
sequencing_run_colours <- setNames(
  unique(sce$sequencing_run_colours),
  unique(names(sce$sequencing_run_colours)))
cluster_colours <- setNames(
  unique(sce$cluster_colours),
  unique(names(sce$cluster_colours)))

# Some useful gene sets
mito_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME == "MT")]

ribo_set <- grep("^Rp(s|l)", rownames(sce), value = TRUE)
# NOTE: A more curated approach for identifying ribosomal protein genes 
#       (https://github.com/Bioconductor/OrchestratingSingleCellAnalysis-base/blob/ae201bf26e3e4fa82d9165d8abf4f4dc4b8e5a68/feature-selection.Rmd#L376-L380)
library(msigdbr)
c2_sets <- msigdbr(species = "Mus musculus", category = "C2")
ribo_set <- union(
  ribo_set,
  c2_sets[c2_sets$gs_name == "KEGG_RIBOSOME", ]$gene_symbol)

sex_set <- rownames(sce)[any(rowData(sce)$ENSEMBL.SEQNAME %in% c("X", "Y"))]

pseudogene_set <- rownames(sce)[
  any(grepl("pseudogene", rowData(sce)$ENSEMBL.GENEBIOTYPE))]
```

# Motivation

The most challenging task in scRNA-seq data analysis is arguably the annotation of cells with meaningful labels.
We might attempt to label cells individually or to label clusters (with cells assigned to those clusters 'inheriting' the cluster-level label).

In ['Selection of biologically relevant cells for the Grant (C075) retinal epithelial cells data set'](C075_Grant_Coultas.cell_selection.html), we obtained one set of labels:

1. 'cluster' labels using clustering. Obtaining clusters of cells is fairly straightforward, but it is more difficult to determine what biological state is represented by each of those clusters.

In this section we use [Pre-defined marker genes] and perform [Cluster marker gene detection] to construct *de novo* marker gene lists that can be used to aid the interpretation of the clustering results.
The *de novo* marker genes lists use a data driven approach to identify those genes that drive separation between clusters whereas the pre-defined marker genes are a more targeted approach to cluster annotation.
These two approaches complement one another.

As a reminder, Figure \@ref(fig:clusterplot-umap) shows the clustering results overlaid on the UMAP plot and Figure \@ref(fig:cluster-barplot) breaks down the clusters by key experimental factors.

```{r clusterplot-umap, fig.cap = "UMAP plot, where each point represents a cell and is coloured according to the legend.", fig.asp = 0.7}
p1 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = cluster), size = 0.25) +
  scale_colour_manual(values = cluster_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p2 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = genotype), size = 0.25) +
  scale_colour_manual(values = genotype_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p3 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = sex), size = 0.25) +
  scale_colour_manual(values = sex_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")
p4 <- ggcells(sce, aes(x = UMAP.1, y = UMAP.2)) +
  geom_point(aes(colour = plate_number), size = 0.25) +
  scale_colour_manual(values = plate_number_colours) + 
  theme_cowplot(font_size = 8) + 
  xlab("Dimension 1") + 
  ylab("Dimension 2")

(p1 | p2) / (p3 | p4)
```

```{r cluster-barplot, fig.asp = 1 / 2, fig.cap = "Breakdown of clusters by experimental factors."}
p1 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = genotype),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = genotype_colours) +
  theme_cowplot(font_size = 8)
p2 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = sex),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = sex_colours) +
  theme_cowplot(font_size = 8)
p3 <- ggcells(sce) + 
  geom_bar(
    aes(x = cluster, fill = plate_number),
    position = position_fill(reverse = TRUE)) +
  coord_flip() +
  ylab("Frequency") +
  scale_fill_manual(values = plate_number_colours) +
  theme_cowplot(font_size = 8)
p4 <- ggcells(sce) +
  geom_bar(aes(x = cluster, fill = cluster)) +
  coord_flip() +
  ylab("Number of cells") +
  scale_fill_manual(values = cluster_colours) +
  theme_cowplot(font_size = 8) + 
  guides(fill = FALSE)

(p1 | p2) / (p3 | p4)
```

We conclude by performing [Cell cycle assignment] and by describing the choice of [Visualization] options for the plots used in this report, as well instructions for how to make your own plots.

# Pre-defined marker genes

We examine [Zoe's gene list] and also perform a simple visualization of the [FACS markers].

## Zoe's gene list

### Motivation

Zoe sent me a list of marker genes to help characterise the cells^[Email 2020-05-06.].

```{r}
zoes_markers <- readxl::read_excel(
  here(
    "data", 
    "zoes_marker_genes",
    "Genes for violin plots Hbo1 retina ECs.xlsx")) %>%
  # NOTE: Zoe forgot to label the cell cycle genes
  dplyr::mutate(
    `Endothelial identity` = ifelse(
      is.na(`Endothelial identity`),
      "Cell cycle",
      `Endothelial identity`),
    Expressed = Gene %in% rownames(sce))

zoes_stem_markers <- readxl::read_excel(
  here(
    "data", 
    "zoes_marker_genes",
    "Stalk endothelial cell genes.xlsx"),
  skip = 2) %>%
  # NOTE: I'm assuming the reference field continues down the rows.
  tidyr::fill(Reference) %>%
  dplyr::mutate(
    `Endothelial identity` = "Stem",
    Expressed = Gene %in% rownames(sce))

zoes_markers <- dplyr::full_join(zoes_markers, zoes_stem_markers) %>%
  # Order `Endothelial identity` as Zoe did in her spreadsheets.
  dplyr::mutate(
    `Endothelial identity` = factor(
      `Endothelial identity`, 
      unique(`Endothelial identity`)))

DT::datatable(
  zoes_markers,
  caption = "Zoe's supplied list of marker genes, their 'Endothelial identity', the reference (if supplied), and whether they are expressed in the current dataset.")

# Exclude non-expressed genes.
zoes_markers <- dplyr::filter(zoes_markers, Expressed)
```

### Analysis

```{r}
zoes_marker_gene_dir <- here("output", "zoes_marker_genes")
dir.create(zoes_marker_gene_dir)
```

```{r, results = "hide"}
violin_plot_dir <- file.path(zoes_marker_gene_dir, "violin_plots")
dir.create(violin_plot_dir)

pdf(
  file.path(violin_plot_dir, "zoes_markers.violin_plot.logcounts.pdf"),
  width = 5,
  height = 5)
lapply(seq_len(nrow(zoes_markers)), function(i) {
  feature <- zoes_markers$Gene[i]
  plotExpression(
    sce, 
    features = feature,
    x = "cluster",
    exprs_values = "logcounts",
    colour_by = "cluster",
    other_fields = "genotype") +
    facet_grid(~ genotype) +
    scale_fill_manual(values = cluster_colours) +
    guides(fill = guide_legend(title = "cluster")) +
    ggtitle(feature)
})
dev.off()

pdf(
  file.path(violin_plot_dir, "zoes_markers.violin_plot.reconstructed.pdf"),
  width = 5,
  height = 5)
lapply(seq_len(nrow(zoes_markers)), function(i) {
  message(i)
  feature <- zoes_markers$Gene[i]
  plotExpression(
    sce, 
    features = feature,
    x = "cluster",
    exprs_values = "reconstructed",
    colour_by = "cluster",
    other_fields = c("genotype")) + 
    facet_grid(~ genotype) + 
    scale_fill_manual(values = cluster_colours) +
    guides(fill = guide_legend(title = "cluster")) +
    ggtitle(feature)
})
dev.off()
```

Figure \@ref(fig:zoes-markers-heatmap) is a heatmap of Zoe's marker genes.
It is challenging to immediately associate clusters with a cell type on the basis of these marker genes, but some patterns are clear, such as:

- `Artery` marker genes are highly expressed in cluster 7.
- `Artery, tip` marker genes are highly expressed in clusters 5 and 7.
- `Tip` marker genes are highly expressed in cluster 5.

The expressions patterns for the other marker gene sets are more complex, such as a subset of the gene set being not expressed, restricted to particular cluster(s), or ubiquitously expressed across clusters.

<aside>
Violin plots of per-gene expression measurements and heatmaps (`logcounts` and `reconstructed`) can be found in the [`output/zoes_marker_genes/`](../output/zoes_marker_genes/) directory.
</aside>

```{r, results = "hide"}
features <- dplyr::filter(zoes_markers, Expressed) %>% 
  dplyr::pull(Gene)
gene_set <- dplyr::filter(zoes_markers, Expressed) %>% 
  dplyr::pull(`Endothelial identity`)

o <- order(sce$cluster, sce$genotype)

mat <- assay(sce, "logcounts")[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
p1 <- pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_row = data.frame(
    gene_set = gene_set,
    row.names = features),
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    genotype = genotype_colours),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  silent = TRUE,
  main = "logcounts (row-normalized)",
  fontsize = 6)

mat <- assay(sce, "reconstructed")[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
p2 <- pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_row = data.frame(
    gene_set = gene_set,
    row.names = features),
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    genotype = genotype_colours),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  silent = TRUE,
  main = "reconstructed (row-normalized)",
  fontsize = 6)
```

```{r zoes-markers-heatmap, fig.wide = TRUE, fig.cap = "Heatmap of row-normalized `logcounts` for Zoe's marker genes. Each column is a sample, each row a gene. Samples are ordered by `cluster` then by `genotype`.", fig.asp = 1.1, results = "hide"}
p1
```

```{r, results = "hide"}
heatmap_dir <- file.path(zoes_marker_gene_dir, "heatmaps")
dir.create(heatmap_dir)

pdf(
  file.path(heatmap_dir, "zoes_markers.heatmap.logcounts.pdf"),
  width = 5,
  height = 5)
p1
dev.off()

pdf(
  file.path(heatmap_dir, "zoes_markers.heatmap.reconstructed.pdf"),
  width = 5,
  height = 5)
p2
dev.off()
```

## FACS markers

```{r}
facs_markers <- grep(
  "^B|^V|^R|^Y|^FSC|^SSC",
  colnames(colData(sce)),
  value = TRUE)
```

Several FACS markers were collected for these cells: `r glue::glue_collapse(facs_markers, sep = ", ", last = ", and ")`.

Absolute FACS measurements are generally not comparable across samples for a number of reasons^[E.g., different machines and reagents] and normalization methods are underdeveloped.
We apply a simple 'scaled rank' normalization method, which is to rank within each plate the FACS measurements between 0 and 1 (0 being the lowest and 1 being the highest).
This is a very drastic normalization that removes much of the quantitative information potentially available in the FACS data.

<aside>
Any sort of sophisticated analysis of the FACS data is very challenging because of this lack of standardization in data collection. The normalization of this data and integration with scRNA-seq data is a research question that is beyond the scope of this analysis.
</aside>

Figure \@ref(fig:umap-with-facs) presents a simple summary of the FACS data by overlaying it on the UMAP plot. 

```{r umap-with-facs, fig.asp = 3, fig.cap = "Overlay of index sorting data on UMAP plot. For each marker, the left-hand plot shows the 'raw' or 'pseudo-logged' fluorescence intensity and the right-side plots the 'scaled rank' of the raw intensity. The pseudo-log transformation is a transformation mapping numbers to a signed logarithmic scale with a smooth transition to linear scale around 0. This transformation is commonly used when plotting fluorescence intensities from FACS. The scaled rank is applied within each `plate_number` and assigns the maximum fluorescence intensity a value of one and the minimum fluorescence intensities a value of zero. It can be thought of as a crude normalization of the FACS data that allows us to compare fluorescence intensities from different plates."}
p <- lapply(facs_markers, function(m) {
  new_umap <- makePerCellDF(sce)
  new_umap <- new_umap[!is.na(dplyr::select(new_umap, all_of(m))), ]
  p1 <- ggplot(new_umap, aes(x = UMAP.1, y = UMAP.2)) +
    geom_point(aes_string(colour = m, fill = m), alpha = 1, size = 0.2) +
    scale_fill_viridis_c(trans = ifelse(
      grepl("^FSC|^SSC", m), 
      "identity",
      scales::pseudo_log_trans(sigma = 150 / 2)),
      breaks = c(0, 10, 100, 1000, 10000, 100000),
      name = "") +
    scale_colour_viridis_c(trans = ifelse(
      grepl("^FSC|^SSC", m), 
      "identity",
      scales::pseudo_log_trans(sigma = 150 / 2)),
      breaks = c(0, 10, 100, 1000, 10000, 100000),
      name = "") +
    theme_cowplot(font_size = 8) + 
    xlab("Dimension 1") + 
    ylab("Dimension 2") + 
    ggtitle(ifelse(grepl("^FSC|^SSC", m), "Raw", "Pseudo-logged"), m) + 
    facet_grid(. ~ genotype)
  
  new_umap$scaled_rank <- as.data.frame(colData(sce[, !is.na(sce[[m]])])) %>%
    dplyr::group_by(plate_number) %>%
    dplyr::mutate(
      rank = rank(!!sym(m), na.last = "keep"),
      max_rank = max(rank, na.rm = TRUE),
      scaled_rank = rank / max_rank) %>%
    dplyr::pull(scaled_rank)
  
  p2 <- ggplot(data = new_umap, aes(x = UMAP.1, y = UMAP.2)) +
    geom_point(aes(colour = scaled_rank, fill = scaled_rank), alpha = 1, size = 0.2) +
    scale_fill_viridis_c(name = "") +
    scale_colour_viridis_c(name = "") +
    theme_cowplot(font_size = 8) + 
    xlab("Dimension 1") + 
    ylab("Dimension 2") + 
    ggtitle("Scaled rank", m) +
    facet_grid(. ~ genotype)
  list(p1, p2)
})
multiplot(
  plotlist = unlist(p, recursive = FALSE),
  layout = matrix(seq_len(length(p) * 2), ncol = 2, byrow = TRUE))
```



# Cluster marker gene detection

## Motivation

To interpret our clustering results, we identify the genes that drive separation between clusters.
These marker genes allow us to assign biological meaning to each cluster based on their functional annotation.
In the most obvious case, the marker genes for each cluster are *a priori* associated with particular cell types, allowing us to treat the clustering as a proxy for cell type identity.
The same principle can be applied to more subtle differences in activation status or differentiation state.

Identification of marker genes is usually based around the retrospective detection of differential expression between clusters^[Differential expression analyses is performed on the original log-expression values.].
Genes that are more strongly DE are more likely to have driven cluster separation in the first place.
The top DE genes are likely to be good candidate markers as they can effectively distinguish between cells in different clusters.

## Statistical methodology and considerations

It is important to have some understanding of the statistical methodology used for marker gene detection.
This section gives an overview.

### Gene lists

For each cluster, the DE results of the relevant comparisons are consolidated into a single output table.
This allows a set of marker genes to be easily defined by taking the top DE genes from each pairwise comparison between clusters.
Other statistics are also reported for each gene, including the adjusted p-values^[It must be stressed that the (adjusted) p-values computed here cannot be properly interpreted as measures of significance because the clusters have been empirically identified from the data. Nonetheless, they can still be used as way to rank 'interesting' genes.] and the log-fold changes relative to every other cluster.

### Use of pairwise comparisons

We intentionally use pairwise comparisons between clusters rather than comparing each cluster to the average of all other cells.
The latter approach is sensitive to the population composition, potentially resulting in wildly different sets of markers when cell type abundances change in different contexts.
In the worst case, the presence of a single dominant subpopulation will drive the selection of top markers for every other cluster, pushing out useful genes that can resolve the various minor subpopulations.
Moreover, pairwise comparisons naturally provide more information to interpret of the utility of a marker, e.g., by providing log-fold changes to indicate which clusters are distinguished by this gene.

### Blocking

We perform intra-plate comparisons by blocking on the `plate_number` to avoid confounding effects from differential expression between plates^[Intra-plate comparisons are robust to differences between libraries but assume that each pair of clusters is present in at least one library. When this is not the case, we cannot commute a logFC for that pair of clusters.].

```{r}
sce$block <- sce$plate_number
```

### Choice of test statistic

The Welch $t$-test is an obvious choice of statistical method to test for differences in expression between clusters.
It is quickly computed and has good statistical properties for large numbers of cells [@soneson2018bias].

Alternatively, we could consider the Wilcoxon rank sum test (also known as the Wilcoxon-Mann-Whitney test, or WMW test).
Its strength lies in the fact that it directly assesses separation between the expression distributions of different clusters.
The WMW test statistic is proportional to the area-under-the-curve (AUC), i.e., the concordance probability, which is the probability of a random cell from one cluster having higher expression than a random cell from another cluster.
In a pairwise comparison, AUCs of 1 or 0 indicate that the two clusters have perfectly separated expression distributions. 
Thus, the WMW test directly addresses the most desirable property of a candidate marker gene, while the $t$-test only does so indirectly via the difference in the means and the intra-group variance.

Another alternative, the binomial test, identifies genes that differ in the proportion of expressing cells between clusters^[For the purposes of this test, a cell is considered to express a gene simply if it has one or more counts for that gene.].
This represents a much more stringent definition of marker genes compared to the other methods, as differences in expression between clusters are effectively ignored if both distributions of expression values are not near zero.
The premise is that genes are more likely to contribute to important biological decisions if they were active in one cluster and silent in another, compared to more subtle 'tuning' effects from changing the expression of an active gene.
From a practical perspective, a binary measure of presence/absence is easier to validate.

The Welch $t$-test is our default method for identifying marker genes.

### Direction and magnitude of the log-fold change

There are 3 ways of specifying the direction of the log-fold change used in the statistical test when comparing cluster $X$ to cluster $Y$:

1. `direction = "any"`: Identify genes that are upregulated or downregulated in $X$ compared to $Y$.
2. `direction = "up"`: Identify genes that are upregulated in $X$ compared to $Y$.
3. `direction = "down"`: Identify genes that are downregulated in  $X$ compared to $Y$.

Generally speaking, downregulated genes are less appealing as markers as it is more difficult to interpret and experimentally validate an absence of expression.
We therefore mostly focus on identifying genes that are upregulated (i.e. `direction = "up"`) in the chosen cluster relative to any/some/all other clusters .
Of course, this increased stringency is not without cost.
If only upregulated genes are requested then any cluster defined by downregulation of a marker gene will not contain that gene among the top set of features in its gene list.
This is occasionally relevant for subtypes or other states that are distinguished by high versus low expression of particular genes.

### Choice of combining pairwise DE results into a marker list

There are 3 ways of combining the pairwise DE results to obtain the per-cluster marker gene lists:

1. [`pval.type = "any"`: Consolidating with DE against *any* other cluster]
2. [`pval.type = "all"`: Consolidating with DE against *all* other clusters]
3. [`pval.type = "some"`: Consolidating with DE against *some* other clusters]

It is important to understand the differences between these strategies, which are described below in some detail.

#### `pval.type = "any"`: Consolidating with DE against *any* other cluster

If `pval.type = "any"`, the null hypothesis is that the gene is not DE in *any* pairwise comparison.
The genes in each cluster's gene list are sorted by the minimum rank (by significance) across all pairwise comparisons (called the `Top` value).
Taking all rows with `Top` values less than or equal to $T$ yields a marker set containing the top $T$ genes from each pairwise comparison.

This strategy guarantees the inclusion of genes that can distinguish between any two clusters.

To demonstrate, let us define a marker set with a $T$ of 1 for a given cluster. 
The set of genes with `Top`$\leq1$ will contain the top gene from each pairwise comparison to every other cluster.
If $T$ is instead, say, $5$, the set will consist of the union of the top 5 genes from each pairwise comparison.
Obviously, multiple genes can have the same `Top` as different genes may have the same rank across different pairwise comparisons.
Conversely, the marker set may be smaller than the product of `Top` and the number of other clusters, as the same gene may be shared across different comparisons.

This approach does not explicitly favour genes that are uniquely expressed in a cluster.
Rather, it focuses on combinations of genes that - together - drive separation of a cluster from the others.
This is more general and robust but tends to yield a less focused marker set compared to the other `pval.type` settings.

For each gene and cluster, the summary effect size is defined as the effect size from the pairwise comparison with the lowest p-value.
The combined p-value is computed by applying Simes' method to all p-values. 
Neither of these values are directly used for ranking and are only reported for the sake of the user.

#### `pval.type = "all"`: Consolidating with DE against *all* other clusters

If `pval.type = "all"`, the null hypothesis is that the gene is not DE in *all* pairwise comparisons.
A combined p-value for each gene is computed using Berger's intersection union test (IUT).
Ranking based on the IUT p-value will focus on genes that are DE in that cluster compared to all other clusters.

This strategy is particularly effective when dealing with distinct clusters that have a unique expression profile.
In such cases, it yields a highly focused marker set that concisely captures the differences between clusters.

However, it can be too stringent if the cluster's separation is driven by combinations of gene expression. For example, consider a situation involving four clusters expressing each combination of two marker genes A and B.
With `pval.type = "all"`, neither A nor B would be detected as markers as it is not uniquely defined in any one cluster.
This is especially detrimental with overclustering where an otherwise acceptable marker is discarded if it is not DE between two adjacent clusters.

For each gene and cluster, the summary effect size is defined as the effect size from the pairwise comparison with the largest p-value.
This reflects the fact that, with this approach, a gene is only as significant as its weakest DE.
Again, this value is not directly used for ranking and are only reported for the sake of the user.

#### `pval.type = "some"`: Consolidating with DE against *some* other clusters

If `pval.type = "some"`, the null hypothesis is that the gene is not DE in *some* of the pairwise comparisons.
Thus, `pval.type = "some"` serves as a compromise between the `pval.type = "all"` and `pval.type = "any"` strategies.
The definition of 'some' is formalized by the minimum proportion (`min.prop`) of significant comparisons per gene and can be tuned to the specifics of the dataset.

For example, suppose we require that the gene is significant in at least `min.prop = 0.5` (i.e. 50%) of comparisons.
A combined p-value is calculated by taking the middlemost value of the Holm-corrected p-values for each gene.
Here, the null hypothesis is that the gene is not DE in at least half of the contrasts.

Genes are then ranked by the combined p-value.
The aim is to provide a more focused marker set without being overly stringent,
However, a downside is that it loses the theoretical guarantees of the `pval.type = "all"` and `pval.type = "any"` strategies.
For example, there is no guarantee that the top set contains genes that can distinguish a cluster from any other cluster, which would have been possible with `pval.type = "any"`.

For each gene and cluster, the summary effect size is defined as the effect size from the pairwise comparison with the `min.prop`-smallest p-value.
This mirrors the p-value calculation but, again, is reported only for the benefit of the user.

## Analysis

CSV files of the cluster marker gene lists and PDFs files of heatmaps for *selected* genes from these cluster marker gene lists can be found in the [`output/cluster_marker_genes/`](../output/cluster_marker_genes/) directory.

<aside>
See [Visualization] for further details of these heatmaps.
</aside>

```{r}
cluster_marker_gene_dir <- here("output", "cluster_marker_genes")
dir.create(cluster_marker_gene_dir)
```

These files are organised as follows:

| prefix      | `test.type` | `direction` | `pval.type` |
|-------------|-------------|-------------|-------------|
| `t_up_all`  | t           | up          | all         |
| `t_up_some` | t           | up          | some        |
| `t_any_any` | t           | any         | any         |


### `t_up_all`

```{r}
test.type <- "t"
direction <- "up"
pval.type <- "all"
min.prop <- NULL

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce))
t_up_all_markers <- markers
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_all/`](../output/cluster_marker_genes/t_up_all/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- paste(test.type, direction, pval.type, sep = "_")
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir, 
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

### `t_up_some`

```{r}
# Sanity check
stopifnot(nlevels(sce$cluster) == 7)
```

We set `min.prop` = $\frac{2}{n - 1}$ where $n$ is the number of clusters (i.e. `min.prop` = $\frac{2}{7 - 1} = \frac{1}{3}$) so that 'some' means 'two or more pairwise comparisons'. 

```{r}
test.type <- "t"
direction <- "up"
pval.type <- "some"
min.prop <- 2 / (nlevels(sce$cluster) - 1)

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce))
t_up_some_markers <- markers
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_some/`](../output/cluster_marker_genes/t_up_some/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- paste(test.type, direction, pval.type, sep = "_")
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir,
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

### `t_any_any`

```{r}
test.type <- "t"
direction <- "any"
pval.type <- "any"
min.prop <- NULL

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce))
t_any_any_markers <- markers
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_any/`](../output/cluster_marker_genes/t_up_any/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- paste(test.type, direction, pval.type, sep = "_")
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir, 
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

### Targeted

We can also perform targeted comparisons between particular clusters of interest.
For example, this can be useful when two clusters ($A$ and $B$) share a set of marker genes that distinguish $(A, B)$ from the remaining clusters and we now want to hone in on what specifically distinguishes $A$ from $B$.

#### `t_up_all_clusters_01_and_02`

This analysis compares clusters 1 and 2.

```{r}
test.type <- "t"
direction <- "up"
pval.type <- "all"
min.prop <- NULL
restrict <- c("1", "2")

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce),
  restrict = restrict)
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_all_clusters_01_and_02/`](../output/cluster_marker_genes/t_up_all_clusters_01_and_02/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- sprintf(
  "%s_%s_%s_clusters_%02d_and_%02d", 
  test.type, 
  direction, 
  pval.type, 
  as.integer(restrict[1]), 
  as.integer(restrict[2]))
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir, 
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

#### `t_up_all_clusters_03_and_04`

This analysis compares clusters 3 and 4.

```{r}
test.type <- "t"
direction <- "up"
pval.type <- "all"
min.prop <- NULL
restrict <- c("3", "4")

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce),
  restrict = restrict)
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_all_clusters_03_and_04/`](../output/cluster_marker_genes/t_up_all_clusters_03_and_04/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- sprintf(
  "%s_%s_%s_clusters_%02d_and_%02d", 
  test.type, 
  direction, 
  pval.type, 
  as.integer(restrict[1]), 
  as.integer(restrict[2]))
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir, 
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

#### `t_up_all_clusters_03_and_06`

This analysis compares clusters 3 and 6.

```{r}
test.type <- "t"
direction <- "up"
pval.type <- "all"
min.prop <- NULL
restrict <- c("3", "6")

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce),
  restrict = restrict)
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_all_clusters_03_and_06/`](../output/cluster_marker_genes/t_up_all_clusters_03_and_06/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- sprintf(
  "%s_%s_%s_clusters_%02d_and_%02d", 
  test.type, 
  direction, 
  pval.type, 
  as.integer(restrict[1]), 
  as.integer(restrict[2]))
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir, 
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

#### `t_up_all_clusters_04_and_06`

This analysis compares clusters 4 and 6.

```{r}
test.type <- "t"
direction <- "up"
pval.type <- "all"
min.prop <- NULL
restrict <- c("4", "6")

markers <- findMarkers(
  sce,
  groups = sce$cluster,
  test = test.type,
  direction = direction,
  block = sce$block,
  pval.type = pval.type,
  min.prop = min.prop,
  row.data = rowData(sce),
  restrict = restrict)
```

The following table gives the number of marker genes for each cluster by this approach.

<aside>
For spreadsheets and heatmaps of these gene lists see [`output/cluster_marker_genes/t_up_all_clusters_04_and_06/`](../output/cluster_marker_genes/t_up_all_clusters_04_and_06/).
</aside>

```{r}
n <- sapply(markers, function(x) sum(x$FDR < 0.05))
knitr::kable(
  data.frame(
    cluster = names(markers),
    n = sapply(markers, function(x) sum(x$FDR < 0.05))),
  caption = "Number of marker genes for each cluster (FDR < 0.05).")
```

```{r, message = FALSE}
# Create outputs
prefix <- sprintf(
  "%s_%s_%s_clusters_%02d_and_%02d", 
  test.type, 
  direction, 
  pval.type, 
  as.integer(restrict[1]), 
  as.integer(restrict[2]))
marker_dir <- file.path(cluster_marker_gene_dir, prefix)
dir.create(marker_dir)
createClusterMarkerOutputs(
  prefix = prefix, 
  outdir = marker_dir, 
  markers = markers,
  exprs_values = c("logcounts", "reconstructed"), 
  k = 100,
  width = 6,
  height = 7.5)
```

# Cell cycle assignment

On occasion, it can be desirable to determine cell cycle activity from scRNA-seq data.
In and of itself, the distribution of cells across phases of the cell cycle is not usually informative, but we can use this to determine if there are differences in cell cycle progression between subpopulations or across treatment conditions.
Many of the key events in the cell cycle (e.g., passage through checkpoints) are post-translational and thus not directly visible in transcriptomic data; nonetheless, there are enough changes in expression that can be exploited to determine cell cycle phase.

## Using the cyclins

```{r, results = "hide"}
cyclin_genes <- grep("^Ccn[abde][0-9]$", rownames(sce))
cyclin_genes <- rownames(sce)[cyclin_genes]
cyclin_genes
```

The cyclins control progression through the cell cycle and have well-characterized patterns of expression across cell cycle phases.
The cyclins available in this dataset are as follows: `r glue::glue_collapse(cyclin_genes, ", ", last = ", and ")`.

- Cyclin A is expressed across S and G2.
- cyclin B is expressed highest in late G2 and mitosis.
- Cyclin D is expressed throughout but peaks at G1.
- Cyclin E is expressed highest in the G1/S transition.

Inspection of the relative expression of cyclins across the population can often be sufficient to determine the relative cell cycle activity in each cluster.

For example, Figure \@ref(fig:cyclin-heatmap) shows that clusters 1 and 2 have higher expression of cyclins A, B, and E and therefore may be in S,G2 or G2/M phase while the other clusters are scattered across the later phases.

```{r cyclin-heatmap, fig.wide = TRUE, fig.cap = "Heatmap of `logcounts` (left) and row-normalized `logcounts` values (right) for Zoe's marker genes. Each column is a sample, each row a gene. Samples are ordered by `cluster` then by `genotype`. Row (gene) and column (cell) order are preserved across the two heatmaps."}
features <- sort(cyclin_genes)

o <- order(sce$cluster, sce$genotype)

mat <- assay(sce, "logcounts")[features, ]
p1 <- pheatmap(
  mat = mat[, o],
  color = viridisLite::magma(101),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    genotype = genotype_colours),
  silent = TRUE,
  main = "logcounts",
  fontsize = 6)

mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
p2 <- pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    genotype = genotype_colours),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  silent = TRUE,
  main = "logcounts (row-normalized)",
  fontsize = 6)

gridExtra::grid.arrange(p1$gtable, p2$gtable, ncol = 2)
```

We can also use standard differential expression methods to look for upregulation of each cyclin, allowing us to determine if there are more cells in the corresponding phase of the cell cycle between subpopulations.

Here we use pairwise Wilcoxon rank sum tests to compute differential expression of cyclin genes between each pair of clusters.
The Wilcoxon test estimates effect sizes as overlap proportions (`overlap` in the tables below).
The overlap proportion is defined as the probability that a randomly selected cell in $A$ has a greater expression value of $X$ than a randomly selected cell in $B$.
Overlap proportions near 0 ($A$ is lower than $B$) or 1 ($A$ is higher than $B$) indicate that the expression distributions are well-separated.
The Wilcoxon rank sum test effectively tests for significant deviations from an overlap proportion of 0.5.

```{r}
cyclin_markers <- findMarkers(
  sce, 
  subset.row = sort(cyclin_genes), 
  group = sce$cluster, 
  block = sce$block, 
  direction = "up", 
  pval.type = "any",
  test.type = "wilcox")

g2_clusters <- c("1", "2")
```

We can infer that clusters `r glue::glue_collapse(g2_clusters, sep = " and ")` likely have more cells in S, G2, and M phases than the other clusters, based on higher expression of some of the cyclin A and B genes.

```{r}
DT::datatable(
  as.data.frame(cyclin_markers[[g2_clusters[[1]]]]),
  caption = paste0(
    "Cyclin DE analysis comparing cluster ",
    g2_clusters[1],
    " to all other clusters.")) %>%
  DT::formatSignif(
    c("p.value", "FDR", "summary.AUC", paste0("AUC.", 2:7)),
    digits = 2)
```

```{r}
DT::datatable(
  as.data.frame(cyclin_markers[[g2_clusters[[2]]]]),
  caption = paste0(
    "Cyclin DE analysis comparing cluster ",
    g2_clusters[2],
    " to all other clusters.")) %>%
  DT::formatSignif(
    c("p.value", "FDR", "summary.AUC", paste0("AUC.", c(1, 3:7))),
    digits = 2)
```

Direct examination of cyclin expression is easily understood, interpreted and validated with other technologies.
However, it is best suited for statements about relative cell cycle activity; for example, we would find it difficult to assign cell cycle phase in Figure \@ref(fig:cyclin-heatmap) without the presence of clusters spanning all phases to provide benchmarks for "high" and "low" expression of each cyclin.
We also assume that cyclin expression is not affected by biological processes other than the cell cycle, which may be a strong assumption in some cases, e.g., malignant cells.

## Using reference profiles

Cell cycle assignment can be considered a specialized case of cell annotation, which suggests that we can use the `r BiocStyle::Biocpkg("SingleR")` package with an appropriate reference dataset.
For example, given a reference dataset containing mouse ESCs with known cell cycle phases^[Buettner et al. 2015], we could use `r BiocStyle::Biocpkg('SingleR')` to determine the phase of each cell in a test dataset.

We use the reference dataset to identify phase-specific markers from genes with annotated roles in cell cycle.
The use of prior annotation aims to avoid detecting markers for other biological processes that happen to be correlated with the cell cycle in the reference dataset, which would reduce classification performance if those processes are absent or uncorrelated in the test dataset.

```{r}
# Adapted from 
# https://osca.bioconductor.org/cell-cycle-assignment.html#using-reference-profiles
library(scRNAseq)
sce.ref <- BuettnerESCData()
# Find genes that are present in both datasets and are cell cycle-related.
library(org.Mm.eg.db)
cycle.anno <- select(
  org.Mm.eg.db,
  keytype = "GOALL",
  keys = "GO:0007049", 
  columns = "ENSEMBL")[,"ENSEMBL"]
candidates <- Reduce(
  intersect, 
  list(rownames(sce.ref), sub("\\.[0-9]+$", "", rowData(sce)$ENSEMBL.GENEID), cycle.anno))
# Identifying markers between cell cycle phases.
sce.ref <- logNormCounts(sce.ref)
phase.stats <- pairwiseWilcox(
  logcounts(sce.ref), 
  sce.ref$phase, 
  direction = "up",
  subset.row = candidates)
cycle.markers <- getTopMarkers(phase.stats[[1]], phase.stats[[2]])
```

Unlike the cyclin-based analysis, this approach yields "absolute" assignments of cell cycle phase that do not need to be interpreted relative to other cells in the same dataset.

```{r}
library(SingleR)
# Switching row names back to Ensembl to match the reference.
test.data <- logcounts(sce)
rownames(test.data) <- sub("\\.[0-9]+$", "", rowData(sce)$ENSEMBL.GENEID)
assignments <- SingleR(
  test.data, 
  ref = sce.ref,
  label = sce.ref$phase, 
  genes = cycle.markers,
  )
janitor::tabyl(
  data.frame(labels = assignments$labels, cluster = sce$cluster),
  cluster,
  labels) %>%
  knitr::kable(
    caption = 
      "Assignement of cells in each cluster to a cell cycle phase with the 'reference profiles' approach.")
```

The key assumption here is that the cell cycle is orthogonal to cell type and other aspects of cell behaviour.
This justifies the use of a reference involving cell types that are quite different from the cells in the test dataset, provided that the cell cycle transcriptional program is conserved across datasets^[Bertoli, Skotheim, and Bruin 2013; Conboy et al. 2007].
However, it is not difficult to find routine violations of this assumption^[see [https://osca.bioconductor.org/cell-cycle-assignment.html#using-reference-profiles](https://osca.bioconductor.org/cell-cycle-assignment.html#using-reference-profiles) for an illustration].

Thus, a healthy dose of scepticism is required when interpreting these assignments. 
Our hope is that any systematic assignment error is consistent across clusters and conditions such that they cancel out in comparisons of phase frequencies, which is the more interesting analysis anyway.
Indeed, while the availability of absolute phase calls may be more appealing, it may not make much practical difference to the conclusions if the frequencies are ultimately interpreted in a relative sense (e.g., using a chi-squared test).

## Using the `cyclone()` classifier

The prediction method described by [@scialdone2015computational] is another approach for classifying cells into cell cycle phases.
Using a reference dataset, we first compute the sign of the difference in expression between each pair of genes.
Pairs with changes in the sign across cell cycle phases are chosen as markers.
Cells in a test dataset can then be classified into the appropriate phase, based on whether the observed sign for each marker pair is consistent with one phase or another. 
This approach is implemented in the `cyclone()` function from the `R BiocStyle::Biocpkg("scran")` package.

```{r}
mm_pairs <- readRDS(
  system.file("exdata", "mouse_cycle_markers.rds", package = "scran"))
# Using Ensembl IDs to match up with the annotation in `mm_pairs`.
set.seed(100)
assignments <- cyclone(
  sce,
  mm_pairs,
  gene.names = sub("\\.[0-9]+$", "", rowData(sce)$ENSEMBL.GENEID))
sce$cyclone_phases <- assignments$phases
```

The phase assignment result for each cell in the dataset is shown in Figure \@ref(fig:cyclone). 
For each cell, a higher score for a phase corresponds to a higher probability that the cell is in that phase.
We recommend focusing on the G1 and G2/M scores as these are the most informative for classification.

```{r cyclone, fig.cap = "Cell cycle phase scores from applying the pair-based classifier on the dataset. Each point represents a cell, plotted according to its scores for G1 and G2/M phases, and coloured according to cluster.", fig.asp = 1}
plot(
  x = assignments$score$G1, 
  y = assignments$score$G2M,
  xlab = "G1 score",
  ylab = "G2/M score",
  pch = 16,
  col = cluster_colours[sce$cluster])
legend(
  "bottomleft",
  pch = 16,
  col = cluster_colours[levels(sce$cluster)],
  legend = names(cluster_colours[levels(sce$cluster)]),
  title = "Cluster",
  ncol = 2)
```

Cells are classified as being in G1 phase if the G1 score is above 0.5 and greater than the G2/M score; in G2/M phase if the G2/M score is above 0.5 and greater than the G1 score; and in S phase if neither score is above 0.5.

Figure \@ref(fig:clusters-by-cyclone-phase) shows most cells are estimated as being in G1 phase.

```{r clusters-by-cyclone-phase, fig.asp = 2 / 5, fig.cap = "Breakdown of clusters by phase (as predicted using cyclone)."}
phase_colours <- setNames(
  RColorBrewer::brewer.pal(3, "Accent"),
  sort(unique(assignments$phases)))
plot_grid(
  ggplot(data.frame(phase = assignments$phases, cluster = sce$cluster)) +
    geom_bar(
      aes(x = cluster, fill = phase),
      position = position_fill(reverse = TRUE)) + 
    coord_flip() +
    ylab("Frequency") +
    scale_fill_manual(values = phase_colours) +
    theme_cowplot(font_size = 6),
  ggplot(as.data.frame(colData(sce))) + 
    geom_bar(aes(x = cluster, fill = cluster)) +
    coord_flip() +
    ylab("Number of cells") +
    scale_fill_manual(values = cluster_colours) +
    theme_cowplot(font_size = 6),
  align = "h",
  ncol = 2)
```

The data used to generate Figure \@ref(fig:clusters-by-cyclone-phase) are also given in the table below.

```{r}
makePerCellDF(sce) %>%
  janitor::tabyl(cluster, cyclone_phases) %>%
  knitr::kable(
    caption = "Number of cells from each cluster estimated to be in each cell cycle phase using the `cyclone()` classifier.")
```

The same considerations and caveats described for the 'reference profiles' approach are also applicable here.

## Summary

We can conclude that clusters `r glue::glue_collapse(g2_clusters, sep = " and ")` likely have more cells in S, G2, and M phases than the other clusters and that the remaining clusters are likely in G1 (based on fairly ubiquitous expression of cyclin D genes and no strong upregulation of the other cyclins).

For some time, it was popular to regress out the cell cycle phase prior to downstream analyses.
The aim was to remove uninteresting variation due to cell cycle, thus improving resolution of other biological processes of interest.
However, we do not consider adjusting for cell cycle to be a necessary step in routine scRNA-seq analyses.

Any attempt at removal would need to assume that the cell cycle effect is orthogonal to other biological processes.
If this was violated (i.e. the cell cycle effect was associated with other biological processes), then the regression would potentially remove interesting signal if cell cycle activity varied across clusters or conditions.

# Visualization

For each marker gene strategy (e.g. `t_up_all`) we provide PDF heatmaps of *selected* marker genes for each cluster in [`output/cluster_marker_genes/`](../output/cluster_marker_genes/).

<aside>
The supplied heatmaps are limited to showing a maximum of 100 genes (any more and they fast become illegible without considerable manual tweaking).
We can assist with customised figures for publication.
</aside>

In this section we describe the rationale behind choices made in creating these heatmaps, namely:

- The considerations behind the [Choice of expression values: `logcounts` vs. `corrected`] used for visualization.
- The [Exclusion of certain types of genes] from the provided heatmaps.

We also use the heatmaps to illustrate the [Limitations of the `t_up_all` strategy] to further motivate the need for the other marker gene lists.

## Choice of expression values: `logcounts` vs. `corrected`  

A reminder that the `logcounts` (log-transformed normalized expression values) are the basis for any and all gene-based analyses, like DE-based marker gene detection.
However, we can visualize the results of such an analysis using either the `logcounts` or the `reconstructed` values (batch-corrected `logcounts`).
The advantage of using the `reconstructed` values for visualization is that these will generally eliminate some of the visual differences between the batches (in this case, between the plates), which can result in 'cleaner' figures.
As always, however, it is essential to also examine the 'raw data' (in this case the `logcounts`) to ensure that the 'cleaner' figures do not belie the truth. 
If you opt to use the `reconstructed` data for a figure it is vital that you make note of this (e.g., in the figure caption).

<aside>
See the 'Data integration' section of [`docs/C075_Grant_Coultas.cell_selection.html`](../docs/C075_Grant_Coultas.cell_selection.html) for details of the `reconstructed` data.
</aside>

To give some examples of the difference between visualizing `logcounts` and `reconstructed` values, Figure \@ref(fig:visualization-violin) illustrates this with a violin plot of *Sox17* expression while Figure \@ref(fig:visualization-heatmap) illustrates this with a heatmap of ribosomal protein gene expression.
Together these figures demonstrate that:

- An arbitrary correction algorithm, such as MNN, is not obliged to preserve the magnitude (or even direction) of differences in per-gene expression when attempting to align multiple batches. This can lead to 'exaggerated' differences (Figure \@ref(fig:visualization-violin)) or, conversely, artificial agreement across batches (Figure \@ref(fig:visualization-heatmap)).
- This means that absolute values (like zero) lose their meaning, i.e. `reconstructed` = 0 does not mean anything special. In contrast, `logcounts` = 0 means that there were zero `counts`, i.e. the gene was not detected in that sample.

```{r visualization-violin, fig.cap = "Violin plot of *Sox17* gene expression using the `logcounts` (left) and the `reconstructed` values (right) for cells in each cluster."}
features <- "Sox17"

p1 <- plotExpression(
  sce, 
  features = features,
  x = "cluster",
  exprs_values = "logcounts",
  colour_by = "cluster",
  other_fields = c("sex", "genotype")) + 
  scale_fill_manual(values = cluster_colours) +
  guides(fill = FALSE) +
  ggtitle(features)

p2 <- plotExpression(
  sce, 
  features = features,
  x = "cluster",
  exprs_values = "reconstructed",
  colour_by = "cluster",
  other_fields = c("sex", "genotype")) + 
  scale_fill_manual(values = cluster_colours) +
  guides(fill = FALSE) +
  ggtitle(features)

p1 | p2
```

```{r visualization-heatmap, fig.cap = "Heatmap of some ribosomoal protein gene expression using the `logcounts` (left) and the `reconstructed` values (right) for cells in each cluster. Row (gene) and column (cell) order are preserved across the two heatmaps.", fig.asp = 1.1}
features <- intersect(rownames(sce), ribo_set)[1:50]
o <- order(sce$cluster)

mat <- assay(sce, "logcounts")[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
p1 <- pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)]),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  silent = TRUE,
  fontsize = 6,
  main = "logcounts (row-normalized)")

mat <- assay(sce, "reconstructed")[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
p2 <- pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)]),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  silent = TRUE,
  fontsize = 6,
  main = "reconstructed (row-normalized)")

gridExtra::grid.arrange(p1$gtable, p2$gtable, ncol = 2)
```

## Limitations of the `t_up_all` strategy

Figure \@ref(fig:t-up-all-heatmap) is a heatmap showing the expression of the top-10 marker genes ($FDR < 0.05$) for each cluster from the [`t_up_all`] analysis. 
From this figure we can observe that some clusters do not have any marker genes, and some other clusters only have a small number of marker genes, using the `t_up_all` strategy.
The lack of marker genes for some clusters under the [`t_up_all`] strategy necessitates the other strategies ([`t_up_some`] and [`t_any_any`]).

```{r t-up-all-heatmap, fig.wide = TRUE, fig.cap = "Heatmap of top-10 `t_up_all` marker genes (FDR < 0.05) using the row-normalized `logcounts` for cells in each cluster. Each column is a sample, each row a gene. Note that some clusters have as few as zero upregulated cluster-specific marker genes.", fig.asp = 1.1}
markers <- t_up_all_markers

k <- 10
marker_names <- lapply(markers, function(x) {
  xx <- x[x$FDR < 0.05, ]
  rownames(xx[order(xx$summary.logFC, decreasing = TRUE), ])
})
features <- unlist(lapply(marker_names, head, n = k))
features <- unique(features)

o <- order(sce$cluster, sce$genotype)

mat <- assay(sce, "logcounts")[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_row = data.frame(
    cluster = rep(levels(sce$cluster), times = pmin(lengths(marker_names), k)),
    row.names = features),
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    genotype = genotype_colours),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  main = "logcounts  (row-normalized)",
  fontsize = 6)
```

## Exclusion of certain types of genes

Figure \@ref(fig:t-up-some-cluster-2-heatmap) is a heatmap showing the expression of the top-50 marker genes for cluster 2 from the [`t_up_some`] analysis. 
From this figure we can observe that many of the cluster 2 `t_up_some` marker genes are pseudogenes, while several others are ribosomal protein genes and sex chromosome genes.
This prompted Zoe and Anne to request heatmaps that exclude mitochondrial genes, ribosomal protein genes, sex chromosome genes, and pseudogenes.

```{r t-up-some-cluster-2-heatmap, fig.wide = TRUE, fig.cap = "Heatmap of top-50 `t_up_all` marker genes for cluster 2 using the row-normalized `logcounts` for cells in each cluster. Each column is a sample, each row a gene. Genes are annotated according to `gene_type`.", fig.asp = 1.1}
markers <- t_up_some_markers[[2]]

k <- 50
features <- head(rownames(markers[markers$FDR < 0.05, ]), k)

o <- order(sce$cluster, sce$genotype)

mat <- assay(sce, "logcounts")[features, ]
mat <- mat - rowMeans(mat)
zlim <- c(-max(abs(mat)), max(abs(mat)))
mat[mat < zlim[1]] <- zlim[1]
mat[mat > zlim[2]] <- zlim[2]
pheatmap(
  mat = mat[, o],
  color = hcl.colors(101, "Blue-Red 3"),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_colnames = FALSE,
  annotation_col = data.frame(
    cluster = sce$cluster[o],
    genotype = sce$genotype[o],
    row.names = colnames(sce)[o]),
  annotation_row = data.frame(
    gene_type = dplyr::case_when(
      features %in% mito_set ~ "mito",
      features %in% ribo_set ~ "ribo",
      features %in% sex_set ~ "sex",
      features %in% pseudogene_set ~ "pseudo",
      TRUE ~ "other"),
    row.names = features),
  annotation_colors = list(
    cluster = cluster_colours[levels(sce$cluster)],
    genotype = genotype_colours,
    gene_type = c(
      "pseudo" = "#2297E6",
      "ribo" = "#61D04F",
      "sex" = "#28E2E5",
      "mito" = "#DF536B", 
      "other" = "white")),
  breaks = seq(zlim[1], zlim[2], length.out = 101),
  main = "logcounts  (row-normalized)",
  fontsize = 6)
```

## Choice of gradient colour scale

A commonly used plot is a UMAP plot overlaid with the expression of a selected gene using a gradient colour scale.
For example, Figure \@ref(fig:sox17-umap-viridis) shows this for *Sox17*.

<aside>
Similar plots are shown in Figure \@ref(fig:umap-with-facs) but visualizing cell surface protein expression as measured by FACS index sorting instead of gene expression. 
</aside>

```{r sox17-umap-viridis, fig.cap = "UMAP plot with points coloured by Sox17 expression using the viridis colour gradient.", fig.asp = 0.8}
ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17)) +
  theme_cowplot() +
  scale_colour_viridis_c(option = "viridis")
```

The gene expression measurements are shown using the 'viridis' colour gradient, which is one of the five viridis colour palettes ('viridis', 'magma', 'plasma', 'inferno', and 'cividis').

<aside>
Yep, *viridis* is both the name of the set of palettes *and* the name of one of the individual palettes. ¯\\\_(ツ)\_/¯
</aside>

I strongly recommend using one of the viridis colour palettes to show the expression values because they are designed to be^[See [https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html](https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html) for details.]:

- **Colourful**, spanning as wide a palette as possible so as to make differences easy to see
- **Perceptually uniform**, meaning that values close to each other have similar-appearing colours and values far away from each other have more different-appearing colours, consistently across the range of values
- **Robust to colour blindness**, so that the above properties hold true for people with common forms of colourblindness, as well as in grey scale printing

Figure \@ref(fig:sox-17-lightblue-red) compares the viridis palette to the commonly used lightblue-red colour gradient for these same data, revealing that the lightblue-red colour gradient is not perceptually uniform, e.g.,:

- The bottom half of the lightblue-red colour gradient is perceptually 'compressed'
- It is hard to distinguish values between 3--5 using the lightblue-red gradient
- The perceptual midpoint of the viridis colour gradient matches the midpoint of the 0--5 range of expression values whereas the perceptual midpoint of the lightblue-red colour gradient occurs somewhere below 1

```{r sox-17-lightblue-red, fig.cap = "UMAP plot with points coloured by Sox17 expression using the lightblue-red colour gradient.", fig.asp = 2 / 5}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "viridis")

plot_grid(
  p,
  p + scale_colour_gradient(low = "lightblue", high = "red"),
  ncol = 2)
```

Figure \@ref(fig:sox17-umap-viridis-cvd) to \@ref(fig:sox17-umap-cividis-cvd) simulate what this same plot would like to people with various common colour-deficiencies using each of the viridis palettes and shows that the data are still very much interpretable in all cases using any of these palettes.

```{r sox17-umap-viridis-cvd, fig.cap = "UMAP plot with points coloured by Sox17 expression using the viridis colour gradient and simulating various common colour deficiencies.", fig.asp = 1.2}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "viridis")

my_cvd_grid(p)
```

```{r sox17-umap-magma-cvd, fig.cap = "UMAP plot with points coloured by Sox17 expression using the magma colour gradient and simulating various common colour deficiencies.", fig.asp = 1.2}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "magma")

my_cvd_grid(p)
```

```{r sox17-umap-inferno-cvd, fig.cap = "UMAP plot with points coloured by Sox17 expression using the inferno colour gradient and simulating various common colour deficiencies.", fig.asp = 1.2}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "inferno")

my_cvd_grid(p)
```

```{r sox17-umap-plasma-cvd, fig.cap = "UMAP plot with points coloured by Sox17 expression using the plasma colour gradient and simulating various common colour deficiencies.", fig.asp = 1.2}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "plasma")

my_cvd_grid(p)
```

```{r sox17-umap-cividis-cvd, fig.cap = "UMAP plot with points coloured by Sox17 expression using the cividis colour gradient and simulating various common colour deficiencies.", fig.asp = 1.2}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "cividis")

my_cvd_grid(p)
```

Finally, if a greyscale version of the plot is required then it may be preferable to reverse the direction of the colour gradient (i.e. low is lightgrey, high is black).
Figure \@ref(fig:sox17-umap-greyscale) shows both options.

```{r sox17-umap-greyscale, fig.cap = "UMAP plot with points coloured by Sox17 expression using a greyscale colour gradient running in each direction.", fig.asp = 2 / 5}
p <- ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17), size = 0.5) +
  theme_cowplot(font_size = 10) +
  scale_colour_viridis_c(option = "cividis")

plot_grid(
  p + scale_colour_gradient(low = "black", high = "grey90"),
  p + scale_colour_gradient(low = "grey90", high = "black"),
  ncol = 2)  
```

## Making your own UMAP plot coloured by gene expression

To make your own UMAP plot coloured by gene expression is fairly straightforward.
Here I give some example code demonstrating how I made the UMAP plot showing *Sox17* expression and how you might extend this.
Each plot can be saved as a PDF within RStudio (see the 'Export' tab in the 'Plots' pane.)

To run these examples you will need R version 4.0 and I also recommend using RStudio.
You will first need to install a few R packages (if they are not already installed).
These packages can be installed and loaded as follows:

```{r, eval = FALSE, echo = TRUE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

if (!requireNamespace("scater", quietly = TRUE)) {
    BiocManager::install("scater")
}

if (!requireNamespace("cowplot", quietly = TRUE)) {
    BiocManager::install("cowplot")
}
```

Once these are installed, you can load the required packages into the R session as follows:

```{r, eval = FALSE, echo = FALSE}
library(scater)
library(cowplot)
```

You will then need to load the `data/SCEs/C075_Grant_Coultas.cells_selected.SCE.rds` *SingleCellExperiment* object into your R session.
One way of doing this is to adapt the following code:

<aside>
Alternatively, you can load the object from the 'Files' pane in RStudio.
</aside>

```{r, echo = TRUE, eval = FALSE}
# NOTE: You will need to modify this to where the object is saved on your 
#       computer.
path_to_se <- "data/SCEs/C075_Grant_Coultas.annotated.SCE.rds"

sce <- readRDS(path_to_se)
```

You can then create a plot in your R session.
Let's re-make the the UMAP plot showing *Sox17* expression with the viridis colour palette:

```{r, echo = TRUE, fig.asp = 0.8}
ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17)) +
  theme_cowplot() +
  scale_colour_viridis_c(option = "viridis")
```

To make a version of that plot using the greyscale colour palette requires us to change just the final line of that command:

```{r, echo = TRUE, fig.asp = 0.8}
ggcells(sce, features = "Sox17") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Sox17)) +
  theme_cowplot() +
  scale_colour_gradient(low = "grey90", high = "black")
```

To plot a different gene requires just two small changes, e.g., to plot *Hist1h2ap* expression:

```{r, echo = TRUE, fig.asp = 0.8}
ggcells(sce, features = "Hist1h2ap") + 
  geom_point(aes(x = UMAP.1, y = UMAP.2, colour = Hist1h2ap)) +
  theme_cowplot() +
  scale_colour_viridis_c(option = "viridis")
```

# Summary

We have performed a focused analysis using Zoe's marker genes and the cyclin genes, and identified cluster marker genes for each cluster using several strategies.
Some preliminary high-level summaries of the clusters are as follows:

- Clusters 1, 5, and 7 can be readily distinguished from the other clusters (`t_up_all` analysis).
    - Cluster 1 has very high and distinct expression of many genes, most obviously including *Hist1h2* genes.
    - Cluster 5 has high and fairly distinct expression of Zoe's `Artery, Tip` and `Tip` marker genes.
    - Cluster 7 has very high and distinct expression of many genes, including Zoe's `Artery` marker genes.
- Cluster 2 shares many marker genes with cluster 1 (`t_up_some` analysis).
- Clusters 3, 4, and 6 are perhaps less distinct than the other clusters and share some marker genes with each other and also with clusters 1, 2, 5, and 7 (`t_up_all` and `t_up_some` analyses).
    - Cluster 3 shares some markers with clusters $4 - 7$.
    - Cluster 4 is perhaps the least distinct cluster in terms of marker genes.
    - Cluster 6 shares some markers with clusters 1 and 2 and others with clusters 3, 4, and 7.
    
<aside>
The [Targeted] gene lists may help identify genes that specifically distinguish clusters 2 and 6 as well as clusters 3, 4, and 6. We can add further targeted analyses as required.
</aside>

As mentioned, cluster 4 is perhaps the least distinct cluster in terms of marker genes.
Cluster 4 contains the fewest cells of all the clusters, with these cells coming from all plates, but is is also the cluster most enriched for `Mutant` cells (in terms of the percentage of cells in that cluster with each genotype; Figure \@ref(fig:cluster-barplot)).
Figure \@ref(fig:qcplot-by-cluster) shows that cells in this cluster have on average poorer QC metrics than cells in most other clusters, perhaps meaning that this cluster is comprised of a 'mixed bag' of cells that are hard to place with their 'correct' cluster.

```{r qcplot-by-cluster, fig.cap = "Distributions of various QC metrics for all cells in the data set. This includes the library sizes, number of expressed genes, and proportion of reads mapped to spike-in transcripts or mitochondrial genes.", fig.asp = 2 / 3}
plot_grid(
  ggcells(
    sce,
    exprs_values = "counts",
    aes(x = cluster, y = sum, colour = cluster, fill = cluster)) +
    geom_violin(scale = "width", width = 0.8, alpha = 0.2) +
    ggbeeswarm::geom_quasirandom(size = 0.1) +
    theme_cowplot(font_size = 8) +
    scale_colour_manual(values = cluster_colours) + 
    scale_fill_manual(values = cluster_colours) +
    scale_y_log10(limits = c(1, NA)) +
    annotation_logticks(
      sides = "l",
      short = unit(0.03, "cm"),
      mid = unit(0.06, "cm"),
      long = unit(0.09, "cm")) +
    guides(fill = FALSE, colour = FALSE),
  ggcells(
    sce,
    exprs_values = "counts",
    aes(
      x = cluster,
      y = altexps_ERCC_percent,
      colour = cluster,
      fill = cluster)) +
    geom_violin(scale = "width", width = 0.8, alpha = 0.2) +
    ggbeeswarm::geom_quasirandom(size = 0.1) +
    ylim(0, 100) +
    theme_cowplot(font_size = 8) +
    scale_colour_manual(values = cluster_colours) + 
    scale_fill_manual(values = cluster_colours) +
    guides(fill = FALSE, colour = FALSE),
  ggcells(
    sce,
    exprs_values = "counts",
    aes(
      x = cluster,
      y = detected,
      colour = cluster,
      fill = cluster)) +
    geom_violin(scale = "width", width = 0.8, alpha = 0.2) +
    ggbeeswarm::geom_quasirandom(size = 0.1) +
    theme_cowplot(font_size = 8) +
    scale_colour_manual(values = cluster_colours) + 
    scale_fill_manual(values = cluster_colours) +
    scale_y_log10(limits = c(1, NA)) +
    annotation_logticks(
      sides = "l",
      short = unit(0.03, "cm"),
      mid = unit(0.06, "cm"),
      long = unit(0.09, "cm")) +
    guides(fill = FALSE, colour = FALSE),
  ggcells(
    sce,
    exprs_values = "counts",
    aes(
      x = cluster,
      y = subsets_Mt_percent,
      colour = cluster,
      fill = cluster)) +
    geom_violin(scale = "width", width = 0.8, alpha = 0.2) +
    ggbeeswarm::geom_quasirandom(size = 0.1) +
    ylim(0, 100) +
    theme_cowplot(font_size = 8) +
    scale_colour_manual(values = cluster_colours) + 
    scale_fill_manual(values = cluster_colours) +
    guides(fill = FALSE, colour = FALSE),
  ncol = 2,
  align = "v")
```
    
This preliminary summary is just that, preliminary.
The cluster marker gene lists can be analysed like any other gene list, such as by using gene set enrichment analysis against the Gene Ontology (GO) or Kyoto Encyclopedia of Genes and Genomes (KEGG) databases.
Used this way, the provided cluster marker gene lists can help develop biologically relevant, meaningful, and useful labels for cells/clusters in this dataset.

<aside>
We strongly recommend that you visualise the expression patterns of any genes of interest to better understand what is driving their inclusion in the marker gene lists.
</aside>

# Concluding remarks

Zoe requested that clusters 3, 4, and 6 be aggregated for the purposes of differential analyses.

```{r}
colLabels(sce) <- factor(
  dplyr::case_when(
    sce$cluster %in% c("3", "4", "6") ~ "clusters_3_4_6",
    TRUE ~ paste0("cluster_", sce$cluster)),
  levels = 
    c("cluster_1", "cluster_2", "clusters_3_4_6", "cluster_5", "cluster_7"))
sce$label_colours <- setNames(
  dplyr::case_when(
    sce$cluster %in% c("3", "4", "6") ~ "dodgerblue",
    TRUE ~ sce$cluster_colours),
  colLabels(sce))
label_colours <- setNames(
  unique(sce$label_colours),
  unique(names(sce$label_colours)))
```

```{r}
saveRDS(
  sce,
  here("data", "SCEs", "C075_Grant_Coultas.annotated.SCE.rds"),
  compress = "xz")
```

The processed *SingleCellExperiment* object is available (see [`data/SCEs/C075_Grant_Coultas.annotated.SCE.rds`](../data/SCEs/C075_Grant_Coultas.annotated.SCE.rds)).
This will be used in downstream analyses, e.g., differential expression analysis between conditions within each cluster and differential abundance analyses between conditions for each cluster.

# Additional information {.appendix}

The following are available on request:

- Full CSV tables of any data presented.
- PDF/PNG files of any static plots.

# Session info {.appendix}

<summary>The analysis and this document were prepared using the following software (click triangle to expand)</summary>
<details>

```{r}
sessioninfo::session_info()
```
